// test-maker/dist/123.js
exports['id'] = 0x7b, exports['ids'] = [0x7b], exports['modules'] = {
    0x93bc: (_0x47bbc4, _0x38af45, _0x35d68f) => {
        _0x38af45['formatArgs'] = _0x17743b, _0x38af45['save'] = _0x1c2f10, _0x38af45['load'] = _0x267ba5, _0x38af45['useColors'] = _0x438275, _0x38af45['storage'] = _0x360cce(), _0x38af45['destroy'] = ((() => {
            let _0x234613 = ![];
            return () => {
                !_0x234613 && (_0x234613 = !![], console['warn']('Instance\x20method\x20`debug.destroy()`\x20is\x20deprecated\x20and\x20no\x20longer\x20does\x20anything.\x20It\x20will\x20be\x20removed\x20in\x20the\x20next\x20major\x20version\x20of\x20`debug`.'));
            };
        })()), _0x38af45['colors'] = [
            '#0000CC',
            '#0000FF',
            '#0033CC',
            '#0033FF',
            '#0066CC',
            '#0066FF',
            '#0099CC',
            '#0099FF',
            '#00CC00',
            '#00CC33',
            '#00CC66',
            '#00CC99',
            '#00CCCC',
            '#00CCFF',
            '#3300CC',
            '#3300FF',
            '#3333CC',
            '#3333FF',
            '#3366CC',
            '#3366FF',
            '#3399CC',
            '#3399FF',
            '#33CC00',
            '#33CC33',
            '#33CC66',
            '#33CC99',
            '#33CCCC',
            '#33CCFF',
            '#6600CC',
            '#6600FF',
            '#6633CC',
            '#6633FF',
            '#66CC00',
            '#66CC33',
            '#9900CC',
            '#9900FF',
            '#9933CC',
            '#9933FF',
            '#99CC00',
            '#99CC33',
            '#CC0000',
            '#CC0033',
            '#CC0066',
            '#CC0099',
            '#CC00CC',
            '#CC00FF',
            '#CC3300',
            '#CC3333',
            '#CC3366',
            '#CC3399',
            '#CC33CC',
            '#CC33FF',
            '#CC6600',
            '#CC6633',
            '#CC9900',
            '#CC9933',
            '#CCCC00',
            '#CCCC33',
            '#FF0000',
            '#FF0033',
            '#FF0066',
            '#FF0099',
            '#FF00CC',
            '#FF00FF',
            '#FF3300',
            '#FF3333',
            '#FF3366',
            '#FF3399',
            '#FF33CC',
            '#FF33FF',
            '#FF6600',
            '#FF6633',
            '#FF9900',
            '#FF9933',
            '#FFCC00',
            '#FFCC33'
        ];
        function _0x438275() {
            if (typeof window !== 'undefined' && window['process'] && (window['process']['type'] === 'renderer' || window['process']['__nwjs']))
                return !![];
            if (typeof navigator !== 'undefined' && navigator['userAgent'] && navigator['userAgent']['toLowerCase']()['match'](/(edge|trident)\/(\d+)/))
                return ![];
            return typeof document !== 'undefined' && document['documentElement'] && document['documentElement']['style'] && document['documentElement']['style']['WebkitAppearance'] || typeof window !== 'undefined' && window['console'] && (window['console']['firebug'] || window['console']['exception'] && window['console']['table']) || typeof navigator !== 'undefined' && navigator['userAgent'] && navigator['userAgent']['toLowerCase']()['match'](/firefox\/(\d+)/) && parseInt(RegExp['$1'], 0xa) >= 0x1f || typeof navigator !== 'undefined' && navigator['userAgent'] && navigator['userAgent']['toLowerCase']()['match'](/applewebkit\/(\d+)/);
        }
        function _0x17743b(_0x474261) {
            _0x474261[0x0] = (this['useColors'] ? '%c' : '') + this['namespace'] + (this['useColors'] ? '\x20%c' : '\x20') + _0x474261[0x0] + (this['useColors'] ? '%c\x20' : '\x20') + '+' + _0x47bbc4['exports']['humanize'](this['diff']);
            if (!this['useColors'])
                return;
            const _0x2931c1 = 'color:\x20' + this['color'];
            _0x474261['splice'](0x1, 0x0, _0x2931c1, 'color:\x20inherit');
            let _0x526ef6 = 0x0, _0x2138ee = 0x0;
            _0x474261[0x0]['replace'](/%[a-zA-Z%]/g, _0x1ddf96 => {
                if (_0x1ddf96 === '%%')
                    return;
                _0x526ef6++, _0x1ddf96 === '%c' && (_0x2138ee = _0x526ef6);
            }), _0x474261['splice'](_0x2138ee, 0x0, _0x2931c1);
        }
        _0x38af45['log'] = console['debug'] || console['log'] || (() => {
        });
        function _0x1c2f10(_0x31e58d) {
            try {
                _0x31e58d ? _0x38af45['storage']['setItem']('debug', _0x31e58d) : _0x38af45['storage']['removeItem']('debug');
            } catch (_0x536657) {
            }
        }
        function _0x267ba5() {
            let _0x70ff50;
            try {
                _0x70ff50 = _0x38af45['storage']['getItem']('debug');
            } catch (_0x148108) {
            }
            return !_0x70ff50 && typeof process !== 'undefined' && 'env' in process && (_0x70ff50 = process['env']['DEBUG']), _0x70ff50;
        }
        function _0x360cce() {
            try {
                return localStorage;
            } catch (_0x1e874d) {
            }
        }
        _0x47bbc4['exports'] = _0x35d68f(0x4817)(_0x38af45);
        const {formatters: _0xe4f9e9} = _0x47bbc4['exports'];
        _0xe4f9e9['j'] = function (_0x56d7e0) {
            try {
                return JSON['stringify'](_0x56d7e0);
            } catch (_0x2edf16) {
                return '[UnexpectedJSONParseError]:\x20' + _0x2edf16['message'];
            }
        };
    },
    0x4817: (_0x509be5, _0x4294af, _0x5afdf0) => {
        function _0x5aad82(_0xefa9da) {
            _0x51ed9e['debug'] = _0x51ed9e, _0x51ed9e['default'] = _0x51ed9e, _0x51ed9e['coerce'] = _0x3ae71a, _0x51ed9e['disable'] = _0x3850da, _0x51ed9e['enable'] = _0x861377, _0x51ed9e['enabled'] = _0x4ed06e, _0x51ed9e['humanize'] = _0x5afdf0(0xd552), _0x51ed9e['destroy'] = _0x43aea1, Object['keys'](_0xefa9da)['forEach'](_0x209c3a => {
                _0x51ed9e[_0x209c3a] = _0xefa9da[_0x209c3a];
            }), _0x51ed9e['names'] = [], _0x51ed9e['skips'] = [], _0x51ed9e['formatters'] = {};
            function _0x47f138(_0x132df0) {
                let _0xa9ceca = 0x0;
                for (let _0x78a608 = 0x0; _0x78a608 < _0x132df0['length']; _0x78a608++) {
                    _0xa9ceca = (_0xa9ceca << 0x5) - _0xa9ceca + _0x132df0['charCodeAt'](_0x78a608), _0xa9ceca |= 0x0;
                }
                return _0x51ed9e['colors'][Math['abs'](_0xa9ceca) % _0x51ed9e['colors']['length']];
            }
            _0x51ed9e['selectColor'] = _0x47f138;
            function _0x51ed9e(_0x57e96b) {
                let _0x3c124e, _0x52e982 = null, _0x163f99, _0x26733c;
                function _0x47d60c(..._0x2da7a5) {
                    if (!_0x47d60c['enabled'])
                        return;
                    const _0x342f2b = _0x47d60c, _0x32baa4 = Number(new Date()), _0x4f703d = _0x32baa4 - (_0x3c124e || _0x32baa4);
                    _0x342f2b['diff'] = _0x4f703d, _0x342f2b['prev'] = _0x3c124e, _0x342f2b['curr'] = _0x32baa4, _0x3c124e = _0x32baa4, _0x2da7a5[0x0] = _0x51ed9e['coerce'](_0x2da7a5[0x0]);
                    typeof _0x2da7a5[0x0] !== 'string' && _0x2da7a5['unshift']('%O');
                    let _0x4c7c3f = 0x0;
                    _0x2da7a5[0x0] = _0x2da7a5[0x0]['replace'](/%([a-zA-Z%])/g, (_0x4eede4, _0x15d8c2) => {
                        if (_0x4eede4 === '%%')
                            return '%';
                        _0x4c7c3f++;
                        const _0x5e090f = _0x51ed9e['formatters'][_0x15d8c2];
                        if (typeof _0x5e090f === 'function') {
                            const _0xd9bacb = _0x2da7a5[_0x4c7c3f];
                            _0x4eede4 = _0x5e090f['call'](_0x342f2b, _0xd9bacb), _0x2da7a5['splice'](_0x4c7c3f, 0x1), _0x4c7c3f--;
                        }
                        return _0x4eede4;
                    }), _0x51ed9e['formatArgs']['call'](_0x342f2b, _0x2da7a5);
                    const _0x3e5d17 = _0x342f2b['log'] || _0x51ed9e['log'];
                    _0x3e5d17['apply'](_0x342f2b, _0x2da7a5);
                }
                return _0x47d60c['namespace'] = _0x57e96b, _0x47d60c['useColors'] = _0x51ed9e['useColors'](), _0x47d60c['color'] = _0x51ed9e['selectColor'](_0x57e96b), _0x47d60c['extend'] = _0x567c86, _0x47d60c['destroy'] = _0x51ed9e['destroy'], Object['defineProperty'](_0x47d60c, 'enabled', {
                    'enumerable': !![],
                    'configurable': ![],
                    'get': () => {
                        if (_0x52e982 !== null)
                            return _0x52e982;
                        return _0x163f99 !== _0x51ed9e['namespaces'] && (_0x163f99 = _0x51ed9e['namespaces'], _0x26733c = _0x51ed9e['enabled'](_0x57e96b)), _0x26733c;
                    },
                    'set': _0x25159d => {
                        _0x52e982 = _0x25159d;
                    }
                }), typeof _0x51ed9e['init'] === 'function' && _0x51ed9e['init'](_0x47d60c), _0x47d60c;
            }
            function _0x567c86(_0x338ffa, _0x488691) {
                const _0x2e784c = _0x51ed9e(this['namespace'] + (typeof _0x488691 === 'undefined' ? ':' : _0x488691) + _0x338ffa);
                return _0x2e784c['log'] = this['log'], _0x2e784c;
            }
            function _0x861377(_0x93cfaa) {
                _0x51ed9e['save'](_0x93cfaa), _0x51ed9e['namespaces'] = _0x93cfaa, _0x51ed9e['names'] = [], _0x51ed9e['skips'] = [];
                let _0x1d1907;
                const _0x12caba = (typeof _0x93cfaa === 'string' ? _0x93cfaa : '')['split'](/[\s,]+/), _0x1318f4 = _0x12caba['length'];
                for (_0x1d1907 = 0x0; _0x1d1907 < _0x1318f4; _0x1d1907++) {
                    if (!_0x12caba[_0x1d1907])
                        continue;
                    _0x93cfaa = _0x12caba[_0x1d1907]['replace'](/\*/g, '.*?'), _0x93cfaa[0x0] === '-' ? _0x51ed9e['skips']['push'](new RegExp('^' + _0x93cfaa['slice'](0x1) + '$')) : _0x51ed9e['names']['push'](new RegExp('^' + _0x93cfaa + '$'));
                }
            }
            function _0x3850da() {
                const _0x52558c = [
                    ..._0x51ed9e['names']['map'](_0x265742),
                    ..._0x51ed9e['skips']['map'](_0x265742)['map'](_0x5f4b90 => '-' + _0x5f4b90)
                ]['join'](',');
                return _0x51ed9e['enable'](''), _0x52558c;
            }
            function _0x4ed06e(_0x4d48e6) {
                if (_0x4d48e6[_0x4d48e6['length'] - 0x1] === '*')
                    return !![];
                let _0x12a2e3, _0xc2745d;
                for (_0x12a2e3 = 0x0, _0xc2745d = _0x51ed9e['skips']['length']; _0x12a2e3 < _0xc2745d; _0x12a2e3++) {
                    if (_0x51ed9e['skips'][_0x12a2e3]['test'](_0x4d48e6))
                        return ![];
                }
                for (_0x12a2e3 = 0x0, _0xc2745d = _0x51ed9e['names']['length']; _0x12a2e3 < _0xc2745d; _0x12a2e3++) {
                    if (_0x51ed9e['names'][_0x12a2e3]['test'](_0x4d48e6))
                        return !![];
                }
                return ![];
            }
            function _0x265742(_0x3f4be6) {
                return _0x3f4be6['toString']()['substring'](0x2, _0x3f4be6['toString']()['length'] - 0x2)['replace'](/\.\*\?$/, '*');
            }
            function _0x3ae71a(_0x41333d) {
                if (_0x41333d instanceof Error)
                    return _0x41333d['stack'] || _0x41333d['message'];
                return _0x41333d;
            }
            function _0x43aea1() {
                console['warn']('Instance\x20method\x20`debug.destroy()`\x20is\x20deprecated\x20and\x20no\x20longer\x20does\x20anything.\x20It\x20will\x20be\x20removed\x20in\x20the\x20next\x20major\x20version\x20of\x20`debug`.');
            }
            return _0x51ed9e['enable'](_0x51ed9e['load']()), _0x51ed9e;
        }
        _0x509be5['exports'] = _0x5aad82;
    },
    0x566b: (_0x31ca7a, _0x351dac, _0x55480a) => {
        typeof process === 'undefined' || process['type'] === 'renderer' || process['browser'] === !![] || process['__nwjs'] ? _0x31ca7a['exports'] = _0x55480a(0x93bc) : _0x31ca7a['exports'] = _0x55480a(0xc933);
    },
    0xc933: (_0x62708d, _0x4ea277, _0x1b0212) => {
        const _0xeabe4d = _0x1b0212(0x129c0), _0x592956 = _0x1b0212(0x1206d);
        _0x4ea277['init'] = _0x5601da, _0x4ea277['log'] = _0x5f3674, _0x4ea277['formatArgs'] = _0x50a268, _0x4ea277['save'] = _0x161532, _0x4ea277['load'] = _0x5edb2b, _0x4ea277['useColors'] = _0x571038, _0x4ea277['destroy'] = _0x592956['deprecate'](() => {
        }, 'Instance\x20method\x20`debug.destroy()`\x20is\x20deprecated\x20and\x20no\x20longer\x20does\x20anything.\x20It\x20will\x20be\x20removed\x20in\x20the\x20next\x20major\x20version\x20of\x20`debug`.'), _0x4ea277['colors'] = [
            0x6,
            0x2,
            0x3,
            0x4,
            0x5,
            0x1
        ];
        try {
            const _0x318285 = _0x1b0212(0x9583);
            _0x318285 && (_0x318285['stderr'] || _0x318285)['level'] >= 0x2 && (_0x4ea277['colors'] = [
                0x14,
                0x15,
                0x1a,
                0x1b,
                0x20,
                0x21,
                0x26,
                0x27,
                0x28,
                0x29,
                0x2a,
                0x2b,
                0x2c,
                0x2d,
                0x38,
                0x39,
                0x3e,
                0x3f,
                0x44,
                0x45,
                0x4a,
                0x4b,
                0x4c,
                0x4d,
                0x4e,
                0x4f,
                0x50,
                0x51,
                0x5c,
                0x5d,
                0x62,
                0x63,
                0x70,
                0x71,
                0x80,
                0x81,
                0x86,
                0x87,
                0x94,
                0x95,
                0xa0,
                0xa1,
                0xa2,
                0xa3,
                0xa4,
                0xa5,
                0xa6,
                0xa7,
                0xa8,
                0xa9,
                0xaa,
                0xab,
                0xac,
                0xad,
                0xb2,
                0xb3,
                0xb8,
                0xb9,
                0xc4,
                0xc5,
                0xc6,
                0xc7,
                0xc8,
                0xc9,
                0xca,
                0xcb,
                0xcc,
                0xcd,
                0xce,
                0xcf,
                0xd0,
                0xd1,
                0xd6,
                0xd7,
                0xdc,
                0xdd
            ]);
        } catch (_0x1bac5c) {
        }
        _0x4ea277['inspectOpts'] = Object['keys'](process['env'])['filter'](_0x5242f5 => {
            return /^debug_/i['test'](_0x5242f5);
        })['reduce']((_0x5cd2c5, _0x59cabc) => {
            const _0xfd17bd = _0x59cabc['substring'](0x6)['toLowerCase']()['replace'](/_([a-z])/g, (_0x13ff4a, _0x17eb36) => {
                return _0x17eb36['toUpperCase']();
            });
            let _0x574f42 = process['env'][_0x59cabc];
            if (/^(yes|on|true|enabled)$/i['test'](_0x574f42))
                _0x574f42 = !![];
            else {
                if (/^(no|off|false|disabled)$/i['test'](_0x574f42))
                    _0x574f42 = ![];
                else
                    _0x574f42 === 'null' ? _0x574f42 = null : _0x574f42 = Number(_0x574f42);
            }
            return _0x5cd2c5[_0xfd17bd] = _0x574f42, _0x5cd2c5;
        }, {});
        function _0x571038() {
            return 'colors' in _0x4ea277['inspectOpts'] ? Boolean(_0x4ea277['inspectOpts']['colors']) : _0xeabe4d['isatty'](process['stderr']['fd']);
        }
        function _0x50a268(_0x582cb4) {
            const {
                namespace: _0x420e45,
                useColors: _0xba9a8c
            } = this;
            if (_0xba9a8c) {
                const _0x1291b0 = this['color'], _0x34dca3 = '\x1b[3' + (_0x1291b0 < 0x8 ? _0x1291b0 : '8;5;' + _0x1291b0), _0x5ca2ec = '\x20\x20' + _0x34dca3 + ';1m' + _0x420e45 + '\x20\x1b[0m';
                _0x582cb4[0x0] = _0x5ca2ec + _0x582cb4[0x0]['split']('\x0a')['join']('\x0a' + _0x5ca2ec), _0x582cb4['push'](_0x34dca3 + 'm+' + _0x62708d['exports']['humanize'](this['diff']) + '\x1b[0m');
            } else
                _0x582cb4[0x0] = _0x51a906() + _0x420e45 + '\x20' + _0x582cb4[0x0];
        }
        function _0x51a906() {
            if (_0x4ea277['inspectOpts']['hideDate'])
                return '';
            return new Date()['toISOString']() + '\x20';
        }
        function _0x5f3674(..._0x49039a) {
            return process['stderr']['write'](_0x592956['format'](..._0x49039a) + '\x0a');
        }
        function _0x161532(_0x4703bd) {
            _0x4703bd ? process['env']['DEBUG'] = _0x4703bd : delete process['env']['DEBUG'];
        }
        function _0x5edb2b() {
            return process['env']['DEBUG'];
        }
        function _0x5601da(_0x385e76) {
            _0x385e76['inspectOpts'] = {};
            const _0x13d247 = Object['keys'](_0x4ea277['inspectOpts']);
            for (let _0x3266f9 = 0x0; _0x3266f9 < _0x13d247['length']; _0x3266f9++) {
                _0x385e76['inspectOpts'][_0x13d247[_0x3266f9]] = _0x4ea277['inspectOpts'][_0x13d247[_0x3266f9]];
            }
        }
        _0x62708d['exports'] = _0x1b0212(0x4817)(_0x4ea277);
        const {formatters: _0x1472c3} = _0x62708d['exports'];
        _0x1472c3['o'] = function (_0x2f34d1) {
            return this['inspectOpts']['colors'] = this['useColors'], _0x592956['inspect'](_0x2f34d1, this['inspectOpts'])['split']('\x0a')['map'](_0x4c85f1 => _0x4c85f1['trim']())['join']('\x20');
        }, _0x1472c3['O'] = function (_0x5cd469) {
            return this['inspectOpts']['colors'] = this['useColors'], _0x592956['inspect'](_0x5cd469, this['inspectOpts']);
        };
    },
    0xd552: _0xd5a58c => {
        var _0x42f1e4 = 0x3e8, _0x2721c4 = _0x42f1e4 * 0x3c, _0x4f0fff = _0x2721c4 * 0x3c, _0x17ed42 = _0x4f0fff * 0x18, _0x3d6c4d = _0x17ed42 * 0x7, _0x4330cd = _0x17ed42 * 365.25;
        _0xd5a58c['exports'] = function (_0x5237e3, _0x45ddbd) {
            _0x45ddbd = _0x45ddbd || {};
            var _0x3b9507 = typeof _0x5237e3;
            if (_0x3b9507 === 'string' && _0x5237e3['length'] > 0x0)
                return _0x2b529c(_0x5237e3);
            else {
                if (_0x3b9507 === 'number' && isFinite(_0x5237e3))
                    return _0x45ddbd['long'] ? _0xa5e6d3(_0x5237e3) : _0x2c8fee(_0x5237e3);
            }
            throw new Error('val\x20is\x20not\x20a\x20non-empty\x20string\x20or\x20a\x20valid\x20number.\x20val=' + JSON['stringify'](_0x5237e3));
        };
        function _0x2b529c(_0x1727bc) {
            _0x1727bc = String(_0x1727bc);
            if (_0x1727bc['length'] > 0x64)
                return;
            var _0xaf3676 = /^(-?(?:\d+)?\.?\d+) *(milliseconds?|msecs?|ms|seconds?|secs?|s|minutes?|mins?|m|hours?|hrs?|h|days?|d|weeks?|w|years?|yrs?|y)?$/i['exec'](_0x1727bc);
            if (!_0xaf3676)
                return;
            var _0xdfbc97 = parseFloat(_0xaf3676[0x1]), _0x1c53ab = (_0xaf3676[0x2] || 'ms')['toLowerCase']();
            switch (_0x1c53ab) {
            case 'years':
            case 'year':
            case 'yrs':
            case 'yr':
            case 'y':
                return _0xdfbc97 * _0x4330cd;
            case 'weeks':
            case 'week':
            case 'w':
                return _0xdfbc97 * _0x3d6c4d;
            case 'days':
            case 'day':
            case 'd':
                return _0xdfbc97 * _0x17ed42;
            case 'hours':
            case 'hour':
            case 'hrs':
            case 'hr':
            case 'h':
                return _0xdfbc97 * _0x4f0fff;
            case 'minutes':
            case 'minute':
            case 'mins':
            case 'min':
            case 'm':
                return _0xdfbc97 * _0x2721c4;
            case 'seconds':
            case 'second':
            case 'secs':
            case 'sec':
            case 's':
                return _0xdfbc97 * _0x42f1e4;
            case 'milliseconds':
            case 'millisecond':
            case 'msecs':
            case 'msec':
            case 'ms':
                return _0xdfbc97;
            default:
                return undefined;
            }
        }
        function _0x2c8fee(_0x4e3bc7) {
            var _0x1de5e2 = Math['abs'](_0x4e3bc7);
            if (_0x1de5e2 >= _0x17ed42)
                return Math['round'](_0x4e3bc7 / _0x17ed42) + 'd';
            if (_0x1de5e2 >= _0x4f0fff)
                return Math['round'](_0x4e3bc7 / _0x4f0fff) + 'h';
            if (_0x1de5e2 >= _0x2721c4)
                return Math['round'](_0x4e3bc7 / _0x2721c4) + 'm';
            if (_0x1de5e2 >= _0x42f1e4)
                return Math['round'](_0x4e3bc7 / _0x42f1e4) + 's';
            return _0x4e3bc7 + 'ms';
        }
        function _0xa5e6d3(_0x402a30) {
            var _0x5c7857 = Math['abs'](_0x402a30);
            if (_0x5c7857 >= _0x17ed42)
                return _0x1b9e41(_0x402a30, _0x5c7857, _0x17ed42, 'day');
            if (_0x5c7857 >= _0x4f0fff)
                return _0x1b9e41(_0x402a30, _0x5c7857, _0x4f0fff, 'hour');
            if (_0x5c7857 >= _0x2721c4)
                return _0x1b9e41(_0x402a30, _0x5c7857, _0x2721c4, 'minute');
            if (_0x5c7857 >= _0x42f1e4)
                return _0x1b9e41(_0x402a30, _0x5c7857, _0x42f1e4, 'second');
            return _0x402a30 + '\x20ms';
        }
        function _0x1b9e41(_0x390d15, _0x2dbae7, _0x19c483, _0x4c491a) {
            var _0x5a0833 = _0x2dbae7 >= _0x19c483 * 1.5;
            return Math['round'](_0x390d15 / _0x19c483) + '\x20' + _0x4c491a + (_0x5a0833 ? 's' : '');
        }
    }
};
;
