// test-maker/dist/131.js
exports['id'] = 0x83, exports['ids'] = [0x83], exports['modules'] = {
    0x1869: _0x1960aa => {
        'use strict';
        function _0x15faa0() {
            this['_events_'] = {}, this['publish'] = this['trigger'] = this['emit'] = _0x31ff75, this['subscribe'] = this['on'] = _0x5d26f7, this['once'] = _0x42c3a0, this['unSubscribe'] = this['off'] = _0x1b39ca, this['emit$'] = _0x2cc072;
            function _0x5d26f7(_0x2834d7, _0x366644, _0x4ea2c5) {
                if (!_0x366644)
                    throw new ReferenceError('handler\x20not\x20defined.');
                return !this['_events_'][_0x2834d7] && (this['_events_'][_0x2834d7] = []), _0x4ea2c5 && (_0x366644['_once_'] = _0x4ea2c5), this['_events_'][_0x2834d7]['push'](_0x366644), this;
            }
            function _0x42c3a0(_0x246e64, _0xe7b306) {
                return this['on'](_0x246e64, _0xe7b306, !![]);
            }
            function _0x1b39ca(_0x389cbb, _0x388567) {
                if (!this['_events_'][_0x389cbb])
                    return this;
                if (!_0x388567)
                    throw new ReferenceError('handler\x20not\x20defined.\x20if\x20you\x20wish\x20to\x20remove\x20all\x20handlers\x20from\x20the\x20event\x20please\x20pass\x20\x22*\x22\x20as\x20the\x20handler');
                if (_0x388567 == '*')
                    return delete this['_events_'][_0x389cbb], this;
                var _0x15b990 = this['_events_'][_0x389cbb];
                while (_0x15b990['includes'](_0x388567)) {
                    _0x15b990['splice'](_0x15b990['indexOf'](_0x388567), 0x1);
                }
                return _0x15b990['length'] < 0x1 && delete this['_events_'][_0x389cbb], this;
            }
            function _0x31ff75(_0x78ce8d) {
                this['emit$']['apply'](this, arguments);
                if (!this['_events_'][_0x78ce8d])
                    return this;
                arguments['splice'] = Array['prototype']['splice'], arguments['splice'](0x0, 0x1);
                var _0x47d13b = this['_events_'][_0x78ce8d], _0x1e5e40 = [];
                for (var _0x13ecd3 in _0x47d13b) {
                    var _0x48b700 = _0x47d13b[_0x13ecd3];
                    _0x48b700['apply'](this, arguments), _0x48b700['_once_'] && _0x1e5e40['push'](_0x48b700);
                }
                for (var _0x13ecd3 in _0x1e5e40) {
                    this['off'](_0x78ce8d, _0x1e5e40[_0x13ecd3]);
                }
                return this;
            }
            function _0x2cc072(_0x45d39c, _0x3897ec) {
                if (!this['_events_']['*'])
                    return this;
                var _0x3633b9 = this['_events_']['*'];
                _0x3897ec['shift'] = Array['prototype']['shift'], _0x3897ec['shift'](_0x45d39c);
                for (var _0x13ba0c of _0x3633b9) {
                    _0x13ba0c['apply'](this, _0x3897ec);
                }
                return this;
            }
            return this;
        }
        !Array['prototype']['includes'] && (Array['prototype']['includes'] = function (_0x2b4499) {
            'use strict';
            if (this == null)
                throw new TypeError('Array.prototype.includes\x20called\x20on\x20null\x20or\x20undefined');
            var _0x57db26 = Object(this), _0x40c046 = parseInt(_0x57db26['length'], 0xa) || 0x0;
            if (_0x40c046 === 0x0)
                return ![];
            var _0x230c16 = parseInt(arguments[0x1], 0xa) || 0x0, _0x1bf0a5;
            _0x230c16 >= 0x0 ? _0x1bf0a5 = _0x230c16 : (_0x1bf0a5 = _0x40c046 + _0x230c16, _0x1bf0a5 < 0x0 && (_0x1bf0a5 = 0x0));
            var _0x1b8ef8;
            while (_0x1bf0a5 < _0x40c046) {
                _0x1b8ef8 = _0x57db26[_0x1bf0a5];
                if (_0x2b4499 === _0x1b8ef8 || _0x2b4499 !== _0x2b4499 && _0x1b8ef8 !== _0x1b8ef8)
                    return !![];
                _0x1bf0a5++;
            }
            return ![];
        }), _0x1960aa['exports'] = _0x15faa0;
    },
    0xb916: _0x16b3af => {
        'use strict';
        class _0x296cee {
            constructor(_0xe0479e) {
                this['_events_'] = {}, this['publish'] = this['trigger'] = this['emit'], this['subscribe'] = this['on'], this['unSubscribe'] = this['off'];
            }
            ['on'](_0x48b9b9, _0x5e06e5, _0x16c69d) {
                if (!_0x5e06e5)
                    throw new ReferenceError('handler\x20not\x20defined.');
                return !this['_events_'][_0x48b9b9] && (this['_events_'][_0x48b9b9] = []), _0x16c69d && (_0x5e06e5['_once_'] = _0x16c69d), this['_events_'][_0x48b9b9]['push'](_0x5e06e5), this;
            }
            ['once'](_0x37c2fe, _0x15e88a) {
                return this['on'](_0x37c2fe, _0x15e88a, !![]);
            }
            ['off'](_0x170a69, _0x12c3bc) {
                if (!this['_events_'][_0x170a69])
                    return this;
                if (!_0x12c3bc)
                    throw new ReferenceError('handler\x20not\x20defined.\x20if\x20you\x20wish\x20to\x20remove\x20all\x20handlers\x20from\x20the\x20event\x20please\x20pass\x20\x22*\x22\x20as\x20the\x20handler');
                if (_0x12c3bc == '*')
                    return delete this['_events_'][_0x170a69], this;
                const _0xa72ff9 = this['_events_'][_0x170a69];
                while (_0xa72ff9['includes'](_0x12c3bc)) {
                    _0xa72ff9['splice'](_0xa72ff9['indexOf'](_0x12c3bc), 0x1);
                }
                return _0xa72ff9['length'] < 0x1 && delete this['_events_'][_0x170a69], this;
            }
            ['emit'](_0x43e464, ..._0x42f40f) {
                if (!this['_events_'][_0x43e464])
                    return this['emit$'](_0x43e464, ..._0x42f40f);
                const _0x51a540 = this['_events_'][_0x43e464], _0x134c08 = [];
                for (let _0x1f82df of _0x51a540) {
                    _0x1f82df['apply'](this, _0x42f40f), _0x1f82df['_once_'] && _0x134c08['push'](_0x1f82df);
                }
                for (let _0x5b75d9 of _0x134c08) {
                    this['off'](_0x43e464, _0x5b75d9);
                }
                return this['emit$'](_0x43e464, ..._0x42f40f);
            }
            ['emit$'](_0x1d6782, ..._0x4615de) {
                if (!this['_events_']['*'])
                    return this;
                const _0x1514e9 = this['_events_']['*'];
                for (let _0x3be63e of _0x1514e9) {
                    _0x3be63e['call'](this, _0x1d6782, ..._0x4615de);
                }
                return this;
            }
        }
        _0x16b3af['exports'] = _0x296cee;
    },
    0x35e: (_0x8cf921, _0x2af967, _0x528f4f) => {
        'use strict';
        let _0x105e56 = _0x528f4f(0x1869);
        process['version'][0x1] > 0x5 && (_0x105e56 = _0x528f4f(0xb916)), _0x8cf921['exports'] = _0x105e56;
    },
    0x100ed: _0x4a09a3 => {
        function _0x29b02f() {
            Object['defineProperties'](this, {
                'data': {
                    'enumerable': !![],
                    'get': _0x4074b2,
                    'set': _0x53a0b2
                },
                'type': {
                    'enumerable': !![],
                    'get': _0x54ad9d,
                    'set': _0x48dffe
                },
                'load': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x420a1f
                },
                'JSON': {
                    'enumerable': !![],
                    'get': _0x69e1c6
                }
            });
            var _0x51774a = '', _0x6b8a89 = {};
            function _0x54ad9d() {
                return _0x51774a;
            }
            function _0x4074b2() {
                return _0x6b8a89;
            }
            function _0x69e1c6() {
                return JSON['stringify']({
                    'type': _0x51774a,
                    'data': _0x6b8a89
                });
            }
            function _0x48dffe(_0x542d61) {
                _0x51774a = _0x542d61;
            }
            function _0x53a0b2(_0x4a9e68) {
                _0x6b8a89 = _0x4a9e68;
            }
            function _0x420a1f(_0xb41b92) {
                try {
                    var _0xb41b92 = JSON['parse'](_0xb41b92);
                    _0x51774a = _0xb41b92['type'], _0x6b8a89 = _0xb41b92['data'];
                } catch (_0x38e462) {
                    var _0x3a35eb = _0xb41b92;
                    _0x51774a = 'error', _0x6b8a89 = {
                        'message': 'Invalid\x20JSON\x20response\x20format',
                        'err': _0x38e462,
                        'response': _0x3a35eb
                    };
                }
            }
        }
        _0x4a09a3['exports'] = _0x29b02f;
    },
    0x10b3: _0x5ac194 => {
        function _0x27604c(_0xe030b1) {
            Object['defineProperties'](this, {
                'add': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x498a8c
                },
                'next': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x5b9670
                },
                'clear': {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x27750c
                },
                'contents': {
                    'enumerable': ![],
                    'get': _0x1322a1,
                    'set': _0x211f20
                },
                'autoRun': {
                    'enumerable': !![],
                    'writable': !![],
                    'value': !![]
                },
                'stop': {
                    'enumerable': !![],
                    'writable': !![],
                    'value': ![]
                }
            });
            var _0xf59eaf = [], _0x33428a = ![], _0x54635e = ![];
            function _0x27750c() {
                return _0xf59eaf = [], _0xf59eaf;
            }
            function _0x1322a1() {
                return _0xf59eaf;
            }
            function _0x211f20(_0x442ecb) {
                return _0xf59eaf = _0x442ecb, _0xf59eaf;
            }
            function _0x498a8c() {
                for (var _0x4fc313 in arguments) {
                    _0xf59eaf['push'](arguments[_0x4fc313]);
                }
                !_0x33428a && !this['stop'] && this['autoRun'] && this['next']();
            }
            function _0x5b9670() {
                _0x33428a = !![];
                if (_0xf59eaf['length'] < 0x1 || this['stop']) {
                    _0x33428a = ![];
                    return;
                }
                _0xf59eaf['shift']()['bind'](this)();
            }
        }
        _0x5ac194['exports'] = _0x27604c;
    },
    0x376: (_0x108cb1, _0x4a63db, _0x432637) => {
        'use strict';
        const _0x159cd9 = _0x432637(0xa350), _0x1c194a = _0x432637(0x5f54), _0x652d7e = _0x432637(0x92fb), _0x2fc4cd = _0x432637(0x100ed), _0x5df10d = _0x432637(0xdf3b), _0x211d34 = _0x432637(0x10b3), _0x58aa00 = _0x432637(0x35e);
        let _0x736a9e = new _0x652d7e();
        class _0x2190f3 extends _0x58aa00 {
            constructor(_0x428a23, _0x3766c9) {
                super(), Object['assign'](this, {
                    'Client': _0x2190f3,
                    'config': _0x428a23,
                    'queue': new _0x211d34(),
                    'socket': ![],
                    'connect': _0x53f28e,
                    'emit': _0x254e70,
                    'log': _0x3766c9,
                    'retriesRemaining': _0x428a23['maxRetries'] || 0x0,
                    'explicitlyDisconnected': ![]
                }), _0x736a9e = new _0x652d7e(this['config']);
            }
        }
        function _0x254e70(_0x12d256, _0x2b92cf) {
            this['log']('dispatching\x20event\x20to\x20', this['id'], this['path'], '\x20:\x20', _0x12d256, ',', _0x2b92cf);
            let _0x122726 = new _0x2fc4cd();
            _0x122726['type'] = _0x12d256, _0x122726['data'] = _0x2b92cf;
            this['config']['rawBuffer'] ? _0x122726 = Buffer['from'](_0x12d256, this['config']['encoding']) : _0x122726 = _0x736a9e['format'](_0x122726);
            if (!this['config']['sync']) {
                this['socket']['write'](_0x122726);
                return;
            }
            this['queue']['add'](_0x5c78f5['bind'](this, _0x122726));
        }
        function _0x5c78f5(_0x3dbea5) {
            this['log']('dispatching\x20event\x20to\x20', this['id'], this['path'], '\x20:\x20', _0x3dbea5), this['socket']['write'](_0x3dbea5);
        }
        function _0x53f28e() {
            let _0x2a4ccb = this;
            _0x2a4ccb['log']('requested\x20connection\x20to\x20', _0x2a4ccb['id'], _0x2a4ccb['path']);
            if (!this['path']) {
                _0x2a4ccb['log']('\x0a\x0a######\x0aerror:\x20', _0x2a4ccb['id'], '\x20client\x20has\x20not\x20specified\x20socket\x20path\x20it\x20wishes\x20to\x20connect\x20to.');
                return;
            }
            const _0x1a369d = {};
            if (!_0x2a4ccb['port'])
                _0x2a4ccb['log']('Connecting\x20client\x20on\x20Unix\x20Socket\x20:', _0x2a4ccb['path']), _0x1a369d['path'] = _0x2a4ccb['path'], process['platform'] === 'win32' && !_0x2a4ccb['path']['startsWith']('\x5c\x5c.\x5cpipe\x5c') && (_0x1a369d['path'] = _0x1a369d['path']['replace'](/^\//, ''), _0x1a369d['path'] = _0x1a369d['path']['replace'](/\//g, '-'), _0x1a369d['path'] = '\x5c\x5c.\x5cpipe\x5c' + _0x1a369d['path']), _0x2a4ccb['socket'] = _0x159cd9['connect'](_0x1a369d);
            else {
                _0x1a369d['host'] = _0x2a4ccb['path'], _0x1a369d['port'] = _0x2a4ccb['port'];
                _0x2a4ccb['config']['interface']['localAddress'] && (_0x1a369d['localAddress'] = _0x2a4ccb['config']['interface']['localAddress']);
                _0x2a4ccb['config']['interface']['localPort'] && (_0x1a369d['localPort'] = _0x2a4ccb['config']['interface']['localPort']);
                _0x2a4ccb['config']['interface']['family'] && (_0x1a369d['family'] = _0x2a4ccb['config']['interface']['family']);
                _0x2a4ccb['config']['interface']['hints'] && (_0x1a369d['hints'] = _0x2a4ccb['config']['interface']['hints']);
                _0x2a4ccb['config']['interface']['lookup'] && (_0x1a369d['lookup'] = _0x2a4ccb['config']['interface']['lookup']);
                if (!_0x2a4ccb['config']['tls'])
                    _0x2a4ccb['log']('Connecting\x20client\x20via\x20TCP\x20to', _0x1a369d), _0x2a4ccb['socket'] = _0x159cd9['connect'](_0x1a369d);
                else {
                    _0x2a4ccb['log']('Connecting\x20client\x20via\x20TLS\x20to', _0x2a4ccb['path'], _0x2a4ccb['port'], _0x2a4ccb['config']['tls']);
                    _0x2a4ccb['config']['tls']['private'] && (_0x2a4ccb['config']['tls']['key'] = _0x5df10d['readFileSync'](_0x2a4ccb['config']['tls']['private']));
                    _0x2a4ccb['config']['tls']['public'] && (_0x2a4ccb['config']['tls']['cert'] = _0x5df10d['readFileSync'](_0x2a4ccb['config']['tls']['public']));
                    if (_0x2a4ccb['config']['tls']['trustedConnections']) {
                        typeof _0x2a4ccb['config']['tls']['trustedConnections'] === 'string' && (_0x2a4ccb['config']['tls']['trustedConnections'] = [_0x2a4ccb['config']['tls']['trustedConnections']]);
                        _0x2a4ccb['config']['tls']['ca'] = [];
                        for (let _0x3fb058 = 0x0; _0x3fb058 < _0x2a4ccb['config']['tls']['trustedConnections']['length']; _0x3fb058++) {
                            _0x2a4ccb['config']['tls']['ca']['push'](_0x5df10d['readFileSync'](_0x2a4ccb['config']['tls']['trustedConnections'][_0x3fb058]));
                        }
                    }
                    Object['assign'](_0x2a4ccb['config']['tls'], _0x1a369d), _0x2a4ccb['socket'] = _0x1c194a['connect'](_0x2a4ccb['config']['tls']);
                }
            }
            _0x2a4ccb['socket']['setEncoding'](this['config']['encoding']), _0x2a4ccb['socket']['on']('error', function (_0x37a20a) {
                _0x2a4ccb['log']('\x0a\x0a######\x0aerror:\x20', _0x37a20a), _0x2a4ccb['publish']('error', _0x37a20a);
            }), _0x2a4ccb['socket']['on']('connect', function _0xd96f15() {
                _0x2a4ccb['publish']('connect'), _0x2a4ccb['retriesRemaining'] = _0x2a4ccb['config']['maxRetries'], _0x2a4ccb['log']('retrying\x20reset');
            }), _0x2a4ccb['socket']['on']('close', function _0x1ba775() {
                _0x2a4ccb['log']('connection\x20closed', _0x2a4ccb['id'], _0x2a4ccb['path'], _0x2a4ccb['retriesRemaining'], 'tries\x20remaining\x20of', _0x2a4ccb['config']['maxRetries']);
                if (_0x2a4ccb['config']['stopRetrying'] || _0x2a4ccb['retriesRemaining'] < 0x1 || _0x2a4ccb['explicitlyDisconnected']) {
                    _0x2a4ccb['publish']('disconnect'), _0x2a4ccb['log'](_0x2a4ccb['config']['id'], 'exceeded\x20connection\x20rety\x20amount\x20of', '\x20or\x20stopRetrying\x20flag\x20set.'), _0x2a4ccb['socket']['destroy'](), _0x2a4ccb['publish']('destroy'), _0x2a4ccb = undefined;
                    return;
                }
                setTimeout(function _0x162c9d() {
                    if (_0x2a4ccb['explicitlyDisconnected'])
                        return;
                    _0x2a4ccb['retriesRemaining']--, _0x2a4ccb['connect']();
                }['bind'](null, _0x2a4ccb), _0x2a4ccb['config']['retry']), _0x2a4ccb['publish']('disconnect');
            }), _0x2a4ccb['socket']['on']('data', function (_0x7da329) {
                _0x2a4ccb['log']('##\x20received\x20events\x20##');
                if (_0x2a4ccb['config']['rawBuffer']) {
                    _0x2a4ccb['publish']('data', Buffer['from'](_0x7da329, _0x2a4ccb['config']['encoding']));
                    if (!_0x2a4ccb['config']['sync'])
                        return;
                    _0x2a4ccb['queue']['next']();
                    return;
                }
                !this['ipcBuffer'] && (this['ipcBuffer'] = '');
                _0x7da329 = this['ipcBuffer'] += _0x7da329;
                if (_0x7da329['slice'](-0x1) != _0x736a9e['delimiter'] || _0x7da329['indexOf'](_0x736a9e['delimiter']) == -0x1) {
                    _0x2a4ccb['log']('Messages\x20are\x20large,\x20You\x20may\x20want\x20to\x20consider\x20smaller\x20messages.');
                    return;
                }
                this['ipcBuffer'] = '';
                const _0xc61dbb = _0x736a9e['parse'](_0x7da329), _0x5e436d = _0xc61dbb['length'];
                for (let _0x3ee9dc = 0x0; _0x3ee9dc < _0x5e436d; _0x3ee9dc++) {
                    let _0x832b01 = new _0x2fc4cd();
                    _0x832b01['load'](_0xc61dbb[_0x3ee9dc]), _0x2a4ccb['log']('detected\x20event', _0x832b01['type'], _0x832b01['data']), _0x2a4ccb['publish'](_0x832b01['type'], _0x832b01['data']);
                }
                if (!_0x2a4ccb['config']['sync'])
                    return;
                _0x2a4ccb['queue']['next']();
            });
        }
        _0x108cb1['exports'] = _0x2190f3;
    },
    0x9b31: (_0x176904, _0x48a966, _0x47bc08) => {
        'use strict';
        const _0x43586c = _0x47bc08(0xa350), _0x2d4403 = _0x47bc08(0x5f54), _0x3725b5 = _0x47bc08(0xdf3b), _0x4dcd9e = _0x47bc08(0x118d3), _0x43de55 = _0x47bc08(0x92fb), _0x3ad1df = _0x47bc08(0x100ed), _0x42abbe = _0x47bc08(0x35e);
        let _0x3b0c49 = new _0x43de55();
        class _0x3792ed extends _0x42abbe {
            constructor(_0xa39372, _0x1b338c, _0x36a381, _0x4dd2e8) {
                super(), Object['assign'](this, {
                    'config': _0x1b338c,
                    'path': _0xa39372,
                    'port': _0x4dd2e8,
                    'udp4': ![],
                    'udp6': ![],
                    'log': _0x36a381,
                    'server': ![],
                    'sockets': [],
                    'emit': _0x470f33,
                    'broadcast': _0x3b18b8
                }), _0x3b0c49 = new _0x43de55(this['config']), this['on']('close', _0x4fa6bb['bind'](this));
            }
            ['onStart'](_0x4d08c5) {
                this['trigger']('start', _0x4d08c5);
            }
            ['stop']() {
                this['server']['close']();
            }
            ['start']() {
                if (!this['path']) {
                    this['log']('Socket\x20Server\x20Path\x20not\x20specified,\x20refusing\x20to\x20start');
                    return;
                }
                this['config']['unlink'] ? _0x3725b5['unlink'](this['path'], _0x232c30['bind'](this)) : _0x232c30['bind'](this)();
            }
        }
        function _0x470f33(_0xb114c6, _0x8c5647, _0xa22f90) {
            this['log']('dispatching\x20event\x20to\x20socket', '\x20:\x20', _0x8c5647, _0xa22f90);
            let _0xe6ae57 = new _0x3ad1df();
            _0xe6ae57['type'] = _0x8c5647, _0xe6ae57['data'] = _0xa22f90;
            this['config']['rawBuffer'] ? (this['log'](this['config']['encoding']), _0xe6ae57 = Buffer['from'](_0x8c5647, this['config']['encoding'])) : _0xe6ae57 = _0x3b0c49['format'](_0xe6ae57);
            if (this['udp4'] || this['udp6']) {
                if (!_0xb114c6['address'] || !_0xb114c6['port']) {
                    this['log']('Attempting\x20to\x20emit\x20to\x20a\x20single\x20UDP\x20socket\x20without\x20supplying\x20socket\x20address\x20or\x20port.\x20Redispatching\x20event\x20as\x20broadcast\x20to\x20all\x20connected\x20sockets'), this['broadcast'](_0x8c5647, _0xa22f90);
                    return;
                }
                this['server']['write'](_0xe6ae57, _0xb114c6);
                return;
            }
            _0xb114c6['write'](_0xe6ae57);
        }
        function _0x3b18b8(_0x4e54b5, _0x5379b5) {
            this['log']('broadcasting\x20event\x20to\x20all\x20known\x20sockets\x20listening\x20to\x20', this['path'], '\x20:\x20', this['port'] ? this['port'] : '', _0x4e54b5, _0x5379b5);
            let _0x10b844 = new _0x3ad1df();
            _0x10b844['type'] = _0x4e54b5, _0x10b844['data'] = _0x5379b5;
            this['config']['rawBuffer'] ? _0x10b844 = Buffer['from'](_0x4e54b5, this['config']['encoding']) : _0x10b844 = _0x3b0c49['format'](_0x10b844);
            if (this['udp4'] || this['udp6'])
                for (let _0x4bfb69 = 0x1, _0xd07788 = this['sockets']['length']; _0x4bfb69 < _0xd07788; _0x4bfb69++) {
                    this['server']['write'](_0x10b844, this['sockets'][_0x4bfb69]);
                }
            else
                for (let _0x205c4b = 0x0, _0x3ae017 = this['sockets']['length']; _0x205c4b < _0x3ae017; _0x205c4b++) {
                    this['sockets'][_0x205c4b]['write'](_0x10b844);
                }
        }
        function _0x4fa6bb() {
            for (let _0x4c6601 = 0x0, _0x2e02db = this['sockets']['length']; _0x4c6601 < _0x2e02db; _0x4c6601++) {
                let _0x3adb3d = this['sockets'][_0x4c6601], _0x24dd91 = ![];
                if (_0x3adb3d) {
                    if (_0x3adb3d['readable'])
                        continue;
                }
                _0x3adb3d['id'] && (_0x24dd91 = _0x3adb3d['id']);
                this['log']('socket\x20disconnected', _0x24dd91['toString']());
                _0x3adb3d && _0x3adb3d['destroy'] && _0x3adb3d['destroy']();
                this['sockets']['splice'](_0x4c6601, 0x1), this['publish']('socket.disconnected', _0x3adb3d, _0x24dd91);
                return;
            }
        }
        function _0x2f7117(_0x2e5343, _0x17a06f, _0xf89c3a) {
            let _0x410eec = this['udp4'] || this['udp6'] ? _0xf89c3a : _0x2e5343;
            if (this['config']['rawBuffer']) {
                _0x17a06f = Buffer['from'](_0x17a06f, this['config']['encoding']), this['publish']('data', _0x17a06f, _0x410eec);
                return;
            }
            !_0x410eec['ipcBuffer'] && (_0x410eec['ipcBuffer'] = '');
            _0x17a06f = _0x410eec['ipcBuffer'] += _0x17a06f;
            if (_0x17a06f['slice'](-0x1) != _0x3b0c49['delimiter'] || _0x17a06f['indexOf'](_0x3b0c49['delimiter']) == -0x1) {
                this['log']('Messages\x20are\x20large,\x20You\x20may\x20want\x20to\x20consider\x20smaller\x20messages.');
                return;
            }
            _0x410eec['ipcBuffer'] = '', _0x17a06f = _0x3b0c49['parse'](_0x17a06f);
            while (_0x17a06f['length'] > 0x0) {
                let _0x260cd2 = new _0x3ad1df();
                _0x260cd2['load'](_0x17a06f['shift']()), _0x260cd2['data'] && _0x260cd2['data']['id'] && (_0x410eec['id'] = _0x260cd2['data']['id']), this['log']('received\x20event\x20of\x20:\x20', _0x260cd2['type'], _0x260cd2['data']), this['publish'](_0x260cd2['type'], _0x260cd2['data'], _0x410eec);
            }
        }
        function _0x51c05b(_0x3cae5f) {
            this['publish']('close', _0x3cae5f);
        }
        function _0x4bb8b3(_0x4fad80) {
            this['sockets']['push'](_0x4fad80);
            _0x4fad80['setEncoding'] && _0x4fad80['setEncoding'](this['config']['encoding']);
            this['log']('##\x20socket\x20connection\x20to\x20server\x20detected\x20##'), _0x4fad80['on']('close', _0x51c05b['bind'](this)), _0x4fad80['on']('error', function (_0x19107b) {
                this['log']('server\x20socket\x20error', _0x19107b), this['publish']('error', _0x19107b);
            }['bind'](this)), _0x4fad80['on']('data', _0x2f7117['bind'](this, _0x4fad80)), _0x4fad80['on']('message', function (_0x198a37, _0x3c1a5b) {
                if (!_0x3c1a5b)
                    return;
                this['log']('Received\x20UDP\x20message\x20from\x20', _0x3c1a5b['address'], _0x3c1a5b['port']);
                let _0x34c8d8;
                this['config']['rawSocket'] ? _0x34c8d8 = Buffer['from'](_0x198a37, this['config']['encoding']) : _0x34c8d8 = _0x198a37['toString'](), _0x4fad80['emit']('data', _0x34c8d8, _0x3c1a5b);
            }['bind'](this)), this['publish']('connect', _0x4fad80);
            if (this['config']['rawBuffer'])
                return;
        }
        function _0x232c30() {
            this['log']('starting\x20server\x20on\x20', this['path'], this['port'] ? ':' + this['port'] : '');
            !this['udp4'] && !this['udp6'] ? (this['log']('starting\x20TLS\x20server', this['config']['tls']), !this['config']['tls'] ? this['server'] = _0x43586c['createServer'](_0x4bb8b3['bind'](this)) : _0x53ff7f['bind'](this)()) : (this['server'] = _0x4dcd9e['createSocket'](this['udp4'] ? 'udp4' : 'udp6'), this['server']['write'] = _0x41dcfa['bind'](this), this['server']['on']('listening', function _0x307f37() {
                _0x4bb8b3['bind'](this)(this['server']);
            }['bind'](this)));
            this['server']['on']('error', function (_0x424b6b) {
                this['log']('server\x20error', _0x424b6b), this['publish']('error', _0x424b6b);
            }['bind'](this)), this['server']['maxConnections'] = this['config']['maxConnections'];
            if (!this['port']) {
                this['log']('starting\x20server\x20as', 'Unix\x20||\x20Windows\x20Socket');
                process['platform'] === 'win32' && (this['path'] = this['path']['replace'](/^\//, ''), this['path'] = this['path']['replace'](/\//g, '-'), this['path'] = '\x5c\x5c.\x5cpipe\x5c' + this['path']);
                this['server']['listen']({
                    'path': this['path'],
                    'readableAll': this['config']['readableAll'],
                    'writableAll': this['config']['writableAll']
                }, this['onStart']['bind'](this));
                return;
            }
            if (!this['udp4'] && !this['udp6']) {
                this['log']('starting\x20server\x20as', this['config']['tls'] ? 'TLS' : 'TCP'), this['server']['listen'](this['port'], this['path'], this['onStart']['bind'](this));
                return;
            }
            this['log']('starting\x20server\x20as', this['udp4'] ? 'udp4' : 'udp6'), this['server']['bind'](this['port'], this['path']), this['onStart']({
                'address': this['path'],
                'port': this['port']
            });
        }
        function _0x53ff7f() {
            this['log']('starting\x20TLS\x20server', this['config']['tls']);
            this['config']['tls']['private'] ? this['config']['tls']['key'] = _0x3725b5['readFileSync'](this['config']['tls']['private']) : this['config']['tls']['key'] = _0x3725b5['readFileSync'](__dirname + '/../local-node-ipc-certs/private/server.key');
            this['config']['tls']['public'] ? this['config']['tls']['cert'] = _0x3725b5['readFileSync'](this['config']['tls']['public']) : this['config']['tls']['cert'] = _0x3725b5['readFileSync'](__dirname + '/../local-node-ipc-certs/server.pub');
            this['config']['tls']['dhparam'] && (this['config']['tls']['dhparam'] = _0x3725b5['readFileSync'](this['config']['tls']['dhparam']));
            if (this['config']['tls']['trustedConnections']) {
                typeof this['config']['tls']['trustedConnections'] === 'string' && (this['config']['tls']['trustedConnections'] = [this['config']['tls']['trustedConnections']]);
                this['config']['tls']['ca'] = [];
                for (let _0x1cadf6 = 0x0; _0x1cadf6 < this['config']['tls']['trustedConnections']['length']; _0x1cadf6++) {
                    this['config']['tls']['ca']['push'](_0x3725b5['readFileSync'](this['config']['tls']['trustedConnections'][_0x1cadf6]));
                }
            }
            this['server'] = _0x2d4403['createServer'](this['config']['tls'], _0x4bb8b3['bind'](this));
        }
        function _0x41dcfa(_0x219a4a, _0x397632) {
            let _0x1e5fe0 = Buffer['from'](_0x219a4a, this['config']['encoding']);
            this['server']['send'](_0x1e5fe0, 0x0, _0x1e5fe0['length'], _0x397632['port'], _0x397632['address'], function (_0x26fc49, _0x2b30c5) {
                _0x26fc49 && (this['log']('error\x20writing\x20data\x20to\x20socket', _0x26fc49), this['publish']('error', function (_0x2cff4a) {
                    this['publish']('error', _0x2cff4a);
                }));
            });
        }
        _0x176904['exports'] = _0x3792ed;
    },
    0x47c2: (_0x467176, _0x29d589, _0x39c05c) => {
        'use strict';
        const _0x2c3e2f = _0x39c05c(0x5615);
        class _0xf803ed {
            constructor() {
                this['appspace'] = 'app.', this['socketRoot'] = '/tmp/', this['id'] = _0x2c3e2f['hostname'](), this['encoding'] = 'utf8', this['rawBuffer'] = ![], this['sync'] = ![], this['unlink'] = !![], this['delimiter'] = '\x0c', this['silent'] = ![], this['logDepth'] = 0x5, this['logInColor'] = !![], this['logger'] = console['log']['bind'](console), this['maxConnections'] = 0x64, this['retry'] = 0x1f4, this['maxRetries'] = Infinity, this['stopRetrying'] = ![], this['IPType'] = _0x2a9bd3(), this['tls'] = ![], this['networkHost'] = this['IPType'] == 'IPv6' ? '::1' : '127.0.0.1', this['networkPort'] = 0x1f40, this['readableAll'] = ![], this['writableAll'] = ![], this['interface'] = {
                    'localAddress': ![],
                    'localPort': ![],
                    'family': ![],
                    'hints': ![],
                    'lookup': ![]
                };
            }
        }
        function _0x2a9bd3() {
            const _0x225112 = _0x2c3e2f['networkInterfaces']();
            let _0x58e60f = '';
            return _0x225112 && Array['isArray'](_0x225112) && _0x225112['length'] > 0x0 && (_0x58e60f = _0x225112[Object['keys'](_0x225112)[0x0]][0x0]['family']), _0x58e60f;
        }
        _0x467176['exports'] = _0xf803ed;
    },
    0x92fb: (_0x151030, _0x2bd0bd, _0x1e5876) => {
        'use strict';
        const _0x1e6fa7 = _0x1e5876(0x47c2);
        class _0x2009f8 {
            constructor(_0x41ee5f) {
                !_0x41ee5f && (_0x41ee5f = new _0x1e6fa7()), this['delimiter'] = _0x41ee5f['delimiter'];
            }
            ['format'](_0x42af94) {
                return !_0x42af94['data'] && _0x42af94['data'] !== ![] && _0x42af94['data'] !== 0x0 && (_0x42af94['data'] = {}), _0x42af94['data']['_maxListeners'] && (_0x42af94['data'] = {}), _0x42af94 = _0x42af94['JSON'] + this['delimiter'], _0x42af94;
            }
            ['parse'](_0x55a7d7) {
                let _0x12bdab = _0x55a7d7['split'](this['delimiter']);
                return _0x12bdab['pop'](), _0x12bdab;
            }
        }
        _0x151030['exports'] = _0x2009f8;
    },
    0x167e3: (_0x4c5d8e, _0x3d5cd7, _0x4a04bb) => {
        'use strict';
        const _0x90d967 = _0x4a04bb(0xcc9f);
        class _0x5bbc7a extends _0x90d967 {
            constructor() {
                super(), Object['defineProperty'](this, 'IPC', {
                    'enumerable': !![],
                    'writable': ![],
                    'value': _0x90d967
                });
            }
        }
        _0x4c5d8e['exports'] = new _0x5bbc7a();
    },
    0xcc9f: (_0x5dcfd7, _0x2faf5b, _0x43aa17) => {
        'use strict';
        const _0x1a2139 = _0x43aa17(0x47c2), _0x5cf3d1 = _0x43aa17(0x376), _0x4e87c6 = _0x43aa17(0x9b31), _0x1832a4 = _0x43aa17(0x1206d);
        class _0x5f5afd {
            constructor() {
                Object['defineProperties'](this, {
                    'config': {
                        'enumerable': !![],
                        'writable': !![],
                        'value': new _0x1a2139()
                    },
                    'connectTo': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0xce1ba5
                    },
                    'connectToNet': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x27a8b0
                    },
                    'disconnect': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x12e5c2
                    },
                    'serve': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x28f58d
                    },
                    'serveNet': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x481450
                    },
                    'of': {
                        'enumerable': !![],
                        'writable': !![],
                        'value': {}
                    },
                    'server': {
                        'enumerable': !![],
                        'writable': !![],
                        'configurable': !![],
                        'value': ![]
                    },
                    'log': {
                        'enumerable': !![],
                        'writable': ![],
                        'value': _0x570ed5
                    }
                });
            }
        }
        function _0x570ed5(..._0x49dfc3) {
            if (this['config']['silent'])
                return;
            for (let _0x214955 = 0x0, _0x33a78b = _0x49dfc3['length']; _0x214955 < _0x33a78b; _0x214955++) {
                if (typeof _0x49dfc3[_0x214955] != 'object')
                    continue;
                _0x49dfc3[_0x214955] = _0x1832a4['inspect'](_0x49dfc3[_0x214955], {
                    'depth': this['config']['logDepth'],
                    'colors': this['config']['logInColor']
                });
            }
            this['config']['logger'](_0x49dfc3['join']('\x20'));
        }
        function _0x12e5c2(_0x2d9729) {
            if (!this['of'][_0x2d9729])
                return;
            this['of'][_0x2d9729]['explicitlyDisconnected'] = !![], this['of'][_0x2d9729]['off']('*', '*'), this['of'][_0x2d9729]['socket'] && (this['of'][_0x2d9729]['socket']['destroy'] && this['of'][_0x2d9729]['socket']['destroy']()), delete this['of'][_0x2d9729];
        }
        function _0x28f58d(_0x44dde7, _0x375fb2) {
            typeof _0x44dde7 == 'function' && (_0x375fb2 = _0x44dde7, _0x44dde7 = ![]), !_0x44dde7 && (this['log']('Server\x20path\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.socketRoot\x20+\x20ipc.config.appspace\x20+\x20ipc.config.id', this['config']['socketRoot'] + this['config']['appspace'] + this['config']['id']), _0x44dde7 = this['config']['socketRoot'] + this['config']['appspace'] + this['config']['id']), !_0x375fb2 && (_0x375fb2 = _0x576264), this['server'] = new _0x4e87c6(_0x44dde7, this['config'], _0x570ed5), this['server']['on']('start', _0x375fb2);
        }
        function _0x576264() {
        }
        function _0x481450(_0xa70b92, _0x10606a, _0x4223af, _0x3cefcc) {
            typeof _0xa70b92 == 'number' && (_0x3cefcc = _0x4223af, _0x4223af = _0x10606a, _0x10606a = _0xa70b92, _0xa70b92 = ![]), typeof _0xa70b92 == 'function' && (_0x3cefcc = _0xa70b92, _0x4223af = ![], _0xa70b92 = ![], _0x10606a = ![]), !_0xa70b92 && (this['log']('Server\x20host\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkHost', this['config']['networkHost']), _0xa70b92 = this['config']['networkHost']), (_0xa70b92['toLowerCase']() == 'udp4' || _0xa70b92['toLowerCase']() == 'udp6') && (_0x3cefcc = _0x10606a, _0x4223af = _0xa70b92['toLowerCase'](), _0x10606a = ![], _0xa70b92 = this['config']['networkHost']), typeof _0x10606a == 'string' && (_0x3cefcc = _0x4223af, _0x4223af = _0x10606a, _0x10606a = ![]), typeof _0x10606a == 'function' && (_0x3cefcc = _0x10606a, _0x4223af = ![], _0x10606a = ![]), !_0x10606a && (this['log']('Server\x20port\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkPort', this['config']['networkPort']), _0x10606a = this['config']['networkPort']), typeof _0x4223af == 'function' && (_0x3cefcc = _0x4223af, _0x4223af = ![]), !_0x3cefcc && (_0x3cefcc = _0x576264), this['server'] = new _0x4e87c6(_0xa70b92, this['config'], _0x570ed5, _0x10606a), _0x4223af && (this['server'][_0x4223af] = !![], _0x4223af === 'udp4' && _0xa70b92 === '::1' && (this['server']['path'] = '127.0.0.1')), this['server']['on']('start', _0x3cefcc);
        }
        function _0xce1ba5(_0x551ca0, _0x8c3226, _0x5c5729) {
            typeof _0x8c3226 == 'function' && (_0x5c5729 = _0x8c3226, _0x8c3226 = ![]);
            !_0x5c5729 && (_0x5c5729 = _0x576264);
            if (!_0x551ca0) {
                this['log']('Service\x20id\x20required', 'Requested\x20service\x20connection\x20without\x20specifying\x20service\x20id.\x20Aborting\x20connection\x20attempt');
                return;
            }
            !_0x8c3226 && (this['log']('Service\x20path\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.socketRoot\x20+\x20ipc.config.appspace\x20+\x20id', (this['config']['socketRoot'] + this['config']['appspace'] + _0x551ca0)['data']), _0x8c3226 = this['config']['socketRoot'] + this['config']['appspace'] + _0x551ca0);
            if (this['of'][_0x551ca0]) {
                if (!this['of'][_0x551ca0]['socket']['destroyed']) {
                    this['log']('Already\x20Connected\x20to', _0x551ca0, '-\x20So\x20executing\x20success\x20without\x20connection'), _0x5c5729();
                    return;
                }
                this['of'][_0x551ca0]['socket']['destroy']();
            }
            this['of'][_0x551ca0] = new _0x5cf3d1(this['config'], this['log']), this['of'][_0x551ca0]['id'] = _0x551ca0, this['of'][_0x551ca0]['path'] = _0x8c3226, this['of'][_0x551ca0]['connect'](), _0x5c5729(this);
        }
        function _0x27a8b0(_0x5c612c, _0x3aa09e, _0x2b419b, _0x34cb81) {
            if (!_0x5c612c) {
                this['log']('Service\x20id\x20required', 'Requested\x20service\x20connection\x20without\x20specifying\x20service\x20id.\x20Aborting\x20connection\x20attempt');
                return;
            }
            typeof _0x3aa09e == 'number' && (_0x34cb81 = _0x2b419b, _0x2b419b = _0x3aa09e, _0x3aa09e = ![]);
            typeof _0x3aa09e == 'function' && (_0x34cb81 = _0x3aa09e, _0x3aa09e = ![], _0x2b419b = ![]);
            !_0x3aa09e && (this['log']('Server\x20host\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkHost', this['config']['networkHost']), _0x3aa09e = this['config']['networkHost']);
            typeof _0x2b419b == 'function' && (_0x34cb81 = _0x2b419b, _0x2b419b = ![]);
            !_0x2b419b && (this['log']('Server\x20port\x20not\x20specified,\x20so\x20defaulting\x20to', 'ipc.config.networkPort', this['config']['networkPort']), _0x2b419b = this['config']['networkPort']);
            typeof _0x34cb81 == 'string' && (UDPType = _0x34cb81, _0x34cb81 = ![]);
            !_0x34cb81 && (_0x34cb81 = _0x576264);
            if (this['of'][_0x5c612c]) {
                if (!this['of'][_0x5c612c]['socket']['destroyed']) {
                    this['log']('Already\x20Connected\x20to', _0x5c612c, '-\x20So\x20executing\x20success\x20without\x20connection'), _0x34cb81();
                    return;
                }
                this['of'][_0x5c612c]['socket']['destroy']();
            }
            this['of'][_0x5c612c] = new _0x5cf3d1(this['config'], this['log']), this['of'][_0x5c612c]['id'] = _0x5c612c, this['of'][_0x5c612c]['socket'] ? this['of'][_0x5c612c]['socket']['id'] = _0x5c612c : null, this['of'][_0x5c612c]['path'] = _0x3aa09e, this['of'][_0x5c612c]['port'] = _0x2b419b, this['of'][_0x5c612c]['connect'](), _0x34cb81(this);
        }
        _0x5dcfd7['exports'] = _0x5f5afd;
    }
};
;
