// test-maker/dist/154.js
'use strict';
exports['id'] = 0x9a, exports['ids'] = [0x9a], exports['modules'] = {
    0x9a: (_0x2f55c2, _0x36678b, _0x4faff5) => {
        _0x4faff5['r'](_0x36678b), _0x4faff5['d'](_0x36678b, { 'ReportersManager': () => _0x43dbb4 });
        var _0x26fe7c = _0x4faff5(0x1145c), _0xa95d8b = _0x4faff5['n'](_0x26fe7c), _0x295ee7 = _0x4faff5(0xc5d1), _0x2cfcb8 = _0x4faff5(0x111e7), _0x143845 = _0x4faff5(0x149e5), _0x1a5f78 = _0x4faff5(0x17784), _0x5a6c67 = _0x4faff5(0x38ed);
        class _0x43dbb4 {
            constructor() {
                this['reporters'] = new Map(), this['registerEvents']();
            }
            async ['install'](_0x42e45e, _0x29b11c) {
                var _0x5c7b30, _0x1c0aec;
                this['engine'] = _0x42e45e, this['pluginsManager'] = _0x29b11c, this['reporters'] = this['engine']['reporters'];
                for (let _0x14b742 of this['reporters']['values']()) {
                    const _0x1328e1 = await ((_0x1c0aec = (_0x5c7b30 = _0x14b742['reporter'])['install']) === null || _0x1c0aec === void 0x0 ? void 0x0 : _0x1c0aec['call'](_0x5c7b30, _0x42e45e, _0x14b742['options']));
                    if (_0x1328e1)
                        _0x295ee7['stateManager']['state']['installQueue']['push'](..._0x1328e1);
                }
            }
            async ['init'](_0x50b331, _0x5506f6) {
                this['engine'] = _0x50b331, this['pluginsManager'] = _0x5506f6, this['reporters'] = this['engine']['reporters'];
                for (let _0x5ce2bc of this['reporters']['values']())
                    await _0x5ce2bc['reporter']['init'](_0x50b331, _0x5ce2bc['options']);
            }
            async ['taskStart'](_0x2ffe21) {
                var _0x59ca8e, _0x2feb2e;
                try {
                    for (let _0x5b767b of this['reporters']['values']())
                        await ((_0x2feb2e = (_0x59ca8e = _0x5b767b['reporter'])['taskStart']) === null || _0x2feb2e === void 0x0 ? void 0x0 : _0x2feb2e['call'](_0x59ca8e, _0x2ffe21));
                    await this['pluginsManager']['runTaskStart']();
                } catch (_0x572530) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x572530 });
                }
            }
            async ['featureStart'](_0x113c6a, _0x490bec) {
                var _0x5e0943, _0x459096;
                try {
                    for (let _0x576ded of this['reporters']['values']())
                        await ((_0x459096 = (_0x5e0943 = _0x576ded['reporter'])['featureStart']) === null || _0x459096 === void 0x0 ? void 0x0 : _0x459096['call'](_0x5e0943, _0x113c6a, _0x490bec));
                } catch (_0x52fc4a) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x52fc4a });
                }
            }
            async ['featureDone'](_0x29fcfd, _0x1ef2ab) {
                var _0x4e8e63, _0x870913;
                try {
                    for (let _0x4ff21f of this['reporters']['values']())
                        await ((_0x870913 = (_0x4e8e63 = _0x4ff21f['reporter'])['featureDone']) === null || _0x870913 === void 0x0 ? void 0x0 : _0x870913['call'](_0x4e8e63, _0x29fcfd, _0x1ef2ab));
                } catch (_0x47dfd6) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x47dfd6 });
                }
            }
            async ['scenarioStart'](_0x6e6d7b, _0xbc856d) {
                var _0x441cbc, _0x51a883;
                try {
                    for (let _0x129940 of this['reporters']['values']())
                        await ((_0x51a883 = (_0x441cbc = _0x129940['reporter'])['scenarioStart']) === null || _0x51a883 === void 0x0 ? void 0x0 : _0x51a883['call'](_0x441cbc, _0x6e6d7b, _0xbc856d));
                } catch (_0x160966) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x160966 });
                }
            }
            async ['scenarioDone'](_0x13b43f, _0x9b7ca1) {
                var _0x4c6713, _0x20f625;
                try {
                    for (let _0x440b98 of this['reporters']['values']())
                        await ((_0x20f625 = (_0x4c6713 = _0x440b98['reporter'])['scenarioDone']) === null || _0x20f625 === void 0x0 ? void 0x0 : _0x20f625['call'](_0x4c6713, _0x13b43f, _0x9b7ca1));
                } catch (_0x806631) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x806631 });
                }
            }
            async ['stepStart'](_0x27abba, _0x571d86) {
                var _0x38885a, _0x3b75ca;
                try {
                    for (let _0x594b08 of this['reporters']['values']())
                        await ((_0x3b75ca = (_0x38885a = _0x594b08['reporter'])['stepStart']) === null || _0x3b75ca === void 0x0 ? void 0x0 : _0x3b75ca['call'](_0x38885a, _0x27abba, _0x571d86));
                } catch (_0x3ea03d) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x3ea03d });
                }
            }
            async ['stepDone'](_0x10af46, _0x1f2d61) {
                var _0xfd0fb7, _0x481b62;
                try {
                    for (let _0x526945 of this['reporters']['values']())
                        await ((_0x481b62 = (_0xfd0fb7 = _0x526945['reporter'])['stepDone']) === null || _0x481b62 === void 0x0 ? void 0x0 : _0x481b62['call'](_0xfd0fb7, _0x10af46, _0x1f2d61));
                } catch (_0x54b0e0) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x54b0e0 });
                }
            }
            async ['subStepStart'](_0x5d59ab, _0x4651ae) {
                var _0x6ff645, _0x17c6c2;
                try {
                    for (let _0x21831f of this['reporters']['values']())
                        await ((_0x17c6c2 = (_0x6ff645 = _0x21831f['reporter'])['subStepStart']) === null || _0x17c6c2 === void 0x0 ? void 0x0 : _0x17c6c2['call'](_0x6ff645, _0x5d59ab, _0x4651ae));
                } catch (_0x32d4aa) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x32d4aa });
                }
            }
            async ['subStepDone'](_0x293d08, _0x29a7cd) {
                var _0x39caa4, _0x43c66a;
                try {
                    for (let _0x5030f1 of this['reporters']['values']())
                        await ((_0x43c66a = (_0x39caa4 = _0x5030f1['reporter'])['subStepDone']) === null || _0x43c66a === void 0x0 ? void 0x0 : _0x43c66a['call'](_0x39caa4, _0x293d08, _0x29a7cd));
                } catch (_0x53a57b) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x53a57b });
                }
            }
            async ['hookStart'](_0x49a345, _0x3ba94a, _0x24b492) {
                var _0x2d2d61, _0x5186d8;
                try {
                    for (let _0x15b564 of this['reporters']['values']())
                        await ((_0x5186d8 = (_0x2d2d61 = _0x15b564['reporter'])['hookStart']) === null || _0x5186d8 === void 0x0 ? void 0x0 : _0x5186d8['call'](_0x2d2d61, _0x49a345, _0x3ba94a, _0x24b492));
                } catch (_0x28c706) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x28c706 });
                }
            }
            async ['hookDone'](_0x4d9028, _0x159cbf, _0x5976f2) {
                var _0x252b28, _0x5415b7;
                try {
                    for (let _0xcabf4a of this['reporters']['values']())
                        await ((_0x5415b7 = (_0x252b28 = _0xcabf4a['reporter'])['hookDone']) === null || _0x5415b7 === void 0x0 ? void 0x0 : _0x5415b7['call'](_0x252b28, _0x4d9028, _0x159cbf, _0x5976f2));
                } catch (_0x1584f6) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x1584f6 });
                }
            }
            async ['selectorActionStart'](_0x522723) {
                var _0x2a04c3, _0x5bcadc;
                try {
                    for (let _0x570b25 of this['reporters']['values']())
                        await ((_0x5bcadc = (_0x2a04c3 = _0x570b25['reporter'])['selectorActionStart']) === null || _0x5bcadc === void 0x0 ? void 0x0 : _0x5bcadc['call'](_0x2a04c3, _0x522723));
                } catch (_0x200745) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x200745 });
                }
            }
            async ['selectorActionDone'](_0x5ebbf1) {
                var _0xdf19da, _0x11d95c;
                try {
                    for (let _0xc89b15 of this['reporters']['values']())
                        await ((_0x11d95c = (_0xdf19da = _0xc89b15['reporter'])['selectorActionDone']) === null || _0x11d95c === void 0x0 ? void 0x0 : _0x11d95c['call'](_0xdf19da, _0x5ebbf1));
                } catch (_0x4aa6e4) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x4aa6e4 });
                }
            }
            async ['controllerActionStart'](_0x275863) {
                var _0x328b02, _0x554ecd;
                try {
                    for (let _0x3544b8 of this['reporters']['values']())
                        await ((_0x554ecd = (_0x328b02 = _0x3544b8['reporter'])['controllerActionStart']) === null || _0x554ecd === void 0x0 ? void 0x0 : _0x554ecd['call'](_0x328b02, _0x275863));
                } catch (_0x2aa714) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x2aa714 });
                }
            }
            async ['controllerActionDone'](_0x252b63) {
                var _0x11a577, _0x4e6dc6;
                try {
                    for (let _0x192a44 of this['reporters']['values']())
                        await ((_0x4e6dc6 = (_0x11a577 = _0x192a44['reporter'])['controllerActionDone']) === null || _0x4e6dc6 === void 0x0 ? void 0x0 : _0x4e6dc6['call'](_0x11a577, _0x252b63));
                } catch (_0x3c9f1a) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x3c9f1a });
                }
            }
            async ['assertionActionStart'](_0x5c69bb) {
                var _0x1611ba, _0x7e1d44;
                try {
                    for (let _0x12347d of this['reporters']['values']())
                        await ((_0x7e1d44 = (_0x1611ba = _0x12347d['reporter'])['assertionActionStart']) === null || _0x7e1d44 === void 0x0 ? void 0x0 : _0x7e1d44['call'](_0x1611ba, _0x5c69bb));
                } catch (_0x41901f) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x41901f });
                }
            }
            async ['assertionActionDone'](_0x1aad6d) {
                var _0x155476, _0x17cb6f;
                try {
                    for (let _0x4591bf of this['reporters']['values']())
                        await ((_0x17cb6f = (_0x155476 = _0x4591bf['reporter'])['assertionActionDone']) === null || _0x17cb6f === void 0x0 ? void 0x0 : _0x17cb6f['call'](_0x155476, _0x1aad6d));
                } catch (_0xbc8d12) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0xbc8d12 });
                }
            }
            async ['taskDone'](_0x3be3db) {
                var _0x4c1b5e, _0x27f06b, _0x2f05d3, _0x1451e5;
                try {
                    for (let _0x4f8b86 of this['reporters']['values']())
                        await ((_0x27f06b = (_0x4c1b5e = _0x4f8b86['reporter'])['taskDone']) === null || _0x27f06b === void 0x0 ? void 0x0 : _0x27f06b['call'](_0x4c1b5e, _0x3be3db));
                    await this['pluginsManager']['runTaskDone'](), ((_0x1451e5 = (_0x2f05d3 = this['engine']['configuration']['gherkin']) === null || _0x2f05d3 === void 0x0 ? void 0x0 : _0x2f05d3['parser']) === null || _0x1451e5 === void 0x0 ? void 0x0 : _0x1451e5['enabled']) && (console['log'](_0xa95d8b()['yellow']('Generating\x20Acceptance\x20Criteria\x20analysis\x20Report')), _0x1a5f78['specsAnalyzer']['analyze'](this['engine']['specs']));
                } catch (_0x44203e) {
                    await (0x0, _0x5a6c67['logGenericError'])({ 'error': _0x44203e });
                }
            }
            ['registerEvents']() {
                _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterTaskStart'], async _0x51a088 => await this['taskStart'](_0x51a088)), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterTaskDone'], async _0x31252a => await this['taskDone'](_0x31252a)), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterFeatureStart'], async _0x364ad6 => await this['featureStart'](_0x364ad6['feature'], _0x364ad6['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterFeatureDone'], async _0x58c32b => await this['featureDone'](_0x58c32b['feature'], _0x58c32b['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterScenarioStart'], async _0xe1cb30 => await this['scenarioStart'](_0xe1cb30['scenario'], _0xe1cb30['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterScenarioDone'], async _0x4beb4f => await this['scenarioDone'](_0x4beb4f['scenario'], _0x4beb4f['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterStepStart'], async _0x45e14d => await this['stepStart'](_0x45e14d['step'], _0x45e14d['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterStepDone'], async _0x462e0c => await this['stepDone'](_0x462e0c['step'], _0x462e0c['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterSubStepStart'], async _0x42c0b8 => await this['subStepStart'](_0x42c0b8['subStep'], _0x42c0b8['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterSubStepDone'], async _0x5bb0f5 => await this['subStepDone'](_0x5bb0f5['subStep'], _0x5bb0f5['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterHookStart'], async _0x3da864 => await this['hookStart'](_0x3da864['name'], _0x3da864['spec'], _0x3da864['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterHookDone'], async _0xd22ca1 => await this['hookDone'](_0xd22ca1['name'], _0xd22ca1['spec'], _0xd22ca1['info'])), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterSelectorActionStart'], async _0x1ddc10 => await this['selectorActionStart'](_0x1ddc10)), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterSelectorActionDone'], async _0x4dec19 => await this['selectorActionDone'](_0x4dec19)), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterControllerActionStart'], async _0x153438 => await this['controllerActionStart'](_0x153438)), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterControllerActionDone'], async _0x1e287c => await this['controllerActionDone'](_0x1e287c)), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterAssertionActionStart'], async _0x45e19c => await this['assertionActionStart'](_0x45e19c)), _0x143845['tmEventsManger']['on'](_0x2cfcb8['TMEvents']['ReporterAssertionActionDone'], async _0x59ffc5 => await this['assertionActionDone'](_0x59ffc5));
            }
        }
    }
};
;
