// test-maker/dist/238.js
exports['id'] = 0xee, exports['ids'] = [0xee], exports['modules'] = {
    0xdcf3: (_0x5043b9, _0x4b7678, _0x400dfc) => {
        'use strict';
        const _0x480350 = _0x400dfc(0x15d1a), _0x18abcf = _0x400dfc(0x11569), _0x460955 = _0x400dfc(0x10c1d)['mkdirsSync'], _0x449af4 = _0x400dfc(0x13743)['utimesMillisSync'], _0x32ebd9 = _0x400dfc(0x7c9d);
        function _0x40e5e9(_0x168478, _0x5d0fce, _0x5d26e0) {
            typeof _0x5d26e0 === 'function' && (_0x5d26e0 = { 'filter': _0x5d26e0 });
            _0x5d26e0 = _0x5d26e0 || {}, _0x5d26e0['clobber'] = 'clobber' in _0x5d26e0 ? !!_0x5d26e0['clobber'] : !![], _0x5d26e0['overwrite'] = 'overwrite' in _0x5d26e0 ? !!_0x5d26e0['overwrite'] : _0x5d26e0['clobber'];
            _0x5d26e0['preserveTimestamps'] && process['arch'] === 'ia32' && process['emitWarning']('Using\x20the\x20preserveTimestamps\x20option\x20in\x2032-bit\x20node\x20is\x20not\x20recommended;\x0a\x0a' + '\x09see\x20https://github.com/jprichardson/node-fs-extra/issues/269', 'Warning', 'fs-extra-WARN0002');
            const {
                srcStat: _0x5c9898,
                destStat: _0x28ade9
            } = _0x32ebd9['checkPathsSync'](_0x168478, _0x5d0fce, 'copy', _0x5d26e0);
            return _0x32ebd9['checkParentPathsSync'](_0x168478, _0x5c9898, _0x5d0fce, 'copy'), _0x324431(_0x28ade9, _0x168478, _0x5d0fce, _0x5d26e0);
        }
        function _0x324431(_0x50f315, _0x2f89e0, _0x408393, _0x57c9f1) {
            if (_0x57c9f1['filter'] && !_0x57c9f1['filter'](_0x2f89e0, _0x408393))
                return;
            const _0x125a4e = _0x18abcf['dirname'](_0x408393);
            if (!_0x480350['existsSync'](_0x125a4e))
                _0x460955(_0x125a4e);
            return _0x2c6c20(_0x50f315, _0x2f89e0, _0x408393, _0x57c9f1);
        }
        function _0x1f5ffa(_0x1dda2f, _0x27c1e9, _0x14bb59, _0x4fb87c) {
            if (_0x4fb87c['filter'] && !_0x4fb87c['filter'](_0x27c1e9, _0x14bb59))
                return;
            return _0x2c6c20(_0x1dda2f, _0x27c1e9, _0x14bb59, _0x4fb87c);
        }
        function _0x2c6c20(_0x46f8f1, _0x545933, _0x2fd6ad, _0x4537f9) {
            const _0xe3af0d = _0x4537f9['dereference'] ? _0x480350['statSync'] : _0x480350['lstatSync'], _0xdef213 = _0xe3af0d(_0x545933);
            if (_0xdef213['isDirectory']())
                return _0x9ec768(_0xdef213, _0x46f8f1, _0x545933, _0x2fd6ad, _0x4537f9);
            else {
                if (_0xdef213['isFile']() || _0xdef213['isCharacterDevice']() || _0xdef213['isBlockDevice']())
                    return _0x348920(_0xdef213, _0x46f8f1, _0x545933, _0x2fd6ad, _0x4537f9);
                else {
                    if (_0xdef213['isSymbolicLink']())
                        return _0x2cf3cc(_0x46f8f1, _0x545933, _0x2fd6ad, _0x4537f9);
                    else {
                        if (_0xdef213['isSocket']())
                            throw new Error('Cannot\x20copy\x20a\x20socket\x20file:\x20' + _0x545933);
                        else {
                            if (_0xdef213['isFIFO']())
                                throw new Error('Cannot\x20copy\x20a\x20FIFO\x20pipe:\x20' + _0x545933);
                        }
                    }
                }
            }
            throw new Error('Unknown\x20file:\x20' + _0x545933);
        }
        function _0x348920(_0x473b16, _0xf8be9b, _0x15ec21, _0x11b0af, _0x95a8f7) {
            if (!_0xf8be9b)
                return _0x461eed(_0x473b16, _0x15ec21, _0x11b0af, _0x95a8f7);
            return _0x417c69(_0x473b16, _0x15ec21, _0x11b0af, _0x95a8f7);
        }
        function _0x417c69(_0x583009, _0xd2ef2a, _0x3b5fce, _0x529b2b) {
            if (_0x529b2b['overwrite'])
                return _0x480350['unlinkSync'](_0x3b5fce), _0x461eed(_0x583009, _0xd2ef2a, _0x3b5fce, _0x529b2b);
            else {
                if (_0x529b2b['errorOnExist'])
                    throw new Error('\x27' + _0x3b5fce + '\x27\x20already\x20exists');
            }
        }
        function _0x461eed(_0xee1fe, _0x8826c8, _0x46fdb5, _0x1c2cd1) {
            _0x480350['copyFileSync'](_0x8826c8, _0x46fdb5);
            if (_0x1c2cd1['preserveTimestamps'])
                _0x47a0a1(_0xee1fe['mode'], _0x8826c8, _0x46fdb5);
            return _0x515113(_0x46fdb5, _0xee1fe['mode']);
        }
        function _0x47a0a1(_0x1be5cd, _0x449773, _0xf63354) {
            if (_0x441199(_0x1be5cd))
                _0x192b63(_0xf63354, _0x1be5cd);
            return _0x66534(_0x449773, _0xf63354);
        }
        function _0x441199(_0x21fa33) {
            return (_0x21fa33 & 0x80) === 0x0;
        }
        function _0x192b63(_0x2c2860, _0x168047) {
            return _0x515113(_0x2c2860, _0x168047 | 0x80);
        }
        function _0x515113(_0x1e24c3, _0x55b25f) {
            return _0x480350['chmodSync'](_0x1e24c3, _0x55b25f);
        }
        function _0x66534(_0x38be69, _0x133503) {
            const _0x484a32 = _0x480350['statSync'](_0x38be69);
            return _0x449af4(_0x133503, _0x484a32['atime'], _0x484a32['mtime']);
        }
        function _0x9ec768(_0xe45fb7, _0x1a532d, _0x3c089f, _0x9e9099, _0x7eba9d) {
            if (!_0x1a532d)
                return _0x1f531b(_0xe45fb7['mode'], _0x3c089f, _0x9e9099, _0x7eba9d);
            return _0x66ab75(_0x3c089f, _0x9e9099, _0x7eba9d);
        }
        function _0x1f531b(_0x2a6180, _0x2abb3a, _0x595d95, _0x2bb364) {
            return _0x480350['mkdirSync'](_0x595d95), _0x66ab75(_0x2abb3a, _0x595d95, _0x2bb364), _0x515113(_0x595d95, _0x2a6180);
        }
        function _0x66ab75(_0x4d8ad4, _0x2ce8d7, _0x646de0) {
            _0x480350['readdirSync'](_0x4d8ad4)['forEach'](_0x5b846e => _0x3f9583(_0x5b846e, _0x4d8ad4, _0x2ce8d7, _0x646de0));
        }
        function _0x3f9583(_0x4ee05c, _0xc09dc9, _0x52733a, _0x5e43fd) {
            const _0x22051a = _0x18abcf['join'](_0xc09dc9, _0x4ee05c), _0x2530c3 = _0x18abcf['join'](_0x52733a, _0x4ee05c), {destStat: _0xf9b558} = _0x32ebd9['checkPathsSync'](_0x22051a, _0x2530c3, 'copy', _0x5e43fd);
            return _0x1f5ffa(_0xf9b558, _0x22051a, _0x2530c3, _0x5e43fd);
        }
        function _0x2cf3cc(_0x501490, _0x212e1f, _0x4216b0, _0x83e29b) {
            let _0x47e9a1 = _0x480350['readlinkSync'](_0x212e1f);
            _0x83e29b['dereference'] && (_0x47e9a1 = _0x18abcf['resolve'](process['cwd'](), _0x47e9a1));
            if (!_0x501490)
                return _0x480350['symlinkSync'](_0x47e9a1, _0x4216b0);
            else {
                let _0xe4ccb9;
                try {
                    _0xe4ccb9 = _0x480350['readlinkSync'](_0x4216b0);
                } catch (_0x5b51fb) {
                    if (_0x5b51fb['code'] === 'EINVAL' || _0x5b51fb['code'] === 'UNKNOWN')
                        return _0x480350['symlinkSync'](_0x47e9a1, _0x4216b0);
                    throw _0x5b51fb;
                }
                _0x83e29b['dereference'] && (_0xe4ccb9 = _0x18abcf['resolve'](process['cwd'](), _0xe4ccb9));
                if (_0x32ebd9['isSrcSubdir'](_0x47e9a1, _0xe4ccb9))
                    throw new Error('Cannot\x20copy\x20\x27' + _0x47e9a1 + '\x27\x20to\x20a\x20subdirectory\x20of\x20itself,\x20\x27' + _0xe4ccb9 + '\x27.');
                if (_0x480350['statSync'](_0x4216b0)['isDirectory']() && _0x32ebd9['isSrcSubdir'](_0xe4ccb9, _0x47e9a1))
                    throw new Error('Cannot\x20overwrite\x20\x27' + _0xe4ccb9 + '\x27\x20with\x20\x27' + _0x47e9a1 + '\x27.');
                return _0x421a9c(_0x47e9a1, _0x4216b0);
            }
        }
        function _0x421a9c(_0x16b1d2, _0x1a2f66) {
            return _0x480350['unlinkSync'](_0x1a2f66), _0x480350['symlinkSync'](_0x16b1d2, _0x1a2f66);
        }
        _0x5043b9['exports'] = _0x40e5e9;
    },
    0x11885: (_0x6cd921, _0x4847d5, _0x53e293) => {
        'use strict';
        const _0x2c5bf0 = _0x53e293(0x15d1a), _0x29e388 = _0x53e293(0x11569), _0x34d676 = _0x53e293(0x10c1d)['mkdirs'], _0x3a485c = _0x53e293(0x15330)['pathExists'], _0x3c3b88 = _0x53e293(0x13743)['utimesMillis'], _0x4729a3 = _0x53e293(0x7c9d);
        function _0x2a8f3f(_0x10dc2b, _0x3ec5f3, _0x2a52ba, _0x2dec25) {
            if (typeof _0x2a52ba === 'function' && !_0x2dec25)
                _0x2dec25 = _0x2a52ba, _0x2a52ba = {};
            else
                typeof _0x2a52ba === 'function' && (_0x2a52ba = { 'filter': _0x2a52ba });
            _0x2dec25 = _0x2dec25 || function () {
            }, _0x2a52ba = _0x2a52ba || {}, _0x2a52ba['clobber'] = 'clobber' in _0x2a52ba ? !!_0x2a52ba['clobber'] : !![], _0x2a52ba['overwrite'] = 'overwrite' in _0x2a52ba ? !!_0x2a52ba['overwrite'] : _0x2a52ba['clobber'], _0x2a52ba['preserveTimestamps'] && process['arch'] === 'ia32' && process['emitWarning']('Using\x20the\x20preserveTimestamps\x20option\x20in\x2032-bit\x20node\x20is\x20not\x20recommended;\x0a\x0a' + '\x09see\x20https://github.com/jprichardson/node-fs-extra/issues/269', 'Warning', 'fs-extra-WARN0001'), _0x4729a3['checkPaths'](_0x10dc2b, _0x3ec5f3, 'copy', _0x2a52ba, (_0x4a0528, _0x1f8189) => {
                if (_0x4a0528)
                    return _0x2dec25(_0x4a0528);
                const {
                    srcStat: _0x59b298,
                    destStat: _0x3041f9
                } = _0x1f8189;
                _0x4729a3['checkParentPaths'](_0x10dc2b, _0x59b298, _0x3ec5f3, 'copy', _0x49a2b6 => {
                    if (_0x49a2b6)
                        return _0x2dec25(_0x49a2b6);
                    if (_0x2a52ba['filter'])
                        return _0x4cc316(_0x37d365, _0x3041f9, _0x10dc2b, _0x3ec5f3, _0x2a52ba, _0x2dec25);
                    return _0x37d365(_0x3041f9, _0x10dc2b, _0x3ec5f3, _0x2a52ba, _0x2dec25);
                });
            });
        }
        function _0x37d365(_0x566e7b, _0x204faa, _0x23fea3, _0x53d6e4, _0x1f2b09) {
            const _0x49dd9f = _0x29e388['dirname'](_0x23fea3);
            _0x3a485c(_0x49dd9f, (_0x1aa2ac, _0x2afc18) => {
                if (_0x1aa2ac)
                    return _0x1f2b09(_0x1aa2ac);
                if (_0x2afc18)
                    return _0x13ec87(_0x566e7b, _0x204faa, _0x23fea3, _0x53d6e4, _0x1f2b09);
                _0x34d676(_0x49dd9f, _0x13c2b6 => {
                    if (_0x13c2b6)
                        return _0x1f2b09(_0x13c2b6);
                    return _0x13ec87(_0x566e7b, _0x204faa, _0x23fea3, _0x53d6e4, _0x1f2b09);
                });
            });
        }
        function _0x4cc316(_0x174e0e, _0x589562, _0x1963ea, _0x5d1f0a, _0x5827f2, _0x4f3b55) {
            Promise['resolve'](_0x5827f2['filter'](_0x1963ea, _0x5d1f0a))['then'](_0x47c55f => {
                if (_0x47c55f)
                    return _0x174e0e(_0x589562, _0x1963ea, _0x5d1f0a, _0x5827f2, _0x4f3b55);
                return _0x4f3b55();
            }, _0xac4b5f => _0x4f3b55(_0xac4b5f));
        }
        function _0x3efa60(_0x35e3aa, _0x34f3b8, _0x8b98d1, _0xaf1bf9, _0x332e92) {
            if (_0xaf1bf9['filter'])
                return _0x4cc316(_0x13ec87, _0x35e3aa, _0x34f3b8, _0x8b98d1, _0xaf1bf9, _0x332e92);
            return _0x13ec87(_0x35e3aa, _0x34f3b8, _0x8b98d1, _0xaf1bf9, _0x332e92);
        }
        function _0x13ec87(_0x26bf79, _0x11de80, _0x350610, _0x2cdee2, _0x5997c5) {
            const _0xd3b80e = _0x2cdee2['dereference'] ? _0x2c5bf0['stat'] : _0x2c5bf0['lstat'];
            _0xd3b80e(_0x11de80, (_0x1179a2, _0x2dd1d8) => {
                if (_0x1179a2)
                    return _0x5997c5(_0x1179a2);
                if (_0x2dd1d8['isDirectory']())
                    return _0x47b56c(_0x2dd1d8, _0x26bf79, _0x11de80, _0x350610, _0x2cdee2, _0x5997c5);
                else {
                    if (_0x2dd1d8['isFile']() || _0x2dd1d8['isCharacterDevice']() || _0x2dd1d8['isBlockDevice']())
                        return _0x47061e(_0x2dd1d8, _0x26bf79, _0x11de80, _0x350610, _0x2cdee2, _0x5997c5);
                    else {
                        if (_0x2dd1d8['isSymbolicLink']())
                            return _0x4bbfc9(_0x26bf79, _0x11de80, _0x350610, _0x2cdee2, _0x5997c5);
                        else {
                            if (_0x2dd1d8['isSocket']())
                                return _0x5997c5(new Error('Cannot\x20copy\x20a\x20socket\x20file:\x20' + _0x11de80));
                            else {
                                if (_0x2dd1d8['isFIFO']())
                                    return _0x5997c5(new Error('Cannot\x20copy\x20a\x20FIFO\x20pipe:\x20' + _0x11de80));
                            }
                        }
                    }
                }
                return _0x5997c5(new Error('Unknown\x20file:\x20' + _0x11de80));
            });
        }
        function _0x47061e(_0x4f1506, _0x21eabc, _0x13e13a, _0x44f40b, _0x188b4b, _0x21e6e4) {
            if (!_0x21eabc)
                return _0x1eebe3(_0x4f1506, _0x13e13a, _0x44f40b, _0x188b4b, _0x21e6e4);
            return _0x10b4a4(_0x4f1506, _0x13e13a, _0x44f40b, _0x188b4b, _0x21e6e4);
        }
        function _0x10b4a4(_0x17a65b, _0xe3f19b, _0x1c6549, _0x4c0efe, _0x25bfda) {
            if (_0x4c0efe['overwrite'])
                _0x2c5bf0['unlink'](_0x1c6549, _0x3c1451 => {
                    if (_0x3c1451)
                        return _0x25bfda(_0x3c1451);
                    return _0x1eebe3(_0x17a65b, _0xe3f19b, _0x1c6549, _0x4c0efe, _0x25bfda);
                });
            else {
                if (_0x4c0efe['errorOnExist'])
                    return _0x25bfda(new Error('\x27' + _0x1c6549 + '\x27\x20already\x20exists'));
                else
                    return _0x25bfda();
            }
        }
        function _0x1eebe3(_0x108895, _0x44eb91, _0x39e192, _0x471ea7, _0x535217) {
            _0x2c5bf0['copyFile'](_0x44eb91, _0x39e192, _0x503582 => {
                if (_0x503582)
                    return _0x535217(_0x503582);
                if (_0x471ea7['preserveTimestamps'])
                    return _0x376a24(_0x108895['mode'], _0x44eb91, _0x39e192, _0x535217);
                return _0x468f4b(_0x39e192, _0x108895['mode'], _0x535217);
            });
        }
        function _0x376a24(_0xeb2de7, _0x55fb0a, _0x2a0141, _0x59f7fb) {
            if (_0x19cd5f(_0xeb2de7))
                return _0xdb9f85(_0x2a0141, _0xeb2de7, _0x127ee8 => {
                    if (_0x127ee8)
                        return _0x59f7fb(_0x127ee8);
                    return _0x56508d(_0xeb2de7, _0x55fb0a, _0x2a0141, _0x59f7fb);
                });
            return _0x56508d(_0xeb2de7, _0x55fb0a, _0x2a0141, _0x59f7fb);
        }
        function _0x19cd5f(_0x3849cf) {
            return (_0x3849cf & 0x80) === 0x0;
        }
        function _0xdb9f85(_0x1ce074, _0x3c7892, _0x5adfe2) {
            return _0x468f4b(_0x1ce074, _0x3c7892 | 0x80, _0x5adfe2);
        }
        function _0x56508d(_0x2815e5, _0x5e39c8, _0x4069c2, _0x3ac9d0) {
            _0x1d6eb4(_0x5e39c8, _0x4069c2, _0x3ae632 => {
                if (_0x3ae632)
                    return _0x3ac9d0(_0x3ae632);
                return _0x468f4b(_0x4069c2, _0x2815e5, _0x3ac9d0);
            });
        }
        function _0x468f4b(_0x3e1531, _0x5d5946, _0x161680) {
            return _0x2c5bf0['chmod'](_0x3e1531, _0x5d5946, _0x161680);
        }
        function _0x1d6eb4(_0x418695, _0x3f3c50, _0x52e341) {
            _0x2c5bf0['stat'](_0x418695, (_0x395f7e, _0x26727d) => {
                if (_0x395f7e)
                    return _0x52e341(_0x395f7e);
                return _0x3c3b88(_0x3f3c50, _0x26727d['atime'], _0x26727d['mtime'], _0x52e341);
            });
        }
        function _0x47b56c(_0x27fba5, _0x3dc895, _0x57cf4f, _0x236586, _0x33f8a4, _0x3b8f9a) {
            if (!_0x3dc895)
                return _0x4edb85(_0x27fba5['mode'], _0x57cf4f, _0x236586, _0x33f8a4, _0x3b8f9a);
            return _0x1843e5(_0x57cf4f, _0x236586, _0x33f8a4, _0x3b8f9a);
        }
        function _0x4edb85(_0x45a231, _0x5186b8, _0x2edc56, _0x524d49, _0x170de8) {
            _0x2c5bf0['mkdir'](_0x2edc56, _0x16b83f => {
                if (_0x16b83f)
                    return _0x170de8(_0x16b83f);
                _0x1843e5(_0x5186b8, _0x2edc56, _0x524d49, _0x12642a => {
                    if (_0x12642a)
                        return _0x170de8(_0x12642a);
                    return _0x468f4b(_0x2edc56, _0x45a231, _0x170de8);
                });
            });
        }
        function _0x1843e5(_0x29bfb7, _0x121613, _0x416fd5, _0x18b2c5) {
            _0x2c5bf0['readdir'](_0x29bfb7, (_0x99aaa4, _0x45ef05) => {
                if (_0x99aaa4)
                    return _0x18b2c5(_0x99aaa4);
                return _0x24aafd(_0x45ef05, _0x29bfb7, _0x121613, _0x416fd5, _0x18b2c5);
            });
        }
        function _0x24aafd(_0x4fb308, _0x5ccdbb, _0x567b81, _0xdcc373, _0x280707) {
            const _0x506e10 = _0x4fb308['pop']();
            if (!_0x506e10)
                return _0x280707();
            return _0x3591cb(_0x4fb308, _0x506e10, _0x5ccdbb, _0x567b81, _0xdcc373, _0x280707);
        }
        function _0x3591cb(_0x384fe8, _0x2fcbdb, _0x43fa3c, _0x59324e, _0x29ee7f, _0x235b7c) {
            const _0x513aec = _0x29e388['join'](_0x43fa3c, _0x2fcbdb), _0x19da95 = _0x29e388['join'](_0x59324e, _0x2fcbdb);
            _0x4729a3['checkPaths'](_0x513aec, _0x19da95, 'copy', _0x29ee7f, (_0x50f103, _0x2b6fad) => {
                if (_0x50f103)
                    return _0x235b7c(_0x50f103);
                const {destStat: _0x59a475} = _0x2b6fad;
                _0x3efa60(_0x59a475, _0x513aec, _0x19da95, _0x29ee7f, _0x513586 => {
                    if (_0x513586)
                        return _0x235b7c(_0x513586);
                    return _0x24aafd(_0x384fe8, _0x43fa3c, _0x59324e, _0x29ee7f, _0x235b7c);
                });
            });
        }
        function _0x4bbfc9(_0x2472bc, _0xea83c6, _0x3164d0, _0x167770, _0x3217f7) {
            _0x2c5bf0['readlink'](_0xea83c6, (_0x38fb18, _0x508279) => {
                if (_0x38fb18)
                    return _0x3217f7(_0x38fb18);
                _0x167770['dereference'] && (_0x508279 = _0x29e388['resolve'](process['cwd'](), _0x508279));
                if (!_0x2472bc)
                    return _0x2c5bf0['symlink'](_0x508279, _0x3164d0, _0x3217f7);
                else
                    _0x2c5bf0['readlink'](_0x3164d0, (_0x3be731, _0x489d1a) => {
                        if (_0x3be731) {
                            if (_0x3be731['code'] === 'EINVAL' || _0x3be731['code'] === 'UNKNOWN')
                                return _0x2c5bf0['symlink'](_0x508279, _0x3164d0, _0x3217f7);
                            return _0x3217f7(_0x3be731);
                        }
                        _0x167770['dereference'] && (_0x489d1a = _0x29e388['resolve'](process['cwd'](), _0x489d1a));
                        if (_0x4729a3['isSrcSubdir'](_0x508279, _0x489d1a))
                            return _0x3217f7(new Error('Cannot\x20copy\x20\x27' + _0x508279 + '\x27\x20to\x20a\x20subdirectory\x20of\x20itself,\x20\x27' + _0x489d1a + '\x27.'));
                        if (_0x2472bc['isDirectory']() && _0x4729a3['isSrcSubdir'](_0x489d1a, _0x508279))
                            return _0x3217f7(new Error('Cannot\x20overwrite\x20\x27' + _0x489d1a + '\x27\x20with\x20\x27' + _0x508279 + '\x27.'));
                        return _0x10ce6a(_0x508279, _0x3164d0, _0x3217f7);
                    });
            });
        }
        function _0x10ce6a(_0x3e331f, _0x591726, _0x5b2211) {
            _0x2c5bf0['unlink'](_0x591726, _0x177234 => {
                if (_0x177234)
                    return _0x5b2211(_0x177234);
                return _0x2c5bf0['symlink'](_0x3e331f, _0x591726, _0x5b2211);
            });
        }
        _0x6cd921['exports'] = _0x2a8f3f;
    },
    0xb839: (_0xd5df02, _0x564c3b, _0x59a69c) => {
        'use strict';
        const _0x4726b = _0x59a69c(0x15f6a)['fromCallback'];
        _0xd5df02['exports'] = {
            'copy': _0x4726b(_0x59a69c(0x11885)),
            'copySync': _0x59a69c(0xdcf3)
        };
    },
    0x112ad: (_0x3c64c5, _0x915262, _0xfc47d3) => {
        'use strict';
        const _0x493574 = _0xfc47d3(0x15f6a)['fromPromise'], _0x1686dc = _0xfc47d3(0x15a17), _0x4cc42b = _0xfc47d3(0x11569), _0x525821 = _0xfc47d3(0x10c1d), _0x244a58 = _0xfc47d3(0x983d), _0x1d50a3 = _0x493574(async function _0x7e23d(_0x4887e7) {
                let _0xef0151;
                try {
                    _0xef0151 = await _0x1686dc['readdir'](_0x4887e7);
                } catch {
                    return _0x525821['mkdirs'](_0x4887e7);
                }
                return Promise['all'](_0xef0151['map'](_0x3ed7b8 => _0x244a58['remove'](_0x4cc42b['join'](_0x4887e7, _0x3ed7b8))));
            });
        function _0xd1f2e(_0x13b313) {
            let _0x45de8d;
            try {
                _0x45de8d = _0x1686dc['readdirSync'](_0x13b313);
            } catch {
                return _0x525821['mkdirsSync'](_0x13b313);
            }
            _0x45de8d['forEach'](_0x1a512b => {
                _0x1a512b = _0x4cc42b['join'](_0x13b313, _0x1a512b), _0x244a58['removeSync'](_0x1a512b);
            });
        }
        _0x3c64c5['exports'] = {
            'emptyDirSync': _0xd1f2e,
            'emptydirSync': _0xd1f2e,
            'emptyDir': _0x1d50a3,
            'emptydir': _0x1d50a3
        };
    },
    0x2c43: (_0x365d21, _0x415035, _0x4ce255) => {
        'use strict';
        const _0x504cbb = _0x4ce255(0x15f6a)['fromCallback'], _0x38ca3a = _0x4ce255(0x11569), _0x252412 = _0x4ce255(0x15d1a), _0x157b54 = _0x4ce255(0x10c1d);
        function _0x23417f(_0x3543a3, _0x1b041e) {
            function _0x3f5cd4() {
                _0x252412['writeFile'](_0x3543a3, '', _0x555e75 => {
                    if (_0x555e75)
                        return _0x1b041e(_0x555e75);
                    _0x1b041e();
                });
            }
            _0x252412['stat'](_0x3543a3, (_0x3af9bc, _0x57f11b) => {
                if (!_0x3af9bc && _0x57f11b['isFile']())
                    return _0x1b041e();
                const _0x2e151c = _0x38ca3a['dirname'](_0x3543a3);
                _0x252412['stat'](_0x2e151c, (_0x2c04bc, _0x4487a0) => {
                    if (_0x2c04bc) {
                        if (_0x2c04bc['code'] === 'ENOENT')
                            return _0x157b54['mkdirs'](_0x2e151c, _0x132b71 => {
                                if (_0x132b71)
                                    return _0x1b041e(_0x132b71);
                                _0x3f5cd4();
                            });
                        return _0x1b041e(_0x2c04bc);
                    }
                    if (_0x4487a0['isDirectory']())
                        _0x3f5cd4();
                    else
                        _0x252412['readdir'](_0x2e151c, _0x571b0c => {
                            if (_0x571b0c)
                                return _0x1b041e(_0x571b0c);
                        });
                });
            });
        }
        function _0x1795a7(_0x3141e1) {
            let _0x39bf62;
            try {
                _0x39bf62 = _0x252412['statSync'](_0x3141e1);
            } catch {
            }
            if (_0x39bf62 && _0x39bf62['isFile']())
                return;
            const _0x214af4 = _0x38ca3a['dirname'](_0x3141e1);
            try {
                !_0x252412['statSync'](_0x214af4)['isDirectory']() && _0x252412['readdirSync'](_0x214af4);
            } catch (_0x58230e) {
                if (_0x58230e && _0x58230e['code'] === 'ENOENT')
                    _0x157b54['mkdirsSync'](_0x214af4);
                else
                    throw _0x58230e;
            }
            _0x252412['writeFileSync'](_0x3141e1, '');
        }
        _0x365d21['exports'] = {
            'createFile': _0x504cbb(_0x23417f),
            'createFileSync': _0x1795a7
        };
    },
    0x143a2: (_0x562241, _0x5b4c7b, _0x52290f) => {
        'use strict';
        const {
                createFile: _0x47109a,
                createFileSync: _0x54722d
            } = _0x52290f(0x2c43), {
                createLink: _0x2f97e9,
                createLinkSync: _0x36c66e
            } = _0x52290f(0x150ea), {
                createSymlink: _0x5a247b,
                createSymlinkSync: _0x545295
            } = _0x52290f(0x16c50);
        _0x562241['exports'] = {
            'createFile': _0x47109a,
            'createFileSync': _0x54722d,
            'ensureFile': _0x47109a,
            'ensureFileSync': _0x54722d,
            'createLink': _0x2f97e9,
            'createLinkSync': _0x36c66e,
            'ensureLink': _0x2f97e9,
            'ensureLinkSync': _0x36c66e,
            'createSymlink': _0x5a247b,
            'createSymlinkSync': _0x545295,
            'ensureSymlink': _0x5a247b,
            'ensureSymlinkSync': _0x545295
        };
    },
    0x150ea: (_0x3a91b2, _0x3005c5, _0x3661e8) => {
        'use strict';
        const _0x4d48bb = _0x3661e8(0x15f6a)['fromCallback'], _0x45a451 = _0x3661e8(0x11569), _0x4648b8 = _0x3661e8(0x15d1a), _0x1d15d5 = _0x3661e8(0x10c1d), _0x118611 = _0x3661e8(0x15330)['pathExists'], {areIdentical: _0x46dbcd} = _0x3661e8(0x7c9d);
        function _0x35b1fe(_0xaf224a, _0x216982, _0x1019f4) {
            function _0x151b56(_0x436af8, _0x49d3f6) {
                _0x4648b8['link'](_0x436af8, _0x49d3f6, _0x914fe7 => {
                    if (_0x914fe7)
                        return _0x1019f4(_0x914fe7);
                    _0x1019f4(null);
                });
            }
            _0x4648b8['lstat'](_0x216982, (_0x37e8db, _0x20fe1e) => {
                _0x4648b8['lstat'](_0xaf224a, (_0x18d420, _0x5b3e32) => {
                    if (_0x18d420)
                        return _0x18d420['message'] = _0x18d420['message']['replace']('lstat', 'ensureLink'), _0x1019f4(_0x18d420);
                    if (_0x20fe1e && _0x46dbcd(_0x5b3e32, _0x20fe1e))
                        return _0x1019f4(null);
                    const _0x16ddb8 = _0x45a451['dirname'](_0x216982);
                    _0x118611(_0x16ddb8, (_0x27c90a, _0x16f921) => {
                        if (_0x27c90a)
                            return _0x1019f4(_0x27c90a);
                        if (_0x16f921)
                            return _0x151b56(_0xaf224a, _0x216982);
                        _0x1d15d5['mkdirs'](_0x16ddb8, _0x361432 => {
                            if (_0x361432)
                                return _0x1019f4(_0x361432);
                            _0x151b56(_0xaf224a, _0x216982);
                        });
                    });
                });
            });
        }
        function _0x539ea1(_0x30a13e, _0x33fff4) {
            let _0x525d2f;
            try {
                _0x525d2f = _0x4648b8['lstatSync'](_0x33fff4);
            } catch {
            }
            try {
                const _0x24ecbb = _0x4648b8['lstatSync'](_0x30a13e);
                if (_0x525d2f && _0x46dbcd(_0x24ecbb, _0x525d2f))
                    return;
            } catch (_0x3674ca) {
                _0x3674ca['message'] = _0x3674ca['message']['replace']('lstat', 'ensureLink');
                throw _0x3674ca;
            }
            const _0x202fe1 = _0x45a451['dirname'](_0x33fff4), _0x1d9697 = _0x4648b8['existsSync'](_0x202fe1);
            if (_0x1d9697)
                return _0x4648b8['linkSync'](_0x30a13e, _0x33fff4);
            return _0x1d15d5['mkdirsSync'](_0x202fe1), _0x4648b8['linkSync'](_0x30a13e, _0x33fff4);
        }
        _0x3a91b2['exports'] = {
            'createLink': _0x4d48bb(_0x35b1fe),
            'createLinkSync': _0x539ea1
        };
    },
    0x1496b: (_0xfacc73, _0x471b41, _0x2734eb) => {
        'use strict';
        const _0x1c5736 = _0x2734eb(0x11569), _0x1f002c = _0x2734eb(0x15d1a), _0x148692 = _0x2734eb(0x15330)['pathExists'];
        function _0x2f598a(_0x2dafb0, _0x58d14e, _0x3394cc) {
            if (_0x1c5736['isAbsolute'](_0x2dafb0))
                return _0x1f002c['lstat'](_0x2dafb0, _0x9b35ea => {
                    if (_0x9b35ea)
                        return _0x9b35ea['message'] = _0x9b35ea['message']['replace']('lstat', 'ensureSymlink'), _0x3394cc(_0x9b35ea);
                    return _0x3394cc(null, {
                        'toCwd': _0x2dafb0,
                        'toDst': _0x2dafb0
                    });
                });
            else {
                const _0x121ed0 = _0x1c5736['dirname'](_0x58d14e), _0x596ec1 = _0x1c5736['join'](_0x121ed0, _0x2dafb0);
                return _0x148692(_0x596ec1, (_0x56ae79, _0x8d5211) => {
                    if (_0x56ae79)
                        return _0x3394cc(_0x56ae79);
                    return _0x8d5211 ? _0x3394cc(null, {
                        'toCwd': _0x596ec1,
                        'toDst': _0x2dafb0
                    }) : _0x1f002c['lstat'](_0x2dafb0, _0x530c0a => {
                        if (_0x530c0a)
                            return _0x530c0a['message'] = _0x530c0a['message']['replace']('lstat', 'ensureSymlink'), _0x3394cc(_0x530c0a);
                        return _0x3394cc(null, {
                            'toCwd': _0x2dafb0,
                            'toDst': _0x1c5736['relative'](_0x121ed0, _0x2dafb0)
                        });
                    });
                });
            }
        }
        function _0x22a7ff(_0x312462, _0x48020f) {
            let _0x20383a;
            if (_0x1c5736['isAbsolute'](_0x312462)) {
                _0x20383a = _0x1f002c['existsSync'](_0x312462);
                if (!_0x20383a)
                    throw new Error('absolute\x20srcpath\x20does\x20not\x20exist');
                return {
                    'toCwd': _0x312462,
                    'toDst': _0x312462
                };
            } else {
                const _0x3e3c99 = _0x1c5736['dirname'](_0x48020f), _0x52a8b2 = _0x1c5736['join'](_0x3e3c99, _0x312462);
                _0x20383a = _0x1f002c['existsSync'](_0x52a8b2);
                if (_0x20383a)
                    return {
                        'toCwd': _0x52a8b2,
                        'toDst': _0x312462
                    };
                else {
                    _0x20383a = _0x1f002c['existsSync'](_0x312462);
                    if (!_0x20383a)
                        throw new Error('relative\x20srcpath\x20does\x20not\x20exist');
                    return {
                        'toCwd': _0x312462,
                        'toDst': _0x1c5736['relative'](_0x3e3c99, _0x312462)
                    };
                }
            }
        }
        _0xfacc73['exports'] = {
            'symlinkPaths': _0x2f598a,
            'symlinkPathsSync': _0x22a7ff
        };
    },
    0x1ca0: (_0x434835, _0xc8fde3, _0x447eff) => {
        'use strict';
        const _0x1e9c5e = _0x447eff(0x15d1a);
        function _0x2e7169(_0x4cbdfd, _0x3a746c, _0x38741e) {
            _0x38741e = typeof _0x3a746c === 'function' ? _0x3a746c : _0x38741e, _0x3a746c = typeof _0x3a746c === 'function' ? ![] : _0x3a746c;
            if (_0x3a746c)
                return _0x38741e(null, _0x3a746c);
            _0x1e9c5e['lstat'](_0x4cbdfd, (_0x2fbddf, _0x2a15e9) => {
                if (_0x2fbddf)
                    return _0x38741e(null, 'file');
                _0x3a746c = _0x2a15e9 && _0x2a15e9['isDirectory']() ? 'dir' : 'file', _0x38741e(null, _0x3a746c);
            });
        }
        function _0xfd7b9b(_0x3d791a, _0x12619b) {
            let _0x4ddf5e;
            if (_0x12619b)
                return _0x12619b;
            try {
                _0x4ddf5e = _0x1e9c5e['lstatSync'](_0x3d791a);
            } catch {
                return 'file';
            }
            return _0x4ddf5e && _0x4ddf5e['isDirectory']() ? 'dir' : 'file';
        }
        _0x434835['exports'] = {
            'symlinkType': _0x2e7169,
            'symlinkTypeSync': _0xfd7b9b
        };
    },
    0x16c50: (_0x3e0e4e, _0x91630, _0x255f3a) => {
        'use strict';
        const _0x3f45cf = _0x255f3a(0x15f6a)['fromCallback'], _0x2b6491 = _0x255f3a(0x11569), _0x28d672 = _0x255f3a(0x15a17), _0x304242 = _0x255f3a(0x10c1d), _0x5e7552 = _0x304242['mkdirs'], _0x2b6052 = _0x304242['mkdirsSync'], _0x25954f = _0x255f3a(0x1496b), _0x44dd69 = _0x25954f['symlinkPaths'], _0x520ffc = _0x25954f['symlinkPathsSync'], _0x212fbd = _0x255f3a(0x1ca0), _0x3d06c7 = _0x212fbd['symlinkType'], _0x52ce73 = _0x212fbd['symlinkTypeSync'], _0x28fbef = _0x255f3a(0x15330)['pathExists'], {areIdentical: _0xaa5606} = _0x255f3a(0x7c9d);
        function _0x1bd1ad(_0x317187, _0x5dd53e, _0x4124e5, _0x445021) {
            _0x445021 = typeof _0x4124e5 === 'function' ? _0x4124e5 : _0x445021, _0x4124e5 = typeof _0x4124e5 === 'function' ? ![] : _0x4124e5, _0x28d672['lstat'](_0x5dd53e, (_0xbffca1, _0x55bb2d) => {
                if (!_0xbffca1 && _0x55bb2d['isSymbolicLink']())
                    Promise['all']([
                        _0x28d672['stat'](_0x317187),
                        _0x28d672['stat'](_0x5dd53e)
                    ])['then'](([_0x3c39db, _0xe66798]) => {
                        if (_0xaa5606(_0x3c39db, _0xe66798))
                            return _0x445021(null);
                        _0xaed7ba(_0x317187, _0x5dd53e, _0x4124e5, _0x445021);
                    });
                else
                    _0xaed7ba(_0x317187, _0x5dd53e, _0x4124e5, _0x445021);
            });
        }
        function _0xaed7ba(_0x337955, _0x2e96d6, _0x3240a6, _0x49e285) {
            _0x44dd69(_0x337955, _0x2e96d6, (_0x155cfe, _0x33f68e) => {
                if (_0x155cfe)
                    return _0x49e285(_0x155cfe);
                _0x337955 = _0x33f68e['toDst'], _0x3d06c7(_0x33f68e['toCwd'], _0x3240a6, (_0xe79a96, _0xa81e52) => {
                    if (_0xe79a96)
                        return _0x49e285(_0xe79a96);
                    const _0x50ac54 = _0x2b6491['dirname'](_0x2e96d6);
                    _0x28fbef(_0x50ac54, (_0x2935d0, _0x3547ad) => {
                        if (_0x2935d0)
                            return _0x49e285(_0x2935d0);
                        if (_0x3547ad)
                            return _0x28d672['symlink'](_0x337955, _0x2e96d6, _0xa81e52, _0x49e285);
                        _0x5e7552(_0x50ac54, _0x336672 => {
                            if (_0x336672)
                                return _0x49e285(_0x336672);
                            _0x28d672['symlink'](_0x337955, _0x2e96d6, _0xa81e52, _0x49e285);
                        });
                    });
                });
            });
        }
        function _0xb0842c(_0x5462f1, _0x4a993f, _0xadcff4) {
            let _0x222d2e;
            try {
                _0x222d2e = _0x28d672['lstatSync'](_0x4a993f);
            } catch {
            }
            if (_0x222d2e && _0x222d2e['isSymbolicLink']()) {
                const _0x1737d3 = _0x28d672['statSync'](_0x5462f1), _0x2ffc42 = _0x28d672['statSync'](_0x4a993f);
                if (_0xaa5606(_0x1737d3, _0x2ffc42))
                    return;
            }
            const _0x2f11a2 = _0x520ffc(_0x5462f1, _0x4a993f);
            _0x5462f1 = _0x2f11a2['toDst'], _0xadcff4 = _0x52ce73(_0x2f11a2['toCwd'], _0xadcff4);
            const _0x16f3e8 = _0x2b6491['dirname'](_0x4a993f), _0x2c26e2 = _0x28d672['existsSync'](_0x16f3e8);
            if (_0x2c26e2)
                return _0x28d672['symlinkSync'](_0x5462f1, _0x4a993f, _0xadcff4);
            return _0x2b6052(_0x16f3e8), _0x28d672['symlinkSync'](_0x5462f1, _0x4a993f, _0xadcff4);
        }
        _0x3e0e4e['exports'] = {
            'createSymlink': _0x3f45cf(_0x1bd1ad),
            'createSymlinkSync': _0xb0842c
        };
    },
    0x15a17: (_0x297253, _0x10e6e7, _0x675d86) => {
        'use strict';
        const _0x36b5c8 = _0x675d86(0x15f6a)['fromCallback'], _0x3da60c = _0x675d86(0x15d1a), _0x45994b = [
                'access',
                'appendFile',
                'chmod',
                'chown',
                'close',
                'copyFile',
                'fchmod',
                'fchown',
                'fdatasync',
                'fstat',
                'fsync',
                'ftruncate',
                'futimes',
                'lchmod',
                'lchown',
                'link',
                'lstat',
                'mkdir',
                'mkdtemp',
                'open',
                'opendir',
                'readdir',
                'readFile',
                'readlink',
                'realpath',
                'rename',
                'rm',
                'rmdir',
                'stat',
                'symlink',
                'truncate',
                'unlink',
                'utimes',
                'writeFile'
            ]['filter'](_0x5e3743 => {
                return typeof _0x3da60c[_0x5e3743] === 'function';
            });
        Object['assign'](_0x10e6e7, _0x3da60c), _0x45994b['forEach'](_0x5cd29f => {
            _0x10e6e7[_0x5cd29f] = _0x36b5c8(_0x3da60c[_0x5cd29f]);
        }), _0x10e6e7['exists'] = function (_0x7db996, _0x48c19f) {
            if (typeof _0x48c19f === 'function')
                return _0x3da60c['exists'](_0x7db996, _0x48c19f);
            return new Promise(_0x38c285 => {
                return _0x3da60c['exists'](_0x7db996, _0x38c285);
            });
        }, _0x10e6e7['read'] = function (_0x15a47c, _0x1daeea, _0xc31daf, _0x553d71, _0xe10f7f, _0x43e00d) {
            if (typeof _0x43e00d === 'function')
                return _0x3da60c['read'](_0x15a47c, _0x1daeea, _0xc31daf, _0x553d71, _0xe10f7f, _0x43e00d);
            return new Promise((_0x5c83fb, _0x15e79a) => {
                _0x3da60c['read'](_0x15a47c, _0x1daeea, _0xc31daf, _0x553d71, _0xe10f7f, (_0x24e7d8, _0x6b236a, _0x50e794) => {
                    if (_0x24e7d8)
                        return _0x15e79a(_0x24e7d8);
                    _0x5c83fb({
                        'bytesRead': _0x6b236a,
                        'buffer': _0x50e794
                    });
                });
            });
        }, _0x10e6e7['write'] = function (_0x2a761b, _0x2ace1c, ..._0x504eea) {
            if (typeof _0x504eea[_0x504eea['length'] - 0x1] === 'function')
                return _0x3da60c['write'](_0x2a761b, _0x2ace1c, ..._0x504eea);
            return new Promise((_0x1b142b, _0x241ac4) => {
                _0x3da60c['write'](_0x2a761b, _0x2ace1c, ..._0x504eea, (_0x157f19, _0x46a7b2, _0x450cad) => {
                    if (_0x157f19)
                        return _0x241ac4(_0x157f19);
                    _0x1b142b({
                        'bytesWritten': _0x46a7b2,
                        'buffer': _0x450cad
                    });
                });
            });
        }, typeof _0x3da60c['writev'] === 'function' && (_0x10e6e7['writev'] = function (_0x33042b, _0xebc7d3, ..._0x25f3a8) {
            if (typeof _0x25f3a8[_0x25f3a8['length'] - 0x1] === 'function')
                return _0x3da60c['writev'](_0x33042b, _0xebc7d3, ..._0x25f3a8);
            return new Promise((_0x338e96, _0xc515d8) => {
                _0x3da60c['writev'](_0x33042b, _0xebc7d3, ..._0x25f3a8, (_0x2fbbaa, _0x2fedb6, _0x40810c) => {
                    if (_0x2fbbaa)
                        return _0xc515d8(_0x2fbbaa);
                    _0x338e96({
                        'bytesWritten': _0x2fedb6,
                        'buffers': _0x40810c
                    });
                });
            });
        }), typeof _0x3da60c['realpath']['native'] === 'function' ? _0x10e6e7['realpath']['native'] = _0x36b5c8(_0x3da60c['realpath']['native']) : process['emitWarning']('fs.realpath.native\x20is\x20not\x20a\x20function.\x20Is\x20fs\x20being\x20monkey-patched?', 'Warning', 'fs-extra-WARN0003');
    },
    0x17fbe: (_0x1fc16e, _0x4d9eba, _0x55309a) => {
        'use strict';
        _0x1fc16e['exports'] = {
            ..._0x55309a(0x15a17),
            ..._0x55309a(0xb839),
            ..._0x55309a(0x112ad),
            ..._0x55309a(0x143a2),
            ..._0x55309a(0x14bf2),
            ..._0x55309a(0x10c1d),
            ..._0x55309a(0x1462d),
            ..._0x55309a(0x7096),
            ..._0x55309a(0x15330),
            ..._0x55309a(0x983d)
        };
    },
    0x14bf2: (_0x38ffa9, _0x5a9b85, _0x392314) => {
        'use strict';
        const _0x408e7a = _0x392314(0x15f6a)['fromPromise'], _0x275d0f = _0x392314(0xd091);
        _0x275d0f['outputJson'] = _0x408e7a(_0x392314(0x10fbe)), _0x275d0f['outputJsonSync'] = _0x392314(0x43a8), _0x275d0f['outputJSON'] = _0x275d0f['outputJson'], _0x275d0f['outputJSONSync'] = _0x275d0f['outputJsonSync'], _0x275d0f['writeJSON'] = _0x275d0f['writeJson'], _0x275d0f['writeJSONSync'] = _0x275d0f['writeJsonSync'], _0x275d0f['readJSON'] = _0x275d0f['readJson'], _0x275d0f['readJSONSync'] = _0x275d0f['readJsonSync'], _0x38ffa9['exports'] = _0x275d0f;
    },
    0xd091: (_0x1f74e1, _0x3d2288, _0x3cf23a) => {
        'use strict';
        const _0x22a38f = _0x3cf23a(0x8517);
        _0x1f74e1['exports'] = {
            'readJson': _0x22a38f['readFile'],
            'readJsonSync': _0x22a38f['readFileSync'],
            'writeJson': _0x22a38f['writeFile'],
            'writeJsonSync': _0x22a38f['writeFileSync']
        };
    },
    0x43a8: (_0x290958, _0x48ce8c, _0x208edd) => {
        'use strict';
        const {stringify: _0x29df1d} = _0x208edd(0x14415), {outputFileSync: _0x3d6df1} = _0x208edd(0x7096);
        function _0x37fe13(_0x40023a, _0x268ba2, _0x262e69) {
            const _0x5611dc = _0x29df1d(_0x268ba2, _0x262e69);
            _0x3d6df1(_0x40023a, _0x5611dc, _0x262e69);
        }
        _0x290958['exports'] = _0x37fe13;
    },
    0x10fbe: (_0x52e5cc, _0x1a31dc, _0x35abe4) => {
        'use strict';
        const {stringify: _0x39355b} = _0x35abe4(0x14415), {outputFile: _0x363d3d} = _0x35abe4(0x7096);
        async function _0x199e82(_0xd5a33d, _0x48a070, _0x45a78c = {}) {
            const _0x3fc102 = _0x39355b(_0x48a070, _0x45a78c);
            await _0x363d3d(_0xd5a33d, _0x3fc102, _0x45a78c);
        }
        _0x52e5cc['exports'] = _0x199e82;
    },
    0x10c1d: (_0x3d6ac2, _0x766b39, _0x1923c8) => {
        'use strict';
        const _0xef76f2 = _0x1923c8(0x15f6a)['fromPromise'], {
                makeDir: _0x139e4d,
                makeDirSync: _0x1d916c
            } = _0x1923c8(0x1010), _0x464241 = _0xef76f2(_0x139e4d);
        _0x3d6ac2['exports'] = {
            'mkdirs': _0x464241,
            'mkdirsSync': _0x1d916c,
            'mkdirp': _0x464241,
            'mkdirpSync': _0x1d916c,
            'ensureDir': _0x464241,
            'ensureDirSync': _0x1d916c
        };
    },
    0x1010: (_0x4e58d9, _0x395d3e, _0x4cfe1e) => {
        'use strict';
        const _0x1118ce = _0x4cfe1e(0x15a17), {checkPath: _0x558c8f} = _0x4cfe1e(0x4517), _0x57abab = _0x4a0dfc => {
                const _0x26048b = { 'mode': 0x1ff };
                if (typeof _0x4a0dfc === 'number')
                    return _0x4a0dfc;
                return {
                    ..._0x26048b,
                    ..._0x4a0dfc
                }['mode'];
            };
        _0x4e58d9['exports']['makeDir'] = async (_0x2e538d, _0xd36940) => {
            return _0x558c8f(_0x2e538d), _0x1118ce['mkdir'](_0x2e538d, {
                'mode': _0x57abab(_0xd36940),
                'recursive': !![]
            });
        }, _0x4e58d9['exports']['makeDirSync'] = (_0x1ecf6b, _0x981d46) => {
            return _0x558c8f(_0x1ecf6b), _0x1118ce['mkdirSync'](_0x1ecf6b, {
                'mode': _0x57abab(_0x981d46),
                'recursive': !![]
            });
        };
    },
    0x4517: (_0x57dca8, _0x4428f8, _0x272a0f) => {
        'use strict';
        const _0x45f84b = _0x272a0f(0x11569);
        _0x57dca8['exports']['checkPath'] = function _0x330071(_0x53b954) {
            if (process['platform'] === 'win32') {
                const _0x29e1c8 = /[<>:"|?*]/['test'](_0x53b954['replace'](_0x45f84b['parse'](_0x53b954)['root'], ''));
                if (_0x29e1c8) {
                    const _0x1ae18c = new Error('Path\x20contains\x20invalid\x20characters:\x20' + _0x53b954);
                    _0x1ae18c['code'] = 'EINVAL';
                    throw _0x1ae18c;
                }
            }
        };
    },
    0x1462d: (_0x492c9f, _0x18ea2a, _0x3fa264) => {
        'use strict';
        const _0x3b2739 = _0x3fa264(0x15f6a)['fromCallback'];
        _0x492c9f['exports'] = {
            'move': _0x3b2739(_0x3fa264(0xda7d)),
            'moveSync': _0x3fa264(0xea9b)
        };
    },
    0xea9b: (_0x246db8, _0x466437, _0x5198ae) => {
        'use strict';
        const _0x33736a = _0x5198ae(0x15d1a), _0x2390ee = _0x5198ae(0x11569), _0x5df08d = _0x5198ae(0xb839)['copySync'], _0x54523d = _0x5198ae(0x983d)['removeSync'], _0x5816bf = _0x5198ae(0x10c1d)['mkdirpSync'], _0x39af41 = _0x5198ae(0x7c9d);
        function _0x5d3cd9(_0x38a9cc, _0x3c4dca, _0x58b51f) {
            _0x58b51f = _0x58b51f || {};
            const _0x39a4e1 = _0x58b51f['overwrite'] || _0x58b51f['clobber'] || ![], {
                    srcStat: _0x52865c,
                    isChangingCase: isChangingCase = ![]
                } = _0x39af41['checkPathsSync'](_0x38a9cc, _0x3c4dca, 'move', _0x58b51f);
            _0x39af41['checkParentPathsSync'](_0x38a9cc, _0x52865c, _0x3c4dca, 'move');
            if (!_0x1cf3f9(_0x3c4dca))
                _0x5816bf(_0x2390ee['dirname'](_0x3c4dca));
            return _0x49046d(_0x38a9cc, _0x3c4dca, _0x39a4e1, isChangingCase);
        }
        function _0x1cf3f9(_0x21a980) {
            const _0x370b84 = _0x2390ee['dirname'](_0x21a980), _0x64963e = _0x2390ee['parse'](_0x370b84);
            return _0x64963e['root'] === _0x370b84;
        }
        function _0x49046d(_0x178697, _0x4e69ee, _0x59a1cb, _0x453915) {
            if (_0x453915)
                return _0x415d1f(_0x178697, _0x4e69ee, _0x59a1cb);
            if (_0x59a1cb)
                return _0x54523d(_0x4e69ee), _0x415d1f(_0x178697, _0x4e69ee, _0x59a1cb);
            if (_0x33736a['existsSync'](_0x4e69ee))
                throw new Error('dest\x20already\x20exists.');
            return _0x415d1f(_0x178697, _0x4e69ee, _0x59a1cb);
        }
        function _0x415d1f(_0x2c103b, _0x346662, _0x182963) {
            try {
                _0x33736a['renameSync'](_0x2c103b, _0x346662);
            } catch (_0x3a4f37) {
                if (_0x3a4f37['code'] !== 'EXDEV')
                    throw _0x3a4f37;
                return _0x151986(_0x2c103b, _0x346662, _0x182963);
            }
        }
        function _0x151986(_0x324e92, _0x54354d, _0x92269) {
            const _0x1b5e60 = {
                'overwrite': _0x92269,
                'errorOnExist': !![]
            };
            return _0x5df08d(_0x324e92, _0x54354d, _0x1b5e60), _0x54523d(_0x324e92);
        }
        _0x246db8['exports'] = _0x5d3cd9;
    },
    0xda7d: (_0xf257a5, _0x2f5a3e, _0x10537a) => {
        'use strict';
        const _0x2a84b3 = _0x10537a(0x15d1a), _0x3db399 = _0x10537a(0x11569), _0x33908b = _0x10537a(0xb839)['copy'], _0x602e30 = _0x10537a(0x983d)['remove'], _0x3e03ad = _0x10537a(0x10c1d)['mkdirp'], _0x3815b6 = _0x10537a(0x15330)['pathExists'], _0x3b450d = _0x10537a(0x7c9d);
        function _0x20f457(_0x4ff72c, _0x557636, _0x3a969a, _0xc82617) {
            typeof _0x3a969a === 'function' && (_0xc82617 = _0x3a969a, _0x3a969a = {});
            _0x3a969a = _0x3a969a || {};
            const _0x4b4e1f = _0x3a969a['overwrite'] || _0x3a969a['clobber'] || ![];
            _0x3b450d['checkPaths'](_0x4ff72c, _0x557636, 'move', _0x3a969a, (_0x410aeb, _0xaaec80) => {
                if (_0x410aeb)
                    return _0xc82617(_0x410aeb);
                const {
                    srcStat: _0x4ed945,
                    isChangingCase: isChangingCase = ![]
                } = _0xaaec80;
                _0x3b450d['checkParentPaths'](_0x4ff72c, _0x4ed945, _0x557636, 'move', _0x3fec3f => {
                    if (_0x3fec3f)
                        return _0xc82617(_0x3fec3f);
                    if (_0x3de61a(_0x557636))
                        return _0x5430f4(_0x4ff72c, _0x557636, _0x4b4e1f, isChangingCase, _0xc82617);
                    _0x3e03ad(_0x3db399['dirname'](_0x557636), _0x211522 => {
                        if (_0x211522)
                            return _0xc82617(_0x211522);
                        return _0x5430f4(_0x4ff72c, _0x557636, _0x4b4e1f, isChangingCase, _0xc82617);
                    });
                });
            });
        }
        function _0x3de61a(_0x5b1fb) {
            const _0x57c75f = _0x3db399['dirname'](_0x5b1fb), _0x4a837e = _0x3db399['parse'](_0x57c75f);
            return _0x4a837e['root'] === _0x57c75f;
        }
        function _0x5430f4(_0x1ceb20, _0x509231, _0x15703b, _0x39f06f, _0x44cf72) {
            if (_0x39f06f)
                return _0x47c365(_0x1ceb20, _0x509231, _0x15703b, _0x44cf72);
            if (_0x15703b)
                return _0x602e30(_0x509231, _0x427900 => {
                    if (_0x427900)
                        return _0x44cf72(_0x427900);
                    return _0x47c365(_0x1ceb20, _0x509231, _0x15703b, _0x44cf72);
                });
            _0x3815b6(_0x509231, (_0x3760e6, _0xa4cd2e) => {
                if (_0x3760e6)
                    return _0x44cf72(_0x3760e6);
                if (_0xa4cd2e)
                    return _0x44cf72(new Error('dest\x20already\x20exists.'));
                return _0x47c365(_0x1ceb20, _0x509231, _0x15703b, _0x44cf72);
            });
        }
        function _0x47c365(_0x5a8383, _0x2df67e, _0x4f79f1, _0x587840) {
            _0x2a84b3['rename'](_0x5a8383, _0x2df67e, _0x503a77 => {
                if (!_0x503a77)
                    return _0x587840();
                if (_0x503a77['code'] !== 'EXDEV')
                    return _0x587840(_0x503a77);
                return _0x568221(_0x5a8383, _0x2df67e, _0x4f79f1, _0x587840);
            });
        }
        function _0x568221(_0x533215, _0x1fb155, _0x4fa459, _0x3be79f) {
            const _0x172f38 = {
                'overwrite': _0x4fa459,
                'errorOnExist': !![]
            };
            _0x33908b(_0x533215, _0x1fb155, _0x172f38, _0x2f2ad7 => {
                if (_0x2f2ad7)
                    return _0x3be79f(_0x2f2ad7);
                return _0x602e30(_0x533215, _0x3be79f);
            });
        }
        _0xf257a5['exports'] = _0x20f457;
    },
    0x7096: (_0x3fa521, _0xb54d32, _0x257d30) => {
        'use strict';
        const _0x276d07 = _0x257d30(0x15f6a)['fromCallback'], _0xc9efce = _0x257d30(0x15d1a), _0x409877 = _0x257d30(0x11569), _0x879dfe = _0x257d30(0x10c1d), _0x2e67ad = _0x257d30(0x15330)['pathExists'];
        function _0x4762d5(_0x2b549a, _0xb1744a, _0x4a7c38, _0x3af1e8) {
            typeof _0x4a7c38 === 'function' && (_0x3af1e8 = _0x4a7c38, _0x4a7c38 = 'utf8');
            const _0x46b64c = _0x409877['dirname'](_0x2b549a);
            _0x2e67ad(_0x46b64c, (_0xacfce9, _0x479ab7) => {
                if (_0xacfce9)
                    return _0x3af1e8(_0xacfce9);
                if (_0x479ab7)
                    return _0xc9efce['writeFile'](_0x2b549a, _0xb1744a, _0x4a7c38, _0x3af1e8);
                _0x879dfe['mkdirs'](_0x46b64c, _0xd39787 => {
                    if (_0xd39787)
                        return _0x3af1e8(_0xd39787);
                    _0xc9efce['writeFile'](_0x2b549a, _0xb1744a, _0x4a7c38, _0x3af1e8);
                });
            });
        }
        function _0x14d59c(_0x5590f3, ..._0xc83e4e) {
            const _0x504ea4 = _0x409877['dirname'](_0x5590f3);
            if (_0xc9efce['existsSync'](_0x504ea4))
                return _0xc9efce['writeFileSync'](_0x5590f3, ..._0xc83e4e);
            _0x879dfe['mkdirsSync'](_0x504ea4), _0xc9efce['writeFileSync'](_0x5590f3, ..._0xc83e4e);
        }
        _0x3fa521['exports'] = {
            'outputFile': _0x276d07(_0x4762d5),
            'outputFileSync': _0x14d59c
        };
    },
    0x15330: (_0x1bded7, _0x41fb13, _0x5b6f38) => {
        'use strict';
        const _0x1b87ed = _0x5b6f38(0x15f6a)['fromPromise'], _0x28a078 = _0x5b6f38(0x15a17);
        function _0x4e3a08(_0x31062e) {
            return _0x28a078['access'](_0x31062e)['then'](() => !![])['catch'](() => ![]);
        }
        _0x1bded7['exports'] = {
            'pathExists': _0x1b87ed(_0x4e3a08),
            'pathExistsSync': _0x28a078['existsSync']
        };
    },
    0x983d: (_0x1e980f, _0x539079, _0x484547) => {
        'use strict';
        const _0x5e9a5f = _0x484547(0x15d1a), _0x4873cb = _0x484547(0x15f6a)['fromCallback'], _0x323eb5 = _0x484547(0x61bd);
        function _0x4f1932(_0x20d72f, _0x269320) {
            if (_0x5e9a5f['rm'])
                return _0x5e9a5f['rm'](_0x20d72f, {
                    'recursive': !![],
                    'force': !![]
                }, _0x269320);
            _0x323eb5(_0x20d72f, _0x269320);
        }
        function _0x4fd8a0(_0x5491c5) {
            if (_0x5e9a5f['rmSync'])
                return _0x5e9a5f['rmSync'](_0x5491c5, {
                    'recursive': !![],
                    'force': !![]
                });
            _0x323eb5['sync'](_0x5491c5);
        }
        _0x1e980f['exports'] = {
            'remove': _0x4873cb(_0x4f1932),
            'removeSync': _0x4fd8a0
        };
    },
    0x61bd: (_0x2dfaf6, _0x3756c9, _0x2dce29) => {
        'use strict';
        const _0x1fa55c = _0x2dce29(0x15d1a), _0x2cf64f = _0x2dce29(0x11569), _0x5138b4 = _0x2dce29(0x9a43), _0x3e2d23 = process['platform'] === 'win32';
        function _0x273b05(_0x386d7a) {
            const _0x4bc617 = [
                'unlink',
                'chmod',
                'stat',
                'lstat',
                'rmdir',
                'readdir'
            ];
            _0x4bc617['forEach'](_0x4b81a6 => {
                _0x386d7a[_0x4b81a6] = _0x386d7a[_0x4b81a6] || _0x1fa55c[_0x4b81a6], _0x4b81a6 = _0x4b81a6 + 'Sync', _0x386d7a[_0x4b81a6] = _0x386d7a[_0x4b81a6] || _0x1fa55c[_0x4b81a6];
            }), _0x386d7a['maxBusyTries'] = _0x386d7a['maxBusyTries'] || 0x3;
        }
        function _0x269af8(_0x16fb13, _0x19e96e, _0x59d6e2) {
            let _0x3febb6 = 0x0;
            typeof _0x19e96e === 'function' && (_0x59d6e2 = _0x19e96e, _0x19e96e = {}), _0x5138b4(_0x16fb13, 'rimraf:\x20missing\x20path'), _0x5138b4['strictEqual'](typeof _0x16fb13, 'string', 'rimraf:\x20path\x20should\x20be\x20a\x20string'), _0x5138b4['strictEqual'](typeof _0x59d6e2, 'function', 'rimraf:\x20callback\x20function\x20required'), _0x5138b4(_0x19e96e, 'rimraf:\x20invalid\x20options\x20argument\x20provided'), _0x5138b4['strictEqual'](typeof _0x19e96e, 'object', 'rimraf:\x20options\x20should\x20be\x20object'), _0x273b05(_0x19e96e), _0x2cac5a(_0x16fb13, _0x19e96e, function _0x2e5deb(_0x57ed51) {
                if (_0x57ed51) {
                    if ((_0x57ed51['code'] === 'EBUSY' || _0x57ed51['code'] === 'ENOTEMPTY' || _0x57ed51['code'] === 'EPERM') && _0x3febb6 < _0x19e96e['maxBusyTries']) {
                        _0x3febb6++;
                        const _0x3ef761 = _0x3febb6 * 0x64;
                        return setTimeout(() => _0x2cac5a(_0x16fb13, _0x19e96e, _0x2e5deb), _0x3ef761);
                    }
                    if (_0x57ed51['code'] === 'ENOENT')
                        _0x57ed51 = null;
                }
                _0x59d6e2(_0x57ed51);
            });
        }
        function _0x2cac5a(_0x1f6c84, _0x50cd07, _0x52daca) {
            _0x5138b4(_0x1f6c84), _0x5138b4(_0x50cd07), _0x5138b4(typeof _0x52daca === 'function'), _0x50cd07['lstat'](_0x1f6c84, (_0x48aa75, _0x3e44de) => {
                if (_0x48aa75 && _0x48aa75['code'] === 'ENOENT')
                    return _0x52daca(null);
                if (_0x48aa75 && _0x48aa75['code'] === 'EPERM' && _0x3e2d23)
                    return _0x2c1fc9(_0x1f6c84, _0x50cd07, _0x48aa75, _0x52daca);
                if (_0x3e44de && _0x3e44de['isDirectory']())
                    return _0xa2c4d4(_0x1f6c84, _0x50cd07, _0x48aa75, _0x52daca);
                _0x50cd07['unlink'](_0x1f6c84, _0x247326 => {
                    if (_0x247326) {
                        if (_0x247326['code'] === 'ENOENT')
                            return _0x52daca(null);
                        if (_0x247326['code'] === 'EPERM')
                            return _0x3e2d23 ? _0x2c1fc9(_0x1f6c84, _0x50cd07, _0x247326, _0x52daca) : _0xa2c4d4(_0x1f6c84, _0x50cd07, _0x247326, _0x52daca);
                        if (_0x247326['code'] === 'EISDIR')
                            return _0xa2c4d4(_0x1f6c84, _0x50cd07, _0x247326, _0x52daca);
                    }
                    return _0x52daca(_0x247326);
                });
            });
        }
        function _0x2c1fc9(_0x1141fa, _0x1d9620, _0x5b85e2, _0x27cf0f) {
            _0x5138b4(_0x1141fa), _0x5138b4(_0x1d9620), _0x5138b4(typeof _0x27cf0f === 'function'), _0x1d9620['chmod'](_0x1141fa, 0x1b6, _0x433122 => {
                _0x433122 ? _0x27cf0f(_0x433122['code'] === 'ENOENT' ? null : _0x5b85e2) : _0x1d9620['stat'](_0x1141fa, (_0x582523, _0x51bda4) => {
                    if (_0x582523)
                        _0x27cf0f(_0x582523['code'] === 'ENOENT' ? null : _0x5b85e2);
                    else
                        _0x51bda4['isDirectory']() ? _0xa2c4d4(_0x1141fa, _0x1d9620, _0x5b85e2, _0x27cf0f) : _0x1d9620['unlink'](_0x1141fa, _0x27cf0f);
                });
            });
        }
        function _0x4f28d9(_0x4fcf30, _0xae434f, _0x1531f8) {
            let _0xebab4c;
            _0x5138b4(_0x4fcf30), _0x5138b4(_0xae434f);
            try {
                _0xae434f['chmodSync'](_0x4fcf30, 0x1b6);
            } catch (_0x291610) {
                if (_0x291610['code'] === 'ENOENT')
                    return;
                else
                    throw _0x1531f8;
            }
            try {
                _0xebab4c = _0xae434f['statSync'](_0x4fcf30);
            } catch (_0x1b9089) {
                if (_0x1b9089['code'] === 'ENOENT')
                    return;
                else
                    throw _0x1531f8;
            }
            _0xebab4c['isDirectory']() ? _0x2dfd25(_0x4fcf30, _0xae434f, _0x1531f8) : _0xae434f['unlinkSync'](_0x4fcf30);
        }
        function _0xa2c4d4(_0x55d9a4, _0x5997da, _0x3d385d, _0x4f41fb) {
            _0x5138b4(_0x55d9a4), _0x5138b4(_0x5997da), _0x5138b4(typeof _0x4f41fb === 'function'), _0x5997da['rmdir'](_0x55d9a4, _0x313dda => {
                if (_0x313dda && (_0x313dda['code'] === 'ENOTEMPTY' || _0x313dda['code'] === 'EEXIST' || _0x313dda['code'] === 'EPERM'))
                    _0xfbaeb8(_0x55d9a4, _0x5997da, _0x4f41fb);
                else
                    _0x313dda && _0x313dda['code'] === 'ENOTDIR' ? _0x4f41fb(_0x3d385d) : _0x4f41fb(_0x313dda);
            });
        }
        function _0xfbaeb8(_0x3251ee, _0x12eb95, _0x641b91) {
            _0x5138b4(_0x3251ee), _0x5138b4(_0x12eb95), _0x5138b4(typeof _0x641b91 === 'function'), _0x12eb95['readdir'](_0x3251ee, (_0x50c699, _0x2173cf) => {
                if (_0x50c699)
                    return _0x641b91(_0x50c699);
                let _0xc073af = _0x2173cf['length'], _0x56f310;
                if (_0xc073af === 0x0)
                    return _0x12eb95['rmdir'](_0x3251ee, _0x641b91);
                _0x2173cf['forEach'](_0x1066be => {
                    _0x269af8(_0x2cf64f['join'](_0x3251ee, _0x1066be), _0x12eb95, _0x3df61f => {
                        if (_0x56f310)
                            return;
                        if (_0x3df61f)
                            return _0x641b91(_0x56f310 = _0x3df61f);
                        --_0xc073af === 0x0 && _0x12eb95['rmdir'](_0x3251ee, _0x641b91);
                    });
                });
            });
        }
        function _0x109996(_0x2278a5, _0x825760) {
            let _0x1efa35;
            _0x825760 = _0x825760 || {}, _0x273b05(_0x825760), _0x5138b4(_0x2278a5, 'rimraf:\x20missing\x20path'), _0x5138b4['strictEqual'](typeof _0x2278a5, 'string', 'rimraf:\x20path\x20should\x20be\x20a\x20string'), _0x5138b4(_0x825760, 'rimraf:\x20missing\x20options'), _0x5138b4['strictEqual'](typeof _0x825760, 'object', 'rimraf:\x20options\x20should\x20be\x20object');
            try {
                _0x1efa35 = _0x825760['lstatSync'](_0x2278a5);
            } catch (_0x4bbe6f) {
                if (_0x4bbe6f['code'] === 'ENOENT')
                    return;
                _0x4bbe6f['code'] === 'EPERM' && _0x3e2d23 && _0x4f28d9(_0x2278a5, _0x825760, _0x4bbe6f);
            }
            try {
                _0x1efa35 && _0x1efa35['isDirectory']() ? _0x2dfd25(_0x2278a5, _0x825760, null) : _0x825760['unlinkSync'](_0x2278a5);
            } catch (_0x1a4912) {
                if (_0x1a4912['code'] === 'ENOENT')
                    return;
                else {
                    if (_0x1a4912['code'] === 'EPERM')
                        return _0x3e2d23 ? _0x4f28d9(_0x2278a5, _0x825760, _0x1a4912) : _0x2dfd25(_0x2278a5, _0x825760, _0x1a4912);
                    else {
                        if (_0x1a4912['code'] !== 'EISDIR')
                            throw _0x1a4912;
                    }
                }
                _0x2dfd25(_0x2278a5, _0x825760, _0x1a4912);
            }
        }
        function _0x2dfd25(_0x6c83ee, _0x5183bd, _0x47c76f) {
            _0x5138b4(_0x6c83ee), _0x5138b4(_0x5183bd);
            try {
                _0x5183bd['rmdirSync'](_0x6c83ee);
            } catch (_0x2b002f) {
                if (_0x2b002f['code'] === 'ENOTDIR')
                    throw _0x47c76f;
                else {
                    if (_0x2b002f['code'] === 'ENOTEMPTY' || _0x2b002f['code'] === 'EEXIST' || _0x2b002f['code'] === 'EPERM')
                        _0xc81598(_0x6c83ee, _0x5183bd);
                    else {
                        if (_0x2b002f['code'] !== 'ENOENT')
                            throw _0x2b002f;
                    }
                }
            }
        }
        function _0xc81598(_0x48e98f, _0x572f6b) {
            _0x5138b4(_0x48e98f), _0x5138b4(_0x572f6b), _0x572f6b['readdirSync'](_0x48e98f)['forEach'](_0x4c4656 => _0x109996(_0x2cf64f['join'](_0x48e98f, _0x4c4656), _0x572f6b));
            if (_0x3e2d23) {
                const _0x35a1dc = Date['now']();
                do {
                    try {
                        const _0x151ad1 = _0x572f6b['rmdirSync'](_0x48e98f, _0x572f6b);
                        return _0x151ad1;
                    } catch {
                    }
                } while (Date['now']() - _0x35a1dc < 0x1f4);
            } else {
                const _0x326fa8 = _0x572f6b['rmdirSync'](_0x48e98f, _0x572f6b);
                return _0x326fa8;
            }
        }
        _0x2dfaf6['exports'] = _0x269af8, _0x269af8['sync'] = _0x109996;
    },
    0x7c9d: (_0x1223a6, _0x5c6231, _0x709f85) => {
        'use strict';
        const _0x424dff = _0x709f85(0x15a17), _0xd37a0e = _0x709f85(0x11569), _0x1c5da0 = _0x709f85(0x1206d);
        function _0x2d6791(_0x5a00fb, _0x27ec18, _0x30d55d) {
            const _0x1973a5 = _0x30d55d['dereference'] ? _0x394fbd => _0x424dff['stat'](_0x394fbd, { 'bigint': !![] }) : _0x7cb311 => _0x424dff['lstat'](_0x7cb311, { 'bigint': !![] });
            return Promise['all']([
                _0x1973a5(_0x5a00fb),
                _0x1973a5(_0x27ec18)['catch'](_0x3cf72f => {
                    if (_0x3cf72f['code'] === 'ENOENT')
                        return null;
                    throw _0x3cf72f;
                })
            ])['then'](([_0xe75c39, _0x3e3c70]) => ({
                'srcStat': _0xe75c39,
                'destStat': _0x3e3c70
            }));
        }
        function _0x1c2993(_0x6d10d8, _0x276b64, _0x4b3b61) {
            let _0x283eac;
            const _0x24b0bf = _0x4b3b61['dereference'] ? _0x2e5416 => _0x424dff['statSync'](_0x2e5416, { 'bigint': !![] }) : _0x34b86b => _0x424dff['lstatSync'](_0x34b86b, { 'bigint': !![] }), _0x2f2fd3 = _0x24b0bf(_0x6d10d8);
            try {
                _0x283eac = _0x24b0bf(_0x276b64);
            } catch (_0xba9636) {
                if (_0xba9636['code'] === 'ENOENT')
                    return {
                        'srcStat': _0x2f2fd3,
                        'destStat': null
                    };
                throw _0xba9636;
            }
            return {
                'srcStat': _0x2f2fd3,
                'destStat': _0x283eac
            };
        }
        function _0x70bdf5(_0x2aed45, _0x2ced3a, _0x32654b, _0x56462b, _0x4ec41f) {
            _0x1c5da0['callbackify'](_0x2d6791)(_0x2aed45, _0x2ced3a, _0x56462b, (_0x270d3f, _0x51551f) => {
                if (_0x270d3f)
                    return _0x4ec41f(_0x270d3f);
                const {
                    srcStat: _0x28bcc8,
                    destStat: _0x557f80
                } = _0x51551f;
                if (_0x557f80) {
                    if (_0x485dc3(_0x28bcc8, _0x557f80)) {
                        const _0x4c94ce = _0xd37a0e['basename'](_0x2aed45), _0xa882a5 = _0xd37a0e['basename'](_0x2ced3a);
                        if (_0x32654b === 'move' && _0x4c94ce !== _0xa882a5 && _0x4c94ce['toLowerCase']() === _0xa882a5['toLowerCase']())
                            return _0x4ec41f(null, {
                                'srcStat': _0x28bcc8,
                                'destStat': _0x557f80,
                                'isChangingCase': !![]
                            });
                        return _0x4ec41f(new Error('Source\x20and\x20destination\x20must\x20not\x20be\x20the\x20same.'));
                    }
                    if (_0x28bcc8['isDirectory']() && !_0x557f80['isDirectory']())
                        return _0x4ec41f(new Error('Cannot\x20overwrite\x20non-directory\x20\x27' + _0x2ced3a + '\x27\x20with\x20directory\x20\x27' + _0x2aed45 + '\x27.'));
                    if (!_0x28bcc8['isDirectory']() && _0x557f80['isDirectory']())
                        return _0x4ec41f(new Error('Cannot\x20overwrite\x20directory\x20\x27' + _0x2ced3a + '\x27\x20with\x20non-directory\x20\x27' + _0x2aed45 + '\x27.'));
                }
                if (_0x28bcc8['isDirectory']() && _0x1bce5f(_0x2aed45, _0x2ced3a))
                    return _0x4ec41f(new Error(_0x2683bd(_0x2aed45, _0x2ced3a, _0x32654b)));
                return _0x4ec41f(null, {
                    'srcStat': _0x28bcc8,
                    'destStat': _0x557f80
                });
            });
        }
        function _0x3b98d0(_0x5a4e62, _0x30546b, _0x365050, _0x3ee35b) {
            const {
                srcStat: _0x862e8f,
                destStat: _0x548a70
            } = _0x1c2993(_0x5a4e62, _0x30546b, _0x3ee35b);
            if (_0x548a70) {
                if (_0x485dc3(_0x862e8f, _0x548a70)) {
                    const _0x32e5c0 = _0xd37a0e['basename'](_0x5a4e62), _0x319657 = _0xd37a0e['basename'](_0x30546b);
                    if (_0x365050 === 'move' && _0x32e5c0 !== _0x319657 && _0x32e5c0['toLowerCase']() === _0x319657['toLowerCase']())
                        return {
                            'srcStat': _0x862e8f,
                            'destStat': _0x548a70,
                            'isChangingCase': !![]
                        };
                    throw new Error('Source\x20and\x20destination\x20must\x20not\x20be\x20the\x20same.');
                }
                if (_0x862e8f['isDirectory']() && !_0x548a70['isDirectory']())
                    throw new Error('Cannot\x20overwrite\x20non-directory\x20\x27' + _0x30546b + '\x27\x20with\x20directory\x20\x27' + _0x5a4e62 + '\x27.');
                if (!_0x862e8f['isDirectory']() && _0x548a70['isDirectory']())
                    throw new Error('Cannot\x20overwrite\x20directory\x20\x27' + _0x30546b + '\x27\x20with\x20non-directory\x20\x27' + _0x5a4e62 + '\x27.');
            }
            if (_0x862e8f['isDirectory']() && _0x1bce5f(_0x5a4e62, _0x30546b))
                throw new Error(_0x2683bd(_0x5a4e62, _0x30546b, _0x365050));
            return {
                'srcStat': _0x862e8f,
                'destStat': _0x548a70
            };
        }
        function _0x46c587(_0x2b02ec, _0x41abbd, _0x1554b9, _0x148d00, _0x283542) {
            const _0x1efd2b = _0xd37a0e['resolve'](_0xd37a0e['dirname'](_0x2b02ec)), _0x1eed39 = _0xd37a0e['resolve'](_0xd37a0e['dirname'](_0x1554b9));
            if (_0x1eed39 === _0x1efd2b || _0x1eed39 === _0xd37a0e['parse'](_0x1eed39)['root'])
                return _0x283542();
            _0x424dff['stat'](_0x1eed39, { 'bigint': !![] }, (_0x5a876c, _0x18b6a3) => {
                if (_0x5a876c) {
                    if (_0x5a876c['code'] === 'ENOENT')
                        return _0x283542();
                    return _0x283542(_0x5a876c);
                }
                if (_0x485dc3(_0x41abbd, _0x18b6a3))
                    return _0x283542(new Error(_0x2683bd(_0x2b02ec, _0x1554b9, _0x148d00)));
                return _0x46c587(_0x2b02ec, _0x41abbd, _0x1eed39, _0x148d00, _0x283542);
            });
        }
        function _0x1b8c84(_0x41c347, _0x4ee10c, _0x25335f, _0x4a1595) {
            const _0x2be169 = _0xd37a0e['resolve'](_0xd37a0e['dirname'](_0x41c347)), _0x433e2e = _0xd37a0e['resolve'](_0xd37a0e['dirname'](_0x25335f));
            if (_0x433e2e === _0x2be169 || _0x433e2e === _0xd37a0e['parse'](_0x433e2e)['root'])
                return;
            let _0x13a2ef;
            try {
                _0x13a2ef = _0x424dff['statSync'](_0x433e2e, { 'bigint': !![] });
            } catch (_0x38ba37) {
                if (_0x38ba37['code'] === 'ENOENT')
                    return;
                throw _0x38ba37;
            }
            if (_0x485dc3(_0x4ee10c, _0x13a2ef))
                throw new Error(_0x2683bd(_0x41c347, _0x25335f, _0x4a1595));
            return _0x1b8c84(_0x41c347, _0x4ee10c, _0x433e2e, _0x4a1595);
        }
        function _0x485dc3(_0xb354f3, _0x24e1d2) {
            return _0x24e1d2['ino'] && _0x24e1d2['dev'] && _0x24e1d2['ino'] === _0xb354f3['ino'] && _0x24e1d2['dev'] === _0xb354f3['dev'];
        }
        function _0x1bce5f(_0x14dc0a, _0x1b568e) {
            const _0x2a4fc0 = _0xd37a0e['resolve'](_0x14dc0a)['split'](_0xd37a0e['sep'])['filter'](_0x45246f => _0x45246f), _0x32e577 = _0xd37a0e['resolve'](_0x1b568e)['split'](_0xd37a0e['sep'])['filter'](_0x347d41 => _0x347d41);
            return _0x2a4fc0['reduce']((_0x19202d, _0x450c46, _0x2baf4f) => _0x19202d && _0x32e577[_0x2baf4f] === _0x450c46, !![]);
        }
        function _0x2683bd(_0x575217, _0x33fa2c, _0x507806) {
            return 'Cannot\x20' + _0x507806 + '\x20\x27' + _0x575217 + '\x27\x20to\x20a\x20subdirectory\x20of\x20itself,\x20\x27' + _0x33fa2c + '\x27.';
        }
        _0x1223a6['exports'] = {
            'checkPaths': _0x70bdf5,
            'checkPathsSync': _0x3b98d0,
            'checkParentPaths': _0x46c587,
            'checkParentPathsSync': _0x1b8c84,
            'isSrcSubdir': _0x1bce5f,
            'areIdentical': _0x485dc3
        };
    },
    0x13743: (_0x3178ba, _0x1f3be6, _0x1c4db2) => {
        'use strict';
        const _0x1de73a = _0x1c4db2(0x15d1a);
        function _0x16ef15(_0x497c31, _0x1ff959, _0x2168a7, _0x5e351b) {
            _0x1de73a['open'](_0x497c31, 'r+', (_0x1216e9, _0x7b184e) => {
                if (_0x1216e9)
                    return _0x5e351b(_0x1216e9);
                _0x1de73a['futimes'](_0x7b184e, _0x1ff959, _0x2168a7, _0x28baf5 => {
                    _0x1de73a['close'](_0x7b184e, _0x4dd53d => {
                        if (_0x5e351b)
                            _0x5e351b(_0x28baf5 || _0x4dd53d);
                    });
                });
            });
        }
        function _0x5a74d0(_0xfdd18, _0x46babb, _0x587b97) {
            const _0x357d55 = _0x1de73a['openSync'](_0xfdd18, 'r+');
            return _0x1de73a['futimesSync'](_0x357d55, _0x46babb, _0x587b97), _0x1de73a['closeSync'](_0x357d55);
        }
        _0x3178ba['exports'] = {
            'utimesMillis': _0x16ef15,
            'utimesMillisSync': _0x5a74d0
        };
    },
    0x1217f: _0xb77171 => {
        'use strict';
        _0xb77171['exports'] = _0x8ea8f0;
        var _0x198ac1 = Object['getPrototypeOf'] || function (_0x4adf81) {
            return _0x4adf81['__proto__'];
        };
        function _0x8ea8f0(_0x150ab8) {
            if (_0x150ab8 === null || typeof _0x150ab8 !== 'object')
                return _0x150ab8;
            if (_0x150ab8 instanceof Object)
                var _0x3e6700 = { '__proto__': _0x198ac1(_0x150ab8) };
            else
                var _0x3e6700 = Object['create'](null);
            return Object['getOwnPropertyNames'](_0x150ab8)['forEach'](function (_0xb752c) {
                Object['defineProperty'](_0x3e6700, _0xb752c, Object['getOwnPropertyDescriptor'](_0x150ab8, _0xb752c));
            }), _0x3e6700;
        }
    },
    0x15d1a: (_0x2bd4c5, _0x1a33be, _0x489063) => {
        var _0x17e4b0 = _0x489063(0xdf3b), _0x89f8d1 = _0x489063(0x15c1b), _0x5c4c95 = _0x489063(0x1458c), _0x361638 = _0x489063(0x1217f), _0x49b314 = _0x489063(0x1206d), _0x227935, _0x2a4bb5;
        typeof Symbol === 'function' && typeof Symbol['for'] === 'function' ? (_0x227935 = Symbol['for']('graceful-fs.queue'), _0x2a4bb5 = Symbol['for']('graceful-fs.previous')) : (_0x227935 = '___graceful-fs.queue', _0x2a4bb5 = '___graceful-fs.previous');
        function _0x385298() {
        }
        function _0x16a2ef(_0x2403f1, _0x5af10b) {
            Object['defineProperty'](_0x2403f1, _0x227935, {
                'get': function () {
                    return _0x5af10b;
                }
            });
        }
        var _0x5b1363 = _0x385298;
        if (_0x49b314['debuglog'])
            _0x5b1363 = _0x49b314['debuglog']('gfs4');
        else {
            if (/\bgfs4\b/i['test'](process['env']['NODE_DEBUG'] || ''))
                _0x5b1363 = function () {
                    var _0x45d8d4 = _0x49b314['format']['apply'](_0x49b314, arguments);
                    _0x45d8d4 = 'GFS4:\x20' + _0x45d8d4['split'](/\n/)['join']('\x0aGFS4:\x20'), console['error'](_0x45d8d4);
                };
        }
        if (!_0x17e4b0[_0x227935]) {
            var _0x4bc237 = global[_0x227935] || [];
            _0x16a2ef(_0x17e4b0, _0x4bc237), _0x17e4b0['close'] = function (_0x44670c) {
                function _0x238d1a(_0x4e096a, _0x16b349) {
                    return _0x44670c['call'](_0x17e4b0, _0x4e096a, function (_0x3d182e) {
                        !_0x3d182e && _0x12d5f3();
                        if (typeof _0x16b349 === 'function')
                            _0x16b349['apply'](this, arguments);
                    });
                }
                return Object['defineProperty'](_0x238d1a, _0x2a4bb5, { 'value': _0x44670c }), _0x238d1a;
            }(_0x17e4b0['close']), _0x17e4b0['closeSync'] = function (_0x6bc4e) {
                function _0x296ea0(_0x22329a) {
                    _0x6bc4e['apply'](_0x17e4b0, arguments), _0x12d5f3();
                }
                return Object['defineProperty'](_0x296ea0, _0x2a4bb5, { 'value': _0x6bc4e }), _0x296ea0;
            }(_0x17e4b0['closeSync']), /\bgfs4\b/i['test'](process['env']['NODE_DEBUG'] || '') && process['on']('exit', function () {
                _0x5b1363(_0x17e4b0[_0x227935]), _0x489063(0x9a43)['equal'](_0x17e4b0[_0x227935]['length'], 0x0);
            });
        }
        !global[_0x227935] && _0x16a2ef(global, _0x17e4b0[_0x227935]);
        _0x2bd4c5['exports'] = _0x53b0a5(_0x361638(_0x17e4b0));
        process['env']['TEST_GRACEFUL_FS_GLOBAL_PATCH'] && !_0x17e4b0['__patched'] && (_0x2bd4c5['exports'] = _0x53b0a5(_0x17e4b0), _0x17e4b0['__patched'] = !![]);
        function _0x53b0a5(_0x3c649c) {
            _0x89f8d1(_0x3c649c), _0x3c649c['gracefulify'] = _0x53b0a5, _0x3c649c['createReadStream'] = _0x354947, _0x3c649c['createWriteStream'] = _0x4e84f5;
            var _0x2348c3 = _0x3c649c['readFile'];
            _0x3c649c['readFile'] = _0x57b8b6;
            function _0x57b8b6(_0x3dda65, _0x4156b2, _0x99e5c7) {
                if (typeof _0x4156b2 === 'function')
                    _0x99e5c7 = _0x4156b2, _0x4156b2 = null;
                return _0x504e05(_0x3dda65, _0x4156b2, _0x99e5c7);
                function _0x504e05(_0x229aa4, _0x2a69b1, _0x5efa43, _0x52fdef) {
                    return _0x2348c3(_0x229aa4, _0x2a69b1, function (_0x20b4f8) {
                        if (_0x20b4f8 && (_0x20b4f8['code'] === 'EMFILE' || _0x20b4f8['code'] === 'ENFILE'))
                            _0x4a4f08([
                                _0x504e05,
                                [
                                    _0x229aa4,
                                    _0x2a69b1,
                                    _0x5efa43
                                ],
                                _0x20b4f8,
                                _0x52fdef || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x5efa43 === 'function')
                                _0x5efa43['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0x1af0c2 = _0x3c649c['writeFile'];
            _0x3c649c['writeFile'] = _0x4f630c;
            function _0x4f630c(_0x1513e3, _0x477fdc, _0x5208f1, _0x86b303) {
                if (typeof _0x5208f1 === 'function')
                    _0x86b303 = _0x5208f1, _0x5208f1 = null;
                return _0x3a5833(_0x1513e3, _0x477fdc, _0x5208f1, _0x86b303);
                function _0x3a5833(_0x3f9932, _0x176452, _0x66483c, _0x1d7770, _0x1a75b5) {
                    return _0x1af0c2(_0x3f9932, _0x176452, _0x66483c, function (_0x47c5f3) {
                        if (_0x47c5f3 && (_0x47c5f3['code'] === 'EMFILE' || _0x47c5f3['code'] === 'ENFILE'))
                            _0x4a4f08([
                                _0x3a5833,
                                [
                                    _0x3f9932,
                                    _0x176452,
                                    _0x66483c,
                                    _0x1d7770
                                ],
                                _0x47c5f3,
                                _0x1a75b5 || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x1d7770 === 'function')
                                _0x1d7770['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0x3973f3 = _0x3c649c['appendFile'];
            if (_0x3973f3)
                _0x3c649c['appendFile'] = _0x3192e8;
            function _0x3192e8(_0x39e299, _0x58b33e, _0x135ded, _0x2e7399) {
                if (typeof _0x135ded === 'function')
                    _0x2e7399 = _0x135ded, _0x135ded = null;
                return _0x2cdb34(_0x39e299, _0x58b33e, _0x135ded, _0x2e7399);
                function _0x2cdb34(_0x5e9c6f, _0x507a80, _0x48192b, _0x3dea8f, _0x136766) {
                    return _0x3973f3(_0x5e9c6f, _0x507a80, _0x48192b, function (_0x390e6c) {
                        if (_0x390e6c && (_0x390e6c['code'] === 'EMFILE' || _0x390e6c['code'] === 'ENFILE'))
                            _0x4a4f08([
                                _0x2cdb34,
                                [
                                    _0x5e9c6f,
                                    _0x507a80,
                                    _0x48192b,
                                    _0x3dea8f
                                ],
                                _0x390e6c,
                                _0x136766 || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x3dea8f === 'function')
                                _0x3dea8f['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0x28d028 = _0x3c649c['copyFile'];
            if (_0x28d028)
                _0x3c649c['copyFile'] = _0x3c89ec;
            function _0x3c89ec(_0x2d1501, _0x59097c, _0x1c1a23, _0x1c775f) {
                typeof _0x1c1a23 === 'function' && (_0x1c775f = _0x1c1a23, _0x1c1a23 = 0x0);
                return _0x40288e(_0x2d1501, _0x59097c, _0x1c1a23, _0x1c775f);
                function _0x40288e(_0x52af6e, _0x47913f, _0x4e7390, _0x2a00aa, _0x2b689d) {
                    return _0x28d028(_0x52af6e, _0x47913f, _0x4e7390, function (_0x387481) {
                        if (_0x387481 && (_0x387481['code'] === 'EMFILE' || _0x387481['code'] === 'ENFILE'))
                            _0x4a4f08([
                                _0x40288e,
                                [
                                    _0x52af6e,
                                    _0x47913f,
                                    _0x4e7390,
                                    _0x2a00aa
                                ],
                                _0x387481,
                                _0x2b689d || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x2a00aa === 'function')
                                _0x2a00aa['apply'](this, arguments);
                        }
                    });
                }
            }
            var _0x3bd633 = _0x3c649c['readdir'];
            _0x3c649c['readdir'] = _0x546bae;
            var _0x4b0e20 = /^v[0-5]\./;
            function _0x546bae(_0x4ee1e9, _0x4756c5, _0x557ab6) {
                if (typeof _0x4756c5 === 'function')
                    _0x557ab6 = _0x4756c5, _0x4756c5 = null;
                var _0x3828cc = _0x4b0e20['test'](process['version']) ? function _0x529c3a(_0x4f31d7, _0x4b9561, _0x45e9ab, _0x45cf2e) {
                    return _0x3bd633(_0x4f31d7, _0x2c3f3e(_0x4f31d7, _0x4b9561, _0x45e9ab, _0x45cf2e));
                } : function _0xc47164(_0x4f61a9, _0x3b017b, _0x55df36, _0x2de60c) {
                    return _0x3bd633(_0x4f61a9, _0x3b017b, _0x2c3f3e(_0x4f61a9, _0x3b017b, _0x55df36, _0x2de60c));
                };
                return _0x3828cc(_0x4ee1e9, _0x4756c5, _0x557ab6);
                function _0x2c3f3e(_0x1fa8f7, _0x317216, _0xc93e77, _0x2eed82) {
                    return function (_0x45d1e0, _0x5a4d8b) {
                        if (_0x45d1e0 && (_0x45d1e0['code'] === 'EMFILE' || _0x45d1e0['code'] === 'ENFILE'))
                            _0x4a4f08([
                                _0x3828cc,
                                [
                                    _0x1fa8f7,
                                    _0x317216,
                                    _0xc93e77
                                ],
                                _0x45d1e0,
                                _0x2eed82 || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (_0x5a4d8b && _0x5a4d8b['sort'])
                                _0x5a4d8b['sort']();
                            if (typeof _0xc93e77 === 'function')
                                _0xc93e77['call'](this, _0x45d1e0, _0x5a4d8b);
                        }
                    };
                }
            }
            if (process['version']['substr'](0x0, 0x4) === 'v0.8') {
                var _0xc07c10 = _0x5c4c95(_0x3c649c);
                _0x391393 = _0xc07c10['ReadStream'], _0x41a019 = _0xc07c10['WriteStream'];
            }
            var _0x19ccc9 = _0x3c649c['ReadStream'];
            _0x19ccc9 && (_0x391393['prototype'] = Object['create'](_0x19ccc9['prototype']), _0x391393['prototype']['open'] = _0x530e89);
            var _0x43d3cb = _0x3c649c['WriteStream'];
            _0x43d3cb && (_0x41a019['prototype'] = Object['create'](_0x43d3cb['prototype']), _0x41a019['prototype']['open'] = _0x395e6a);
            Object['defineProperty'](_0x3c649c, 'ReadStream', {
                'get': function () {
                    return _0x391393;
                },
                'set': function (_0xb5375b) {
                    _0x391393 = _0xb5375b;
                },
                'enumerable': !![],
                'configurable': !![]
            }), Object['defineProperty'](_0x3c649c, 'WriteStream', {
                'get': function () {
                    return _0x41a019;
                },
                'set': function (_0x2a0287) {
                    _0x41a019 = _0x2a0287;
                },
                'enumerable': !![],
                'configurable': !![]
            });
            var _0x5b7926 = _0x391393;
            Object['defineProperty'](_0x3c649c, 'FileReadStream', {
                'get': function () {
                    return _0x5b7926;
                },
                'set': function (_0x317c5f) {
                    _0x5b7926 = _0x317c5f;
                },
                'enumerable': !![],
                'configurable': !![]
            });
            var _0x5d757f = _0x41a019;
            Object['defineProperty'](_0x3c649c, 'FileWriteStream', {
                'get': function () {
                    return _0x5d757f;
                },
                'set': function (_0x40f2d4) {
                    _0x5d757f = _0x40f2d4;
                },
                'enumerable': !![],
                'configurable': !![]
            });
            function _0x391393(_0x460f6d, _0x3fbe37) {
                if (this instanceof _0x391393)
                    return _0x19ccc9['apply'](this, arguments), this;
                else
                    return _0x391393['apply'](Object['create'](_0x391393['prototype']), arguments);
            }
            function _0x530e89() {
                var _0x1866a8 = this;
                _0x5e233b(_0x1866a8['path'], _0x1866a8['flags'], _0x1866a8['mode'], function (_0x45b230, _0x2a5c81) {
                    if (_0x45b230) {
                        if (_0x1866a8['autoClose'])
                            _0x1866a8['destroy']();
                        _0x1866a8['emit']('error', _0x45b230);
                    } else
                        _0x1866a8['fd'] = _0x2a5c81, _0x1866a8['emit']('open', _0x2a5c81), _0x1866a8['read']();
                });
            }
            function _0x41a019(_0x4904b9, _0x109dff) {
                if (this instanceof _0x41a019)
                    return _0x43d3cb['apply'](this, arguments), this;
                else
                    return _0x41a019['apply'](Object['create'](_0x41a019['prototype']), arguments);
            }
            function _0x395e6a() {
                var _0x2a0fd3 = this;
                _0x5e233b(_0x2a0fd3['path'], _0x2a0fd3['flags'], _0x2a0fd3['mode'], function (_0x4763db, _0x21b016) {
                    _0x4763db ? (_0x2a0fd3['destroy'](), _0x2a0fd3['emit']('error', _0x4763db)) : (_0x2a0fd3['fd'] = _0x21b016, _0x2a0fd3['emit']('open', _0x21b016));
                });
            }
            function _0x354947(_0x1a08bd, _0x4c4250) {
                return new _0x3c649c['ReadStream'](_0x1a08bd, _0x4c4250);
            }
            function _0x4e84f5(_0x17a4ca, _0x4bd617) {
                return new _0x3c649c['WriteStream'](_0x17a4ca, _0x4bd617);
            }
            var _0x59751e = _0x3c649c['open'];
            _0x3c649c['open'] = _0x5e233b;
            function _0x5e233b(_0x36a666, _0x535612, _0x33873d, _0x4562ce) {
                if (typeof _0x33873d === 'function')
                    _0x4562ce = _0x33873d, _0x33873d = null;
                return _0x418002(_0x36a666, _0x535612, _0x33873d, _0x4562ce);
                function _0x418002(_0x53eaf2, _0x243534, _0x1f7ef6, _0x45094c, _0x7942ab) {
                    return _0x59751e(_0x53eaf2, _0x243534, _0x1f7ef6, function (_0xef0f23, _0x3751e3) {
                        if (_0xef0f23 && (_0xef0f23['code'] === 'EMFILE' || _0xef0f23['code'] === 'ENFILE'))
                            _0x4a4f08([
                                _0x418002,
                                [
                                    _0x53eaf2,
                                    _0x243534,
                                    _0x1f7ef6,
                                    _0x45094c
                                ],
                                _0xef0f23,
                                _0x7942ab || Date['now'](),
                                Date['now']()
                            ]);
                        else {
                            if (typeof _0x45094c === 'function')
                                _0x45094c['apply'](this, arguments);
                        }
                    });
                }
            }
            return _0x3c649c;
        }
        function _0x4a4f08(_0x558b32) {
            _0x5b1363('ENQUEUE', _0x558b32[0x0]['name'], _0x558b32[0x1]), _0x17e4b0[_0x227935]['push'](_0x558b32), _0x1a729f();
        }
        var _0xed2395;
        function _0x12d5f3() {
            var _0x195503 = Date['now']();
            for (var _0x3d36eb = 0x0; _0x3d36eb < _0x17e4b0[_0x227935]['length']; ++_0x3d36eb) {
                _0x17e4b0[_0x227935][_0x3d36eb]['length'] > 0x2 && (_0x17e4b0[_0x227935][_0x3d36eb][0x3] = _0x195503, _0x17e4b0[_0x227935][_0x3d36eb][0x4] = _0x195503);
            }
            _0x1a729f();
        }
        function _0x1a729f() {
            clearTimeout(_0xed2395), _0xed2395 = undefined;
            if (_0x17e4b0[_0x227935]['length'] === 0x0)
                return;
            var _0x1e5149 = _0x17e4b0[_0x227935]['shift'](), _0x5b0fd7 = _0x1e5149[0x0], _0x4febe3 = _0x1e5149[0x1], _0xc6d923 = _0x1e5149[0x2], _0xfc0593 = _0x1e5149[0x3], _0x5632fd = _0x1e5149[0x4];
            if (_0xfc0593 === undefined)
                _0x5b1363('RETRY', _0x5b0fd7['name'], _0x4febe3), _0x5b0fd7['apply'](null, _0x4febe3);
            else {
                if (Date['now']() - _0xfc0593 >= 0xea60) {
                    _0x5b1363('TIMEOUT', _0x5b0fd7['name'], _0x4febe3);
                    var _0x2d5d65 = _0x4febe3['pop']();
                    if (typeof _0x2d5d65 === 'function')
                        _0x2d5d65['call'](null, _0xc6d923);
                } else {
                    var _0x629bc1 = Date['now']() - _0x5632fd, _0x389518 = Math['max'](_0x5632fd - _0xfc0593, 0x1), _0x496f5a = Math['min'](_0x389518 * 1.2, 0x64);
                    _0x629bc1 >= _0x496f5a ? (_0x5b1363('RETRY', _0x5b0fd7['name'], _0x4febe3), _0x5b0fd7['apply'](null, _0x4febe3['concat']([_0xfc0593]))) : _0x17e4b0[_0x227935]['push'](_0x1e5149);
                }
            }
            _0xed2395 === undefined && (_0xed2395 = setTimeout(_0x1a729f, 0x0));
        }
    },
    0x1458c: (_0x3477f2, _0x1e3464, _0x40166d) => {
        var _0xb07161 = _0x40166d(0x31ed)['Stream'];
        _0x3477f2['exports'] = _0x296d59;
        function _0x296d59(_0x1c8728) {
            return {
                'ReadStream': _0x37bf6f,
                'WriteStream': _0x4a70ff
            };
            function _0x37bf6f(_0x57fbc5, _0x326423) {
                if (!(this instanceof _0x37bf6f))
                    return new _0x37bf6f(_0x57fbc5, _0x326423);
                _0xb07161['call'](this);
                var _0x3e8c81 = this;
                this['path'] = _0x57fbc5, this['fd'] = null, this['readable'] = !![], this['paused'] = ![], this['flags'] = 'r', this['mode'] = 0x1b6, this['bufferSize'] = 0x40 * 0x400, _0x326423 = _0x326423 || {};
                var _0x4e22fa = Object['keys'](_0x326423);
                for (var _0x504ddb = 0x0, _0x46b25c = _0x4e22fa['length']; _0x504ddb < _0x46b25c; _0x504ddb++) {
                    var _0x387c0d = _0x4e22fa[_0x504ddb];
                    this[_0x387c0d] = _0x326423[_0x387c0d];
                }
                if (this['encoding'])
                    this['setEncoding'](this['encoding']);
                if (this['start'] !== undefined) {
                    if ('number' !== typeof this['start'])
                        throw TypeError('start\x20must\x20be\x20a\x20Number');
                    if (this['end'] === undefined)
                        this['end'] = Infinity;
                    else {
                        if ('number' !== typeof this['end'])
                            throw TypeError('end\x20must\x20be\x20a\x20Number');
                    }
                    if (this['start'] > this['end'])
                        throw new Error('start\x20must\x20be\x20<=\x20end');
                    this['pos'] = this['start'];
                }
                if (this['fd'] !== null) {
                    process['nextTick'](function () {
                        _0x3e8c81['_read']();
                    });
                    return;
                }
                _0x1c8728['open'](this['path'], this['flags'], this['mode'], function (_0x118a2f, _0x27105f) {
                    if (_0x118a2f) {
                        _0x3e8c81['emit']('error', _0x118a2f), _0x3e8c81['readable'] = ![];
                        return;
                    }
                    _0x3e8c81['fd'] = _0x27105f, _0x3e8c81['emit']('open', _0x27105f), _0x3e8c81['_read']();
                });
            }
            function _0x4a70ff(_0x1d6205, _0x1eb6e1) {
                if (!(this instanceof _0x4a70ff))
                    return new _0x4a70ff(_0x1d6205, _0x1eb6e1);
                _0xb07161['call'](this), this['path'] = _0x1d6205, this['fd'] = null, this['writable'] = !![], this['flags'] = 'w', this['encoding'] = 'binary', this['mode'] = 0x1b6, this['bytesWritten'] = 0x0, _0x1eb6e1 = _0x1eb6e1 || {};
                var _0x5b435a = Object['keys'](_0x1eb6e1);
                for (var _0x2dfadf = 0x0, _0x4870f1 = _0x5b435a['length']; _0x2dfadf < _0x4870f1; _0x2dfadf++) {
                    var _0x4603af = _0x5b435a[_0x2dfadf];
                    this[_0x4603af] = _0x1eb6e1[_0x4603af];
                }
                if (this['start'] !== undefined) {
                    if ('number' !== typeof this['start'])
                        throw TypeError('start\x20must\x20be\x20a\x20Number');
                    if (this['start'] < 0x0)
                        throw new Error('start\x20must\x20be\x20>=\x20zero');
                    this['pos'] = this['start'];
                }
                this['busy'] = ![], this['_queue'] = [], this['fd'] === null && (this['_open'] = _0x1c8728['open'], this['_queue']['push']([
                    this['_open'],
                    this['path'],
                    this['flags'],
                    this['mode'],
                    undefined
                ]), this['flush']());
            }
        }
    },
    0x15c1b: (_0x387944, _0x4ac4f1, _0x555491) => {
        var _0x3bb68d = _0x555491(0x5629), _0x99ed12 = process['cwd'], _0x1cdb1b = null, _0x1b7438 = process['env']['GRACEFUL_FS_PLATFORM'] || process['platform'];
        process['cwd'] = function () {
            if (!_0x1cdb1b)
                _0x1cdb1b = _0x99ed12['call'](process);
            return _0x1cdb1b;
        };
        try {
            process['cwd']();
        } catch (_0x3180fe) {
        }
        if (typeof process['chdir'] === 'function') {
            var _0x37fcd4 = process['chdir'];
            process['chdir'] = function (_0x32bafd) {
                _0x1cdb1b = null, _0x37fcd4['call'](process, _0x32bafd);
            };
            if (Object['setPrototypeOf'])
                Object['setPrototypeOf'](process['chdir'], _0x37fcd4);
        }
        _0x387944['exports'] = _0x551535;
        function _0x551535(_0x3e3bdd) {
            _0x3bb68d['hasOwnProperty']('O_SYMLINK') && process['version']['match'](/^v0\.6\.[0-2]|^v0\.5\./) && _0x24c4cc(_0x3e3bdd);
            !_0x3e3bdd['lutimes'] && _0x553a63(_0x3e3bdd);
            _0x3e3bdd['chown'] = _0x592bf0(_0x3e3bdd['chown']), _0x3e3bdd['fchown'] = _0x592bf0(_0x3e3bdd['fchown']), _0x3e3bdd['lchown'] = _0x592bf0(_0x3e3bdd['lchown']), _0x3e3bdd['chmod'] = _0x49a47e(_0x3e3bdd['chmod']), _0x3e3bdd['fchmod'] = _0x49a47e(_0x3e3bdd['fchmod']), _0x3e3bdd['lchmod'] = _0x49a47e(_0x3e3bdd['lchmod']), _0x3e3bdd['chownSync'] = _0xf434d3(_0x3e3bdd['chownSync']), _0x3e3bdd['fchownSync'] = _0xf434d3(_0x3e3bdd['fchownSync']), _0x3e3bdd['lchownSync'] = _0xf434d3(_0x3e3bdd['lchownSync']), _0x3e3bdd['chmodSync'] = _0x3b6b4a(_0x3e3bdd['chmodSync']), _0x3e3bdd['fchmodSync'] = _0x3b6b4a(_0x3e3bdd['fchmodSync']), _0x3e3bdd['lchmodSync'] = _0x3b6b4a(_0x3e3bdd['lchmodSync']), _0x3e3bdd['stat'] = _0x1f2e8b(_0x3e3bdd['stat']), _0x3e3bdd['fstat'] = _0x1f2e8b(_0x3e3bdd['fstat']), _0x3e3bdd['lstat'] = _0x1f2e8b(_0x3e3bdd['lstat']), _0x3e3bdd['statSync'] = _0x26f934(_0x3e3bdd['statSync']), _0x3e3bdd['fstatSync'] = _0x26f934(_0x3e3bdd['fstatSync']), _0x3e3bdd['lstatSync'] = _0x26f934(_0x3e3bdd['lstatSync']);
            _0x3e3bdd['chmod'] && !_0x3e3bdd['lchmod'] && (_0x3e3bdd['lchmod'] = function (_0x4467c8, _0x181be6, _0x31c70d) {
                if (_0x31c70d)
                    process['nextTick'](_0x31c70d);
            }, _0x3e3bdd['lchmodSync'] = function () {
            });
            _0x3e3bdd['chown'] && !_0x3e3bdd['lchown'] && (_0x3e3bdd['lchown'] = function (_0x1e69b4, _0x12f460, _0x42c2f2, _0x441110) {
                if (_0x441110)
                    process['nextTick'](_0x441110);
            }, _0x3e3bdd['lchownSync'] = function () {
            });
            _0x1b7438 === 'win32' && (_0x3e3bdd['rename'] = typeof _0x3e3bdd['rename'] !== 'function' ? _0x3e3bdd['rename'] : function (_0xd3b36) {
                function _0x5c49b9(_0x1f96ed, _0x16f55e, _0xcf0ed4) {
                    var _0x1c648b = Date['now'](), _0x30fe6d = 0x0;
                    _0xd3b36(_0x1f96ed, _0x16f55e, function _0x1fba15(_0x8f26ce) {
                        if (_0x8f26ce && (_0x8f26ce['code'] === 'EACCES' || _0x8f26ce['code'] === 'EPERM') && Date['now']() - _0x1c648b < 0xea60) {
                            setTimeout(function () {
                                _0x3e3bdd['stat'](_0x16f55e, function (_0x4a5588, _0x1689d5) {
                                    if (_0x4a5588 && _0x4a5588['code'] === 'ENOENT')
                                        _0xd3b36(_0x1f96ed, _0x16f55e, _0x1fba15);
                                    else
                                        _0xcf0ed4(_0x8f26ce);
                                });
                            }, _0x30fe6d);
                            if (_0x30fe6d < 0x64)
                                _0x30fe6d += 0xa;
                            return;
                        }
                        if (_0xcf0ed4)
                            _0xcf0ed4(_0x8f26ce);
                    });
                }
                if (Object['setPrototypeOf'])
                    Object['setPrototypeOf'](_0x5c49b9, _0xd3b36);
                return _0x5c49b9;
            }(_0x3e3bdd['rename']));
            _0x3e3bdd['read'] = typeof _0x3e3bdd['read'] !== 'function' ? _0x3e3bdd['read'] : function (_0x2de450) {
                function _0x1d889e(_0xe019fc, _0x278d78, _0x6b7cee, _0x4c12cc, _0x11ea6f, _0x2751fc) {
                    var _0x7f3d92;
                    if (_0x2751fc && typeof _0x2751fc === 'function') {
                        var _0x4d9ecd = 0x0;
                        _0x7f3d92 = function (_0x102b3c, _0x30c95c, _0x1f47a3) {
                            if (_0x102b3c && _0x102b3c['code'] === 'EAGAIN' && _0x4d9ecd < 0xa)
                                return _0x4d9ecd++, _0x2de450['call'](_0x3e3bdd, _0xe019fc, _0x278d78, _0x6b7cee, _0x4c12cc, _0x11ea6f, _0x7f3d92);
                            _0x2751fc['apply'](this, arguments);
                        };
                    }
                    return _0x2de450['call'](_0x3e3bdd, _0xe019fc, _0x278d78, _0x6b7cee, _0x4c12cc, _0x11ea6f, _0x7f3d92);
                }
                if (Object['setPrototypeOf'])
                    Object['setPrototypeOf'](_0x1d889e, _0x2de450);
                return _0x1d889e;
            }(_0x3e3bdd['read']), _0x3e3bdd['readSync'] = typeof _0x3e3bdd['readSync'] !== 'function' ? _0x3e3bdd['readSync'] : function (_0x2e2c51) {
                return function (_0x4e5b25, _0x5a7297, _0x2981d9, _0x51dceb, _0x1e43fd) {
                    var _0x4bcc15 = 0x0;
                    while (!![]) {
                        try {
                            return _0x2e2c51['call'](_0x3e3bdd, _0x4e5b25, _0x5a7297, _0x2981d9, _0x51dceb, _0x1e43fd);
                        } catch (_0x13982b) {
                            if (_0x13982b['code'] === 'EAGAIN' && _0x4bcc15 < 0xa) {
                                _0x4bcc15++;
                                continue;
                            }
                            throw _0x13982b;
                        }
                    }
                };
            }(_0x3e3bdd['readSync']);
            function _0x24c4cc(_0x338b81) {
                _0x338b81['lchmod'] = function (_0x3c3995, _0x10ea44, _0x64de3c) {
                    _0x338b81['open'](_0x3c3995, _0x3bb68d['O_WRONLY'] | _0x3bb68d['O_SYMLINK'], _0x10ea44, function (_0x2b8bbc, _0x3982f0) {
                        if (_0x2b8bbc) {
                            if (_0x64de3c)
                                _0x64de3c(_0x2b8bbc);
                            return;
                        }
                        _0x338b81['fchmod'](_0x3982f0, _0x10ea44, function (_0x19cff0) {
                            _0x338b81['close'](_0x3982f0, function (_0x20c101) {
                                if (_0x64de3c)
                                    _0x64de3c(_0x19cff0 || _0x20c101);
                            });
                        });
                    });
                }, _0x338b81['lchmodSync'] = function (_0x5d6845, _0x26778b) {
                    var _0x3a3569 = _0x338b81['openSync'](_0x5d6845, _0x3bb68d['O_WRONLY'] | _0x3bb68d['O_SYMLINK'], _0x26778b), _0xc3852a = !![], _0x46a325;
                    try {
                        _0x46a325 = _0x338b81['fchmodSync'](_0x3a3569, _0x26778b), _0xc3852a = ![];
                    } finally {
                        if (_0xc3852a)
                            try {
                                _0x338b81['closeSync'](_0x3a3569);
                            } catch (_0x30c0ca) {
                            }
                        else
                            _0x338b81['closeSync'](_0x3a3569);
                    }
                    return _0x46a325;
                };
            }
            function _0x553a63(_0x58ffa5) {
                if (_0x3bb68d['hasOwnProperty']('O_SYMLINK') && _0x58ffa5['futimes'])
                    _0x58ffa5['lutimes'] = function (_0x3abefb, _0x14f598, _0x45202c, _0x2937e1) {
                        _0x58ffa5['open'](_0x3abefb, _0x3bb68d['O_SYMLINK'], function (_0x5e12d2, _0x3708da) {
                            if (_0x5e12d2) {
                                if (_0x2937e1)
                                    _0x2937e1(_0x5e12d2);
                                return;
                            }
                            _0x58ffa5['futimes'](_0x3708da, _0x14f598, _0x45202c, function (_0x562b17) {
                                _0x58ffa5['close'](_0x3708da, function (_0x1cab5e) {
                                    if (_0x2937e1)
                                        _0x2937e1(_0x562b17 || _0x1cab5e);
                                });
                            });
                        });
                    }, _0x58ffa5['lutimesSync'] = function (_0x679b60, _0x589bcb, _0x46974d) {
                        var _0x26b1e2 = _0x58ffa5['openSync'](_0x679b60, _0x3bb68d['O_SYMLINK']), _0x25b720, _0x4a7630 = !![];
                        try {
                            _0x25b720 = _0x58ffa5['futimesSync'](_0x26b1e2, _0x589bcb, _0x46974d), _0x4a7630 = ![];
                        } finally {
                            if (_0x4a7630)
                                try {
                                    _0x58ffa5['closeSync'](_0x26b1e2);
                                } catch (_0x35f15f) {
                                }
                            else
                                _0x58ffa5['closeSync'](_0x26b1e2);
                        }
                        return _0x25b720;
                    };
                else
                    _0x58ffa5['futimes'] && (_0x58ffa5['lutimes'] = function (_0x5253fd, _0x4fe693, _0xbfedb5, _0x2ec91b) {
                        if (_0x2ec91b)
                            process['nextTick'](_0x2ec91b);
                    }, _0x58ffa5['lutimesSync'] = function () {
                    });
            }
            function _0x49a47e(_0x5a8a91) {
                if (!_0x5a8a91)
                    return _0x5a8a91;
                return function (_0x41347f, _0x258828, _0x422645) {
                    return _0x5a8a91['call'](_0x3e3bdd, _0x41347f, _0x258828, function (_0x1b79c4) {
                        if (_0x77adfb(_0x1b79c4))
                            _0x1b79c4 = null;
                        if (_0x422645)
                            _0x422645['apply'](this, arguments);
                    });
                };
            }
            function _0x3b6b4a(_0xc7489c) {
                if (!_0xc7489c)
                    return _0xc7489c;
                return function (_0x315bf9, _0x1220ea) {
                    try {
                        return _0xc7489c['call'](_0x3e3bdd, _0x315bf9, _0x1220ea);
                    } catch (_0x2afbb4) {
                        if (!_0x77adfb(_0x2afbb4))
                            throw _0x2afbb4;
                    }
                };
            }
            function _0x592bf0(_0x4bf457) {
                if (!_0x4bf457)
                    return _0x4bf457;
                return function (_0x3f2bf0, _0x4a9b72, _0x5cce15, _0x3d20f9) {
                    return _0x4bf457['call'](_0x3e3bdd, _0x3f2bf0, _0x4a9b72, _0x5cce15, function (_0x58b04e) {
                        if (_0x77adfb(_0x58b04e))
                            _0x58b04e = null;
                        if (_0x3d20f9)
                            _0x3d20f9['apply'](this, arguments);
                    });
                };
            }
            function _0xf434d3(_0x1d61df) {
                if (!_0x1d61df)
                    return _0x1d61df;
                return function (_0x44a6b3, _0x181e35, _0x158d05) {
                    try {
                        return _0x1d61df['call'](_0x3e3bdd, _0x44a6b3, _0x181e35, _0x158d05);
                    } catch (_0x206b77) {
                        if (!_0x77adfb(_0x206b77))
                            throw _0x206b77;
                    }
                };
            }
            function _0x1f2e8b(_0x401a2a) {
                if (!_0x401a2a)
                    return _0x401a2a;
                return function (_0xae5336, _0x107fcf, _0x204fb0) {
                    typeof _0x107fcf === 'function' && (_0x204fb0 = _0x107fcf, _0x107fcf = null);
                    function _0x1a5360(_0xe86c8c, _0x28ecc9) {
                        if (_0x28ecc9) {
                            if (_0x28ecc9['uid'] < 0x0)
                                _0x28ecc9['uid'] += 0x100000000;
                            if (_0x28ecc9['gid'] < 0x0)
                                _0x28ecc9['gid'] += 0x100000000;
                        }
                        if (_0x204fb0)
                            _0x204fb0['apply'](this, arguments);
                    }
                    return _0x107fcf ? _0x401a2a['call'](_0x3e3bdd, _0xae5336, _0x107fcf, _0x1a5360) : _0x401a2a['call'](_0x3e3bdd, _0xae5336, _0x1a5360);
                };
            }
            function _0x26f934(_0x36b983) {
                if (!_0x36b983)
                    return _0x36b983;
                return function (_0x383062, _0x14b6e5) {
                    var _0x476c90 = _0x14b6e5 ? _0x36b983['call'](_0x3e3bdd, _0x383062, _0x14b6e5) : _0x36b983['call'](_0x3e3bdd, _0x383062);
                    if (_0x476c90) {
                        if (_0x476c90['uid'] < 0x0)
                            _0x476c90['uid'] += 0x100000000;
                        if (_0x476c90['gid'] < 0x0)
                            _0x476c90['gid'] += 0x100000000;
                    }
                    return _0x476c90;
                };
            }
            function _0x77adfb(_0x20adbc) {
                if (!_0x20adbc)
                    return !![];
                if (_0x20adbc['code'] === 'ENOSYS')
                    return !![];
                var _0x5c88e7 = !process['getuid'] || process['getuid']() !== 0x0;
                if (_0x5c88e7) {
                    if (_0x20adbc['code'] === 'EINVAL' || _0x20adbc['code'] === 'EPERM')
                        return !![];
                }
                return ![];
            }
        }
    },
    0x8517: (_0x12a1ce, _0x5b9201, _0x49c90d) => {
        let _0x402e58;
        try {
            _0x402e58 = _0x49c90d(0x15d1a);
        } catch (_0x13c57b) {
            _0x402e58 = _0x49c90d(0xdf3b);
        }
        const _0x4a2146 = _0x49c90d(0x15f6a), {
                stringify: _0x1e2561,
                stripBom: _0x2f2d52
            } = _0x49c90d(0x14415);
        async function _0x17804f(_0x42f43a, _0x810f64 = {}) {
            typeof _0x810f64 === 'string' && (_0x810f64 = { 'encoding': _0x810f64 });
            const _0x2403c3 = _0x810f64['fs'] || _0x402e58, _0x36f5a5 = 'throws' in _0x810f64 ? _0x810f64['throws'] : !![];
            let _0xb02625 = await _0x4a2146['fromCallback'](_0x2403c3['readFile'])(_0x42f43a, _0x810f64);
            _0xb02625 = _0x2f2d52(_0xb02625);
            let _0x3ebb3a;
            try {
                _0x3ebb3a = JSON['parse'](_0xb02625, _0x810f64 ? _0x810f64['reviver'] : null);
            } catch (_0x54a5c4) {
                if (_0x36f5a5) {
                    _0x54a5c4['message'] = _0x42f43a + ':\x20' + _0x54a5c4['message'];
                    throw _0x54a5c4;
                } else
                    return null;
            }
            return _0x3ebb3a;
        }
        const _0x2580bd = _0x4a2146['fromPromise'](_0x17804f);
        function _0x333dfa(_0xb354d0, _0x1eba61 = {}) {
            typeof _0x1eba61 === 'string' && (_0x1eba61 = { 'encoding': _0x1eba61 });
            const _0x56c186 = _0x1eba61['fs'] || _0x402e58, _0x437ae3 = 'throws' in _0x1eba61 ? _0x1eba61['throws'] : !![];
            try {
                let _0xf28b44 = _0x56c186['readFileSync'](_0xb354d0, _0x1eba61);
                return _0xf28b44 = _0x2f2d52(_0xf28b44), JSON['parse'](_0xf28b44, _0x1eba61['reviver']);
            } catch (_0x5a9743) {
                if (_0x437ae3) {
                    _0x5a9743['message'] = _0xb354d0 + ':\x20' + _0x5a9743['message'];
                    throw _0x5a9743;
                } else
                    return null;
            }
        }
        async function _0x163eb4(_0x268c12, _0x3d9388, _0xf7e21a = {}) {
            const _0x145ff0 = _0xf7e21a['fs'] || _0x402e58, _0x5f385c = _0x1e2561(_0x3d9388, _0xf7e21a);
            await _0x4a2146['fromCallback'](_0x145ff0['writeFile'])(_0x268c12, _0x5f385c, _0xf7e21a);
        }
        const _0x2a9aab = _0x4a2146['fromPromise'](_0x163eb4);
        function _0x3a1aee(_0x218d0e, _0x2d35d0, _0x100642 = {}) {
            const _0x3027ed = _0x100642['fs'] || _0x402e58, _0x3849f8 = _0x1e2561(_0x2d35d0, _0x100642);
            return _0x3027ed['writeFileSync'](_0x218d0e, _0x3849f8, _0x100642);
        }
        const _0x900ecb = {
            'readFile': _0x2580bd,
            'readFileSync': _0x333dfa,
            'writeFile': _0x2a9aab,
            'writeFileSync': _0x3a1aee
        };
        _0x12a1ce['exports'] = _0x900ecb;
    },
    0x14415: _0x454923 => {
        function _0x12a3f8(_0xa9a89b, {
            EOL: EOL = '\x0a',
            finalEOL: finalEOL = !![],
            replacer: replacer = null,
            spaces: _0x357426
        } = {}) {
            const _0x2a4f5e = finalEOL ? EOL : '', _0x4168aa = JSON['stringify'](_0xa9a89b, replacer, _0x357426);
            return _0x4168aa['replace'](/\n/g, EOL) + _0x2a4f5e;
        }
        function _0x387804(_0x24f644) {
            if (Buffer['isBuffer'](_0x24f644))
                _0x24f644 = _0x24f644['toString']('utf8');
            return _0x24f644['replace'](/^\uFEFF/, '');
        }
        _0x454923['exports'] = {
            'stringify': _0x12a3f8,
            'stripBom': _0x387804
        };
    },
    0x15f6a: (_0x561e8f, _0x408e24) => {
        'use strict';
        _0x408e24['fromCallback'] = function (_0x189ff4) {
            return Object['defineProperty'](function (..._0x13cd5c) {
                if (typeof _0x13cd5c[_0x13cd5c['length'] - 0x1] === 'function')
                    _0x189ff4['apply'](this, _0x13cd5c);
                else
                    return new Promise((_0x4db041, _0x242c3d) => {
                        _0x189ff4['call'](this, ..._0x13cd5c, (_0x17684c, _0x5071c2) => _0x17684c != null ? _0x242c3d(_0x17684c) : _0x4db041(_0x5071c2));
                    });
            }, 'name', { 'value': _0x189ff4['name'] });
        }, _0x408e24['fromPromise'] = function (_0x41b418) {
            return Object['defineProperty'](function (..._0x1ae578) {
                const _0x4561d0 = _0x1ae578[_0x1ae578['length'] - 0x1];
                if (typeof _0x4561d0 !== 'function')
                    return _0x41b418['apply'](this, _0x1ae578);
                else
                    _0x41b418['apply'](this, _0x1ae578['slice'](0x0, -0x1))['then'](_0x41f867 => _0x4561d0(null, _0x41f867), _0x4561d0);
            }, 'name', { 'value': _0x41b418['name'] });
        };
    }
};
;
