// test-maker/dist/32.js
'use strict';
exports['id'] = 0x20, exports['ids'] = [
    0x20,
    0x23d
], exports['modules'] = {
    0xd6f3: (_0xd5ac8c, _0x179aa2, _0x53c2a2) => {
        _0x53c2a2['r'](_0x179aa2), _0x53c2a2['d'](_0x179aa2, {
            'Runner': () => _0x301430,
            'runner': () => _0x312216
        });
        var _0x38909f = _0x53c2a2(0x1145c), _0xbddbf7 = _0x53c2a2['n'](_0x38909f), _0x32d909 = _0x53c2a2(0x5885), _0x5af5d5 = _0x53c2a2(0x177ce), _0x2f11e5 = _0x53c2a2['n'](_0x5af5d5), _0x54884a = _0x53c2a2(0x17fbe), _0x1a95af = _0x53c2a2(0x9677), _0x2a8080 = _0x53c2a2(0x412a), _0x2c1bcc = _0x53c2a2['n'](_0x2a8080), _0x5156c1 = _0x53c2a2(0x3b1), _0x236fb5 = _0x53c2a2['n'](_0x5156c1), _0x5d48e3 = _0x53c2a2(0x135fa), _0x35a5a7 = _0x53c2a2['n'](_0x5d48e3), _0x3d5b3c = _0x53c2a2(0x13833), _0x55a20b = _0x53c2a2(0x1fee), _0x3b81e9 = _0x53c2a2(0xb4de), _0x5c7a3b = _0x53c2a2(0xc5d1), _0xd8b472 = _0x53c2a2(0x111e7), _0x35280b = _0x53c2a2(0x149e5);
        ;
        function _0x226a36(_0x41cecc) {
            const _0x4cc85a = _0x41cecc['toLowerCase']();
            switch (_0x4cc85a) {
            case 'true':
                return !![];
            case 'false':
                return ![];
            case 'nan':
                return NaN;
            case '':
            case 'undefined':
                return undefined;
            case 'null':
                return null;
            default:
                return isNaN(Number(_0x4cc85a)) ? _0x41cecc : Number(_0x4cc85a);
            }
        }
        var _0x541c32 = _0x53c2a2(0x51b9), _0x347c0d = _0x53c2a2(0x845b), _0x330604 = _0x53c2a2(0x1993), _0x23065b = _0x53c2a2(0x17a85), _0x424d23 = _0x53c2a2(0xf590), _0x326303 = _0x53c2a2(0x71b5), _0x20a9dd = _0x53c2a2(0x13b9a), _0x3d246a = _0x53c2a2(0x1e99);
        ;
        class _0x301430 {
            async ['start'](_0x139f64) {
                this['options'] = _0x139f64, await this['compileConfig']();
                if (_0x55a20b['configManager']['configuration']['runner']['liveMode'])
                    await this['watchTestMakerConfig']();
                _0x5c7a3b['stateManager']['state']['cliArgs'] = this['options'], _0x3d246a['internalLogger']['level'] = _0x55a20b['configManager']['configuration']['logging']['globalLogLevel'], await _0x3b81e9['instanceManager']['init'](), await _0x35280b['tmEventsManger']['emit'](_0xd8b472['TMEvents']['EngineStart']);
            }
            async ['watchTestMakerConfig']() {
                if (this['options']['config'])
                    _0x32d909['watch'](_0x236fb5()['joinSafe'](_0x3d5b3c['projectPath'], this['options']['config']), {
                        'persistent': !![],
                        'awaitWriteFinish': !![],
                        'ignoreInitial': !![]
                    })['on']('all', async () => {
                        _0x2f11e5()['warn']('Test\x20Maker\x20Config\x20file\x20has\x20been\x20modified,\x20Test\x20Maker\x20needs\x20to\x20be\x20restarted.'), process['exitCode'] = 0x400, (0x0, _0x541c32['exitProcess'])(0x0);
                    });
            }
            async ['compileConfig']() {
                if (this['options']['config']) {
                    const _0x558b61 = _0x236fb5()['resolve'](_0x236fb5()['normalizeTrim'](this['options']['config']));
                    _0x2f11e5()['info']('Compiling\x20Test\x20Maker\x20Config\x20File:\x20' + _0x558b61);
                    const _0xbd1c1f = await (0x0, _0x20a9dd['webpackMemoryCompiler'])({
                            'webpackCompiler': _0x35a5a7(),
                            'config': (await _0x53c2a2['e'](0x1fe)['then'](_0x53c2a2['bind'](_0x53c2a2, 0x7b16)))['getBundleConfig'](this['options']['config'], _0x558b61)
                        }), _0x26c96a = _0x3d5b3c['testMakerArtifactsPath'] + '/' + _0x236fb5()['basename'](_0x558b61);
                    await _0x54884a['outputFile'](_0x26c96a, _0xbd1c1f);
                    const _0x31ca4d = (0x0, _0x23065b['nativeRequire'])(_0x26c96a), _0x367be4 = this['options']['config'] ? _0x31ca4d['default'] : {};
                    this['setupOptions']({
                        'options': this['options'],
                        'fileConfiguration': _0x367be4
                    });
                } else
                    this['setupOptions']({
                        'options': this['options'],
                        'fileConfiguration': {}
                    });
            }
            ['setupOptions']({
                options: _0x15191a,
                fileConfiguration: _0x4cbe53
            }) {
                this['mergeConfig']({
                    'options': _0x15191a,
                    'fileConfiguration': _0x4cbe53
                }), this['validateNodeversionOption'](), this['normalizeFilteringOptions'](), this['normalizeOverrideOptions'](_0x15191a), this['normalizeSuitesOptions'](_0x15191a), this['normalizeExtractions'](_0x15191a), this['normalizeOutputOptions']();
            }
            ['parseExtraOptions'](_0x204fa2, _0x261f28) {
                try {
                    const _0x31c31f = new _0x424d23['PropertyAccessor'](_0x261f28), _0x3c5a92 = {}, _0x1e8d0e = Array['isArray'](_0x204fa2) ? _0x204fa2 : [exports];
                    return _0x1e8d0e['forEach'](_0x2f891a => {
                        const _0x410d93 = _0x2f891a['split']('=');
                        if (_0x410d93['length'] > 0x1) {
                            const _0x26a9a2 = _0x226a36(_0x410d93[0x1]);
                            _0x31c31f['target'] = (0x0, _0x330604['mergeAndConcat'])(_0x31c31f['target'], _0x31c31f['set'](_0x410d93[0x0], _0x26a9a2)), _0x3c5a92[_0x410d93[0x0]] = _0x26a9a2;
                        } else
                            _0x3c5a92[_0x2f891a] = _0x31c31f['get'](_0x2f891a);
                    }), _0x3c5a92;
                } catch (_0x5d8047) {
                    _0x5d8047['name'] = 'Extra\x20Options\x20Setup', _0x5d8047['message'] = '' + _0x5d8047['message'];
                    throw _0x5d8047;
                }
            }
            ['normalizeFilteringOptions']() {
                const _0x16c8e9 = _0x55a20b['configManager']['configuration']['filtering'], _0x34e57a = _0x673037 => {
                        return _0x673037 ? Array['isArray'](_0x673037) ? _0x673037 : [_0x673037] : [];
                    };
                _0x16c8e9 && (_0x16c8e9['features'] = _0x34e57a(_0x16c8e9['features']), _0x16c8e9['featuresTags'] = _0x34e57a(_0x16c8e9['featuresTags']), _0x16c8e9['scenarios'] = _0x34e57a(_0x16c8e9['scenarios']), _0x16c8e9['scenariosTags'] = _0x34e57a(_0x16c8e9['scenariosTags']), _0x16c8e9['steps'] = _0x34e57a(_0x16c8e9['steps']), _0x16c8e9['stepsTags'] = _0x34e57a(_0x16c8e9['stepsTags']));
            }
            ['validateNodeversionOption']() {
                var _0x296bac, _0x19cdbd;
                if ((_0x296bac = _0x55a20b['configManager']['configuration']['runner']) === null || _0x296bac === void 0x0 ? void 0x0 : _0x296bac['nodeVersion'])
                    !_0x2c1bcc()['satisfies'](process['version'], (_0x19cdbd = _0x55a20b['configManager']['configuration']['runner']) === null || _0x19cdbd === void 0x0 ? void 0x0 : _0x19cdbd['nodeVersion']) && (_0x2f11e5()['error']('The\x20Specified\x20NodeJS\x20version:' + _0xbddbf7()['yellowBright'](_0x55a20b['configManager']['configuration']['runner']['nodeVersion']) + '\x20in\x20config\x20does\x20not\x20satisfy\x20the\x20one\x20installed:\x20' + _0xbddbf7()['yellowBright'](process['version'])), (0x0, _0x541c32['exitProcess'])(0x1));
                else
                    !_0x2c1bcc()['satisfies'](process['version'], '>=14.x') && _0x2f11e5()['warn']('Test\x20Maker\x20may\x20not\x20work\x20correctly\x20with\x20Node\x20Version\x20less\x20than\x20\x2214x\x20-\x20current\x20installed\x20Node\x20version:\x20' + process['version']);
            }
            ['normalizeSuitesOptions'](_0x43f8ee) {
                const _0x346890 = {};
                _0x43f8ee['suites']['forEach'](_0x548204 => {
                    var _0x1fefc3;
                    let _0x48b1c2 = (_0x1fefc3 = _0x55a20b['configManager']['configuration']['suites']) === null || _0x1fefc3 === void 0x0 ? void 0x0 : _0x1fefc3[_0x548204];
                    if (!_0x48b1c2)
                        throw new Error('Suite:\x20\x22' + _0x548204 + '\x22\x20is\x20not\x20defined\x20in\x20Config\x20file.\x20please\x20add\x20one\x20with\x20the\x20same\x20name.');
                    if (Array['isArray'](_0x48b1c2))
                        _0x48b1c2 = { 'path': _0x48b1c2 };
                    _0x346890[_0x548204] = _0x48b1c2;
                });
                if (Object['keys'](_0x346890)['length'])
                    _0x55a20b['configManager']['configuration']['suites'] = _0x346890;
            }
            ['normalizeExtractions'](_0x430320) {
                this['parseExtraOptions'](_0x430320['extra'], _0x55a20b['configManager']['configuration']['extra'] || {});
            }
            ['normalizeOutputOptions']() {
                var _0x39cdcf, _0x36ec4c, _0x3266bb, _0x4ddb3c;
                if ((_0x36ec4c = (_0x39cdcf = _0x55a20b['configManager']['configuration']['build']) === null || _0x39cdcf === void 0x0 ? void 0x0 : _0x39cdcf['output']) === null || _0x36ec4c === void 0x0 ? void 0x0 : _0x36ec4c['path'])
                    _0x55a20b['configManager']['configuration']['build']['output']['path'] = _0x236fb5()['resolve'](((_0x4ddb3c = (_0x3266bb = _0x55a20b['configManager']['configuration']['build']) === null || _0x3266bb === void 0x0 ? void 0x0 : _0x3266bb['output']) === null || _0x4ddb3c === void 0x0 ? void 0x0 : _0x4ddb3c['path']) || '');
            }
            ['normalizeOverrideOptions'](_0x541507) {
                if (_0x541507['override'] && _0x541507['override'] !== 'true' && _0x541507['override']['length'])
                    _0x55a20b['configManager']['configuration'] = (0x0, _0x1a95af['merge'])(_0x55a20b['configManager']['configuration'], (0x0, _0x326303['virtualModule'])('module.exports=\x20' + _0x541507['override'], (0x0, _0x347c0d['generateId'])()));
            }
            ['mergeConfig']({
                options: _0x31d7f9,
                fileConfiguration: _0x2de3cc
            }) {
                _0x55a20b['configManager']['configuration'] = (0x0, _0x1a95af['merge'])(_0x55a20b['configManager']['configuration'], _0x2de3cc);
                const _0x6debdd = {
                    'runner': {
                        'startPage': _0x31d7f9['startPage'],
                        'liveMode': _0x31d7f9['liveMode'],
                        'headless': _0x31d7f9['headless'],
                        'parallel': {
                            'features': _0x31d7f9['parallelFeatures'],
                            'clients': _0x31d7f9['parallelClients'],
                            'adapters': _0x31d7f9['parallelAdapters']
                        }
                    },
                    'reporting': { 'reporters': _0x31d7f9['reporters'] },
                    'source': _0x31d7f9['source'],
                    'filtering': {
                        'glob': _0x31d7f9['filterGlob'],
                        'features': _0x31d7f9['filterFeature'],
                        'featuresTags': _0x31d7f9['filterFeatureTag'],
                        'scenarios': _0x31d7f9['filterScenario'],
                        'scenariosTags': _0x31d7f9['filterScenarioTag'],
                        'steps': _0x31d7f9['filterStep'],
                        'stepsTags': _0x31d7f9['filterStepTag']
                    }
                };
                _0x55a20b['configManager']['configuration'] = (0x0, _0x1a95af['mergeAndCompare'])((_0x17665a, _0x206fdb) => {
                    return _0x206fdb === undefined ? _0x17665a : _0x206fdb;
                }, _0x55a20b['configManager']['configuration'], _0x6debdd);
            }
        }
        const _0x312216 = new _0x301430();
    },
    0x38ed: (_0x467cc7, _0x8f6a28, _0x41d9a9) => {
        _0x41d9a9['r'](_0x8f6a28), _0x41d9a9['d'](_0x8f6a28, { 'logGenericError': () => _0x4b4571 });
        var _0x4a2f9a = _0x41d9a9(0x1145c), _0x3dc027 = _0x41d9a9['n'](_0x4a2f9a);
        async function _0x4b4571({
            error: _0x305d8c,
            isWarning: _0x55a26b,
            show: show = !![]
        }) {
            let _0x324977;
            try {
                _0x324977 ? _0x55a26b ? console['warn'](_0x324977['formatted']['ansi']) : console['error'](_0x324977['formatted']['ansi']) : _0x55a26b ? console['warn'](_0x305d8c) : console['error'](_0x3dc027()['redBright'](_0x305d8c['stack'] || _0x305d8c));
            } catch (_0x43f160) {
                console['error'](_0x3dc027()['redBright'](_0x43f160['stack'] || _0x43f160));
            } finally {
                console['info'](_0x3dc027()['cyanBright']('See\x20Log\x20file\x20for\x20more\x20Details...'));
            }
        }
    },
    0x13b9a: (_0x1ed453, _0x236063, _0x209196) => {
        _0x209196['r'](_0x236063), _0x209196['d'](_0x236063, { 'webpackMemoryCompiler': () => _0x59067c });
        var _0x2f5817 = _0x209196(0x17fbe), _0x41761b = _0x209196['n'](_0x2f5817), _0xef7a7b = _0x209196(0x5b3f), _0x5d6d3d = _0x209196['n'](_0xef7a7b), _0x348e9d = _0x209196(0x5615), _0x1d1158 = _0x209196['n'](_0x348e9d), _0x3ec928 = _0x209196(0x11569), _0x101b56 = _0x209196['n'](_0x3ec928), _0x88e413 = _0x209196(0x845b);
        function _0x273e2d(_0x5d2872 = (0x0, _0x88e413['generateId'])()) {
            const _0x1689d2 = (0x0, _0x348e9d['tmpdir'])(), _0x802c49 = _0x5d2872 + '.js', _0x15b242 = (0x0, _0x3ec928['join'])(_0x1689d2, _0x802c49), _0x54a5cb = async _0x4e0452 => {
                    await (0x0, _0x2f5817['writeFile'])(_0x15b242, _0x4e0452);
                }, _0x5e5e86 = async () => {
                    await (0x0, _0x2f5817['unlink'])(_0x15b242);
                };
            return {
                'path': _0x1689d2,
                'filename': _0x802c49,
                'filepath': _0x15b242,
                'write': _0x54a5cb,
                'remove': _0x5e5e86
            };
        }
        async function _0x59067c({
            webpackCompiler: _0x2f09b4,
            config: config = { 'mode': 'production' },
            code: _0xb38d7d
        }) {
            const _0x25f693 = {
                ...config,
                'output': {
                    ...config['output'],
                    'path': '/',
                    'filename': 'output.js'
                }
            };
            let _0x26f14b;
            _0xb38d7d && (_0x26f14b = _0x273e2d(), await _0x26f14b['write'](_0xb38d7d), _0x25f693['entry'] = _0x26f14b['filepath']);
            const _0x43bf8e = _0x2f09b4(_0x25f693);
            _0x43bf8e['outputFileSystem'] = new (_0x5d6d3d())();
            const _0x381e6a = await new Promise((_0x4be351, _0x100350) => {
                _0x43bf8e['run']((_0x440679, _0x4ffa64) => {
                    if (_0x440679)
                        return _0x100350(_0x440679);
                    if (_0x4ffa64 === null || _0x4ffa64 === void 0x0 ? void 0x0 : _0x4ffa64['hasErrors']())
                        return _0x100350(new Error(_0x4ffa64['toString']({
                            'errorDetails': !![],
                            'warnings': !![]
                        })));
                    _0x4be351(_0x43bf8e['outputFileSystem']['data']['output.js']['toString']());
                });
            });
            if (_0x26f14b)
                await _0x26f14b['remove']();
            return _0x381e6a;
        }
    }
};
;
