// test-maker/dist/386.js
exports['id'] = 0x182, exports['ids'] = [0x182], exports['modules'] = {
    0x7d39: (_0xcf0f25, _0x3bd66e, _0x3ce7c6) => {
        'use strict';
        _0xcf0f25 = _0x3ce7c6['nmd'](_0xcf0f25);
        const _0xc3bbc5 = (_0x3bf911, _0x3ca4ca) => (..._0x110b7d) => {
                const _0x466ff1 = _0x3bf911(..._0x110b7d);
                return '\x1b[' + (_0x466ff1 + _0x3ca4ca) + 'm';
            }, _0x433016 = (_0x502ab4, _0x14424d) => (..._0x373dbb) => {
                const _0x5de9a2 = _0x502ab4(..._0x373dbb);
                return '\x1b[' + (0x26 + _0x14424d) + ';5;' + _0x5de9a2 + 'm';
            }, _0x594c96 = (_0x50ca42, _0x238c22) => (..._0x2336c4) => {
                const _0x5c890 = _0x50ca42(..._0x2336c4);
                return '\x1b[' + (0x26 + _0x238c22) + ';2;' + _0x5c890[0x0] + ';' + _0x5c890[0x1] + ';' + _0x5c890[0x2] + 'm';
            }, _0x21eec1 = _0xcd89ab => _0xcd89ab, _0x54021f = (_0x5c9fee, _0x19f47c, _0x3b8555) => [
                _0x5c9fee,
                _0x19f47c,
                _0x3b8555
            ], _0xb8a62a = (_0x1efa87, _0x35ab12, _0x3e12ef) => {
                Object['defineProperty'](_0x1efa87, _0x35ab12, {
                    'get': () => {
                        const _0x3ddbf3 = _0x3e12ef();
                        return Object['defineProperty'](_0x1efa87, _0x35ab12, {
                            'value': _0x3ddbf3,
                            'enumerable': !![],
                            'configurable': !![]
                        }), _0x3ddbf3;
                    },
                    'enumerable': !![],
                    'configurable': !![]
                });
            };
        let _0x1a7f70;
        const _0xe1e3f4 = (_0x58fde2, _0x3ee709, _0x37835a, _0x11c651) => {
            _0x1a7f70 === undefined && (_0x1a7f70 = _0x3ce7c6(0xad3));
            const _0x4c60b7 = _0x11c651 ? 0xa : 0x0, _0x2a3b2a = {};
            for (const [_0x501250, _0x524da8] of Object['entries'](_0x1a7f70)) {
                const _0x1db249 = _0x501250 === 'ansi16' ? 'ansi' : _0x501250;
                if (_0x501250 === _0x3ee709)
                    _0x2a3b2a[_0x1db249] = _0x58fde2(_0x37835a, _0x4c60b7);
                else
                    typeof _0x524da8 === 'object' && (_0x2a3b2a[_0x1db249] = _0x58fde2(_0x524da8[_0x3ee709], _0x4c60b7));
            }
            return _0x2a3b2a;
        };
        function _0x195d86() {
            const _0x30339d = new Map(), _0x4e76b1 = {
                    'modifier': {
                        'reset': [
                            0x0,
                            0x0
                        ],
                        'bold': [
                            0x1,
                            0x16
                        ],
                        'dim': [
                            0x2,
                            0x16
                        ],
                        'italic': [
                            0x3,
                            0x17
                        ],
                        'underline': [
                            0x4,
                            0x18
                        ],
                        'inverse': [
                            0x7,
                            0x1b
                        ],
                        'hidden': [
                            0x8,
                            0x1c
                        ],
                        'strikethrough': [
                            0x9,
                            0x1d
                        ]
                    },
                    'color': {
                        'black': [
                            0x1e,
                            0x27
                        ],
                        'red': [
                            0x1f,
                            0x27
                        ],
                        'green': [
                            0x20,
                            0x27
                        ],
                        'yellow': [
                            0x21,
                            0x27
                        ],
                        'blue': [
                            0x22,
                            0x27
                        ],
                        'magenta': [
                            0x23,
                            0x27
                        ],
                        'cyan': [
                            0x24,
                            0x27
                        ],
                        'white': [
                            0x25,
                            0x27
                        ],
                        'blackBright': [
                            0x5a,
                            0x27
                        ],
                        'redBright': [
                            0x5b,
                            0x27
                        ],
                        'greenBright': [
                            0x5c,
                            0x27
                        ],
                        'yellowBright': [
                            0x5d,
                            0x27
                        ],
                        'blueBright': [
                            0x5e,
                            0x27
                        ],
                        'magentaBright': [
                            0x5f,
                            0x27
                        ],
                        'cyanBright': [
                            0x60,
                            0x27
                        ],
                        'whiteBright': [
                            0x61,
                            0x27
                        ]
                    },
                    'bgColor': {
                        'bgBlack': [
                            0x28,
                            0x31
                        ],
                        'bgRed': [
                            0x29,
                            0x31
                        ],
                        'bgGreen': [
                            0x2a,
                            0x31
                        ],
                        'bgYellow': [
                            0x2b,
                            0x31
                        ],
                        'bgBlue': [
                            0x2c,
                            0x31
                        ],
                        'bgMagenta': [
                            0x2d,
                            0x31
                        ],
                        'bgCyan': [
                            0x2e,
                            0x31
                        ],
                        'bgWhite': [
                            0x2f,
                            0x31
                        ],
                        'bgBlackBright': [
                            0x64,
                            0x31
                        ],
                        'bgRedBright': [
                            0x65,
                            0x31
                        ],
                        'bgGreenBright': [
                            0x66,
                            0x31
                        ],
                        'bgYellowBright': [
                            0x67,
                            0x31
                        ],
                        'bgBlueBright': [
                            0x68,
                            0x31
                        ],
                        'bgMagentaBright': [
                            0x69,
                            0x31
                        ],
                        'bgCyanBright': [
                            0x6a,
                            0x31
                        ],
                        'bgWhiteBright': [
                            0x6b,
                            0x31
                        ]
                    }
                };
            _0x4e76b1['color']['gray'] = _0x4e76b1['color']['blackBright'], _0x4e76b1['bgColor']['bgGray'] = _0x4e76b1['bgColor']['bgBlackBright'], _0x4e76b1['color']['grey'] = _0x4e76b1['color']['blackBright'], _0x4e76b1['bgColor']['bgGrey'] = _0x4e76b1['bgColor']['bgBlackBright'];
            for (const [_0xc91e9b, _0x25625a] of Object['entries'](_0x4e76b1)) {
                for (const [_0xeaca98, _0x1c46bd] of Object['entries'](_0x25625a)) {
                    _0x4e76b1[_0xeaca98] = {
                        'open': '\x1b[' + _0x1c46bd[0x0] + 'm',
                        'close': '\x1b[' + _0x1c46bd[0x1] + 'm'
                    }, _0x25625a[_0xeaca98] = _0x4e76b1[_0xeaca98], _0x30339d['set'](_0x1c46bd[0x0], _0x1c46bd[0x1]);
                }
                Object['defineProperty'](_0x4e76b1, _0xc91e9b, {
                    'value': _0x25625a,
                    'enumerable': ![]
                });
            }
            return Object['defineProperty'](_0x4e76b1, 'codes', {
                'value': _0x30339d,
                'enumerable': ![]
            }), _0x4e76b1['color']['close'] = '\x1b[39m', _0x4e76b1['bgColor']['close'] = '\x1b[49m', _0xb8a62a(_0x4e76b1['color'], 'ansi', () => _0xe1e3f4(_0xc3bbc5, 'ansi16', _0x21eec1, ![])), _0xb8a62a(_0x4e76b1['color'], 'ansi256', () => _0xe1e3f4(_0x433016, 'ansi256', _0x21eec1, ![])), _0xb8a62a(_0x4e76b1['color'], 'ansi16m', () => _0xe1e3f4(_0x594c96, 'rgb', _0x54021f, ![])), _0xb8a62a(_0x4e76b1['bgColor'], 'ansi', () => _0xe1e3f4(_0xc3bbc5, 'ansi16', _0x21eec1, !![])), _0xb8a62a(_0x4e76b1['bgColor'], 'ansi256', () => _0xe1e3f4(_0x433016, 'ansi256', _0x21eec1, !![])), _0xb8a62a(_0x4e76b1['bgColor'], 'ansi16m', () => _0xe1e3f4(_0x594c96, 'rgb', _0x54021f, !![])), _0x4e76b1;
        }
        Object['defineProperty'](_0xcf0f25, 'exports', {
            'enumerable': !![],
            'get': _0x195d86
        });
    },
    0x1145c: (_0x50fee3, _0x158b38, _0x4e2487) => {
        'use strict';
        const _0x348aa7 = _0x4e2487(0x7d39), {
                stdout: _0x521f96,
                stderr: _0x3600d1
            } = _0x4e2487(0x9583), {
                stringReplaceAll: _0x3d690e,
                stringEncaseCRLFWithFirstIndex: _0x17a28f
            } = _0x4e2487(0x2d), {isArray: _0x3d961d} = Array, _0x2c1228 = [
                'ansi',
                'ansi',
                'ansi256',
                'ansi16m'
            ], _0x3a3f73 = Object['create'](null), _0x4e3526 = (_0x2dfd3d, _0x2e0deb = {}) => {
                if (_0x2e0deb['level'] && !(Number['isInteger'](_0x2e0deb['level']) && _0x2e0deb['level'] >= 0x0 && _0x2e0deb['level'] <= 0x3))
                    throw new Error('The\x20`level`\x20option\x20should\x20be\x20an\x20integer\x20from\x200\x20to\x203');
                const _0x4248f4 = _0x521f96 ? _0x521f96['level'] : 0x0;
                _0x2dfd3d['level'] = _0x2e0deb['level'] === undefined ? _0x4248f4 : _0x2e0deb['level'];
            };
        class _0x5336b1 {
            constructor(_0x222618) {
                return _0x4e3e84(_0x222618);
            }
        }
        const _0x4e3e84 = _0x19d124 => {
            const _0x4e9ffd = {};
            return _0x4e3526(_0x4e9ffd, _0x19d124), _0x4e9ffd['template'] = (..._0x357cf4) => _0x29d6d0(_0x4e9ffd['template'], ..._0x357cf4), Object['setPrototypeOf'](_0x4e9ffd, _0x3029ed['prototype']), Object['setPrototypeOf'](_0x4e9ffd['template'], _0x4e9ffd), _0x4e9ffd['template']['constructor'] = () => {
                throw new Error('`chalk.constructor()`\x20is\x20deprecated.\x20Use\x20`new\x20chalk.Instance()`\x20instead.');
            }, _0x4e9ffd['template']['Instance'] = _0x5336b1, _0x4e9ffd['template'];
        };
        function _0x3029ed(_0x108f3e) {
            return _0x4e3e84(_0x108f3e);
        }
        for (const [_0x36da53, _0x3ab68e] of Object['entries'](_0x348aa7)) {
            _0x3a3f73[_0x36da53] = {
                'get'() {
                    const _0x1a0351 = _0x219f69(this, _0x4dadd4(_0x3ab68e['open'], _0x3ab68e['close'], this['_styler']), this['_isEmpty']);
                    return Object['defineProperty'](this, _0x36da53, { 'value': _0x1a0351 }), _0x1a0351;
                }
            };
        }
        _0x3a3f73['visible'] = {
            'get'() {
                const _0x4bd275 = _0x219f69(this, this['_styler'], !![]);
                return Object['defineProperty'](this, 'visible', { 'value': _0x4bd275 }), _0x4bd275;
            }
        };
        const _0xfc14de = [
            'rgb',
            'hex',
            'keyword',
            'hsl',
            'hsv',
            'hwb',
            'ansi',
            'ansi256'
        ];
        for (const _0x1ed613 of _0xfc14de) {
            _0x3a3f73[_0x1ed613] = {
                'get'() {
                    const {level: _0x30421c} = this;
                    return function (..._0x2c4fb4) {
                        const _0x425e0d = _0x4dadd4(_0x348aa7['color'][_0x2c1228[_0x30421c]][_0x1ed613](..._0x2c4fb4), _0x348aa7['color']['close'], this['_styler']);
                        return _0x219f69(this, _0x425e0d, this['_isEmpty']);
                    };
                }
            };
        }
        for (const _0x48f266 of _0xfc14de) {
            const _0x10519c = 'bg' + _0x48f266[0x0]['toUpperCase']() + _0x48f266['slice'](0x1);
            _0x3a3f73[_0x10519c] = {
                'get'() {
                    const {level: _0x2d6c24} = this;
                    return function (..._0x1a3627) {
                        const _0x40c97e = _0x4dadd4(_0x348aa7['bgColor'][_0x2c1228[_0x2d6c24]][_0x48f266](..._0x1a3627), _0x348aa7['bgColor']['close'], this['_styler']);
                        return _0x219f69(this, _0x40c97e, this['_isEmpty']);
                    };
                }
            };
        }
        const _0x25b2a0 = Object['defineProperties'](() => {
            }, {
                ..._0x3a3f73,
                'level': {
                    'enumerable': !![],
                    'get'() {
                        return this['_generator']['level'];
                    },
                    'set'(_0x4972aa) {
                        this['_generator']['level'] = _0x4972aa;
                    }
                }
            }), _0x4dadd4 = (_0x347d61, _0x4705e4, _0x3b36df) => {
                let _0x5e296a, _0x18b9d0;
                return _0x3b36df === undefined ? (_0x5e296a = _0x347d61, _0x18b9d0 = _0x4705e4) : (_0x5e296a = _0x3b36df['openAll'] + _0x347d61, _0x18b9d0 = _0x4705e4 + _0x3b36df['closeAll']), {
                    'open': _0x347d61,
                    'close': _0x4705e4,
                    'openAll': _0x5e296a,
                    'closeAll': _0x18b9d0,
                    'parent': _0x3b36df
                };
            }, _0x219f69 = (_0x450c51, _0x41ba37, _0xdb21a7) => {
                const _0x3b7fe8 = (..._0x584f25) => {
                    if (_0x3d961d(_0x584f25[0x0]) && _0x3d961d(_0x584f25[0x0]['raw']))
                        return _0x59e418(_0x3b7fe8, _0x29d6d0(_0x3b7fe8, ..._0x584f25));
                    return _0x59e418(_0x3b7fe8, _0x584f25['length'] === 0x1 ? '' + _0x584f25[0x0] : _0x584f25['join']('\x20'));
                };
                return Object['setPrototypeOf'](_0x3b7fe8, _0x25b2a0), _0x3b7fe8['_generator'] = _0x450c51, _0x3b7fe8['_styler'] = _0x41ba37, _0x3b7fe8['_isEmpty'] = _0xdb21a7, _0x3b7fe8;
            }, _0x59e418 = (_0x1dd382, _0x52eaac) => {
                if (_0x1dd382['level'] <= 0x0 || !_0x52eaac)
                    return _0x1dd382['_isEmpty'] ? '' : _0x52eaac;
                let _0x2ed67d = _0x1dd382['_styler'];
                if (_0x2ed67d === undefined)
                    return _0x52eaac;
                const {
                    openAll: _0x257a9a,
                    closeAll: _0x47e604
                } = _0x2ed67d;
                if (_0x52eaac['indexOf']('\x1b') !== -0x1)
                    while (_0x2ed67d !== undefined) {
                        _0x52eaac = _0x3d690e(_0x52eaac, _0x2ed67d['close'], _0x2ed67d['open']), _0x2ed67d = _0x2ed67d['parent'];
                    }
                const _0x2d1af1 = _0x52eaac['indexOf']('\x0a');
                return _0x2d1af1 !== -0x1 && (_0x52eaac = _0x17a28f(_0x52eaac, _0x47e604, _0x257a9a, _0x2d1af1)), _0x257a9a + _0x52eaac + _0x47e604;
            };
        let _0x4701e7;
        const _0x29d6d0 = (_0x3cc8c5, ..._0xd74a72) => {
            const [_0x1f702b] = _0xd74a72;
            if (!_0x3d961d(_0x1f702b) || !_0x3d961d(_0x1f702b['raw']))
                return _0xd74a72['join']('\x20');
            const _0x5eb1fa = _0xd74a72['slice'](0x1), _0x3296ae = [_0x1f702b['raw'][0x0]];
            for (let _0x276c90 = 0x1; _0x276c90 < _0x1f702b['length']; _0x276c90++) {
                _0x3296ae['push'](String(_0x5eb1fa[_0x276c90 - 0x1])['replace'](/[{}\\]/g, '\x5c$&'), String(_0x1f702b['raw'][_0x276c90]));
            }
            return _0x4701e7 === undefined && (_0x4701e7 = _0x4e2487(0xee76)), _0x4701e7(_0x3cc8c5, _0x3296ae['join'](''));
        };
        Object['defineProperties'](_0x3029ed['prototype'], _0x3a3f73);
        const _0x24e1c1 = _0x3029ed();
        _0x24e1c1['supportsColor'] = _0x521f96, _0x24e1c1['stderr'] = _0x3029ed({ 'level': _0x3600d1 ? _0x3600d1['level'] : 0x0 }), _0x24e1c1['stderr']['supportsColor'] = _0x3600d1, _0x50fee3['exports'] = _0x24e1c1;
    },
    0xee76: _0x45b416 => {
        'use strict';
        const _0x119405 = /(?:\\(u(?:[a-f\d]{4}|\{[a-f\d]{1,6}\})|x[a-f\d]{2}|.))|(?:\{(~)?(\w+(?:\([^)]*\))?(?:\.\w+(?:\([^)]*\))?)*)(?:[ \t]|(?=\r?\n)))|(\})|((?:.|[\r\n\f])+?)/gi, _0x24172d = /(?:^|\.)(\w+)(?:\(([^)]*)\))?/g, _0x3cbbfb = /^(['"])((?:\\.|(?!\1)[^\\])*)\1$/, _0x205780 = /\\(u(?:[a-f\d]{4}|{[a-f\d]{1,6}})|x[a-f\d]{2}|.)|([^\\])/gi, _0x346298 = new Map([
                [
                    'n',
                    '\x0a'
                ],
                [
                    'r',
                    '\x0d'
                ],
                [
                    't',
                    '\x09'
                ],
                [
                    'b',
                    '\x08'
                ],
                [
                    'f',
                    '\x0c'
                ],
                [
                    'v',
                    '\x0b'
                ],
                [
                    '0',
                    '\x00'
                ],
                [
                    '\x5c',
                    '\x5c'
                ],
                [
                    'e',
                    '\x1b'
                ],
                [
                    'a',
                    '\x07'
                ]
            ]);
        function _0x4c39b0(_0x5c5761) {
            const _0x2ecf12 = _0x5c5761[0x0] === 'u', _0x53c9f2 = _0x5c5761[0x1] === '{';
            if (_0x2ecf12 && !_0x53c9f2 && _0x5c5761['length'] === 0x5 || _0x5c5761[0x0] === 'x' && _0x5c5761['length'] === 0x3)
                return String['fromCharCode'](parseInt(_0x5c5761['slice'](0x1), 0x10));
            if (_0x2ecf12 && _0x53c9f2)
                return String['fromCodePoint'](parseInt(_0x5c5761['slice'](0x2, -0x1), 0x10));
            return _0x346298['get'](_0x5c5761) || _0x5c5761;
        }
        function _0x188cad(_0x3e1950, _0xfbec5e) {
            const _0x3062c8 = [], _0x3f23d1 = _0xfbec5e['trim']()['split'](/\s*,\s*/g);
            let _0x1ebafe;
            for (const _0x57d6b5 of _0x3f23d1) {
                const _0xfad857 = Number(_0x57d6b5);
                if (!Number['isNaN'](_0xfad857))
                    _0x3062c8['push'](_0xfad857);
                else {
                    if (_0x1ebafe = _0x57d6b5['match'](_0x3cbbfb))
                        _0x3062c8['push'](_0x1ebafe[0x2]['replace'](_0x205780, (_0x421c64, _0x5be0bc, _0x5cc655) => _0x5be0bc ? _0x4c39b0(_0x5be0bc) : _0x5cc655));
                    else
                        throw new Error('Invalid\x20Chalk\x20template\x20style\x20argument:\x20' + _0x57d6b5 + '\x20(in\x20style\x20\x27' + _0x3e1950 + '\x27)');
                }
            }
            return _0x3062c8;
        }
        function _0x1e56db(_0x144d9a) {
            _0x24172d['lastIndex'] = 0x0;
            const _0x3a78e8 = [];
            let _0x37fc32;
            while ((_0x37fc32 = _0x24172d['exec'](_0x144d9a)) !== null) {
                const _0x2f078e = _0x37fc32[0x1];
                if (_0x37fc32[0x2]) {
                    const _0x28e33d = _0x188cad(_0x2f078e, _0x37fc32[0x2]);
                    _0x3a78e8['push']([_0x2f078e]['concat'](_0x28e33d));
                } else
                    _0x3a78e8['push']([_0x2f078e]);
            }
            return _0x3a78e8;
        }
        function _0x5c9a2b(_0x162005, _0x2f32c7) {
            const _0x582142 = {};
            for (const _0xc573ae of _0x2f32c7) {
                for (const _0x1811a2 of _0xc573ae['styles']) {
                    _0x582142[_0x1811a2[0x0]] = _0xc573ae['inverse'] ? null : _0x1811a2['slice'](0x1);
                }
            }
            let _0x13b030 = _0x162005;
            for (const [_0xc969ef, _0x40f8dd] of Object['entries'](_0x582142)) {
                if (!Array['isArray'](_0x40f8dd))
                    continue;
                if (!(_0xc969ef in _0x13b030))
                    throw new Error('Unknown\x20Chalk\x20style:\x20' + _0xc969ef);
                _0x13b030 = _0x40f8dd['length'] > 0x0 ? _0x13b030[_0xc969ef](..._0x40f8dd) : _0x13b030[_0xc969ef];
            }
            return _0x13b030;
        }
        _0x45b416['exports'] = (_0x324ab4, _0x1f75eb) => {
            const _0x5aef21 = [], _0x5cbceb = [];
            let _0x35c779 = [];
            _0x1f75eb['replace'](_0x119405, (_0x249ba8, _0x528afc, _0x42fc50, _0x45dc5f, _0x18646e, _0x18ce97) => {
                if (_0x528afc)
                    _0x35c779['push'](_0x4c39b0(_0x528afc));
                else {
                    if (_0x45dc5f) {
                        const _0x5de465 = _0x35c779['join']('');
                        _0x35c779 = [], _0x5cbceb['push'](_0x5aef21['length'] === 0x0 ? _0x5de465 : _0x5c9a2b(_0x324ab4, _0x5aef21)(_0x5de465)), _0x5aef21['push']({
                            'inverse': _0x42fc50,
                            'styles': _0x1e56db(_0x45dc5f)
                        });
                    } else {
                        if (_0x18646e) {
                            if (_0x5aef21['length'] === 0x0)
                                throw new Error('Found\x20extraneous\x20}\x20in\x20Chalk\x20template\x20literal');
                            _0x5cbceb['push'](_0x5c9a2b(_0x324ab4, _0x5aef21)(_0x35c779['join'](''))), _0x35c779 = [], _0x5aef21['pop']();
                        } else
                            _0x35c779['push'](_0x18ce97);
                    }
                }
            }), _0x5cbceb['push'](_0x35c779['join'](''));
            if (_0x5aef21['length'] > 0x0) {
                const _0x281403 = 'Chalk\x20template\x20literal\x20is\x20missing\x20' + _0x5aef21['length'] + '\x20closing\x20bracket' + (_0x5aef21['length'] === 0x1 ? '' : 's') + '\x20(`}`)';
                throw new Error(_0x281403);
            }
            return _0x5cbceb['join']('');
        };
    },
    0x2d: _0x1692b3 => {
        'use strict';
        const _0x2dc832 = (_0x1a9b49, _0xecb558, _0x56f087) => {
                let _0x196a07 = _0x1a9b49['indexOf'](_0xecb558);
                if (_0x196a07 === -0x1)
                    return _0x1a9b49;
                const _0x506aef = _0xecb558['length'];
                let _0xd0d53f = 0x0, _0x1b7e7d = '';
                do {
                    _0x1b7e7d += _0x1a9b49['substr'](_0xd0d53f, _0x196a07 - _0xd0d53f) + _0xecb558 + _0x56f087, _0xd0d53f = _0x196a07 + _0x506aef, _0x196a07 = _0x1a9b49['indexOf'](_0xecb558, _0xd0d53f);
                } while (_0x196a07 !== -0x1);
                return _0x1b7e7d += _0x1a9b49['substr'](_0xd0d53f), _0x1b7e7d;
            }, _0x47b9eb = (_0x2f4671, _0x356433, _0xc11a7a, _0x1aab85) => {
                let _0x285e3 = 0x0, _0x459ccb = '';
                do {
                    const _0x2dea2a = _0x2f4671[_0x1aab85 - 0x1] === '\x0d';
                    _0x459ccb += _0x2f4671['substr'](_0x285e3, (_0x2dea2a ? _0x1aab85 - 0x1 : _0x1aab85) - _0x285e3) + _0x356433 + (_0x2dea2a ? '\x0d\x0a' : '\x0a') + _0xc11a7a, _0x285e3 = _0x1aab85 + 0x1, _0x1aab85 = _0x2f4671['indexOf']('\x0a', _0x285e3);
                } while (_0x1aab85 !== -0x1);
                return _0x459ccb += _0x2f4671['substr'](_0x285e3), _0x459ccb;
            };
        _0x1692b3['exports'] = {
            'stringReplaceAll': _0x2dc832,
            'stringEncaseCRLFWithFirstIndex': _0x47b9eb
        };
    },
    0x13ed: (_0x578fe3, _0x4ed454, _0x56f316) => {
        const _0x10de59 = _0x56f316(0x1845b), _0x345464 = {};
        for (const _0xc43409 of Object['keys'](_0x10de59)) {
            _0x345464[_0x10de59[_0xc43409]] = _0xc43409;
        }
        const _0x464970 = {
            'rgb': {
                'channels': 0x3,
                'labels': 'rgb'
            },
            'hsl': {
                'channels': 0x3,
                'labels': 'hsl'
            },
            'hsv': {
                'channels': 0x3,
                'labels': 'hsv'
            },
            'hwb': {
                'channels': 0x3,
                'labels': 'hwb'
            },
            'cmyk': {
                'channels': 0x4,
                'labels': 'cmyk'
            },
            'xyz': {
                'channels': 0x3,
                'labels': 'xyz'
            },
            'lab': {
                'channels': 0x3,
                'labels': 'lab'
            },
            'lch': {
                'channels': 0x3,
                'labels': 'lch'
            },
            'hex': {
                'channels': 0x1,
                'labels': ['hex']
            },
            'keyword': {
                'channels': 0x1,
                'labels': ['keyword']
            },
            'ansi16': {
                'channels': 0x1,
                'labels': ['ansi16']
            },
            'ansi256': {
                'channels': 0x1,
                'labels': ['ansi256']
            },
            'hcg': {
                'channels': 0x3,
                'labels': [
                    'h',
                    'c',
                    'g'
                ]
            },
            'apple': {
                'channels': 0x3,
                'labels': [
                    'r16',
                    'g16',
                    'b16'
                ]
            },
            'gray': {
                'channels': 0x1,
                'labels': ['gray']
            }
        };
        _0x578fe3['exports'] = _0x464970;
        for (const _0x4d3612 of Object['keys'](_0x464970)) {
            if (!('channels' in _0x464970[_0x4d3612]))
                throw new Error('missing\x20channels\x20property:\x20' + _0x4d3612);
            if (!('labels' in _0x464970[_0x4d3612]))
                throw new Error('missing\x20channel\x20labels\x20property:\x20' + _0x4d3612);
            if (_0x464970[_0x4d3612]['labels']['length'] !== _0x464970[_0x4d3612]['channels'])
                throw new Error('channel\x20and\x20label\x20counts\x20mismatch:\x20' + _0x4d3612);
            const {
                channels: _0x59a391,
                labels: _0x2117a0
            } = _0x464970[_0x4d3612];
            delete _0x464970[_0x4d3612]['channels'], delete _0x464970[_0x4d3612]['labels'], Object['defineProperty'](_0x464970[_0x4d3612], 'channels', { 'value': _0x59a391 }), Object['defineProperty'](_0x464970[_0x4d3612], 'labels', { 'value': _0x2117a0 });
        }
        _0x464970['rgb']['hsl'] = function (_0x3688d4) {
            const _0x2665f4 = _0x3688d4[0x0] / 0xff, _0x135c8d = _0x3688d4[0x1] / 0xff, _0x40a7bd = _0x3688d4[0x2] / 0xff, _0x303ab5 = Math['min'](_0x2665f4, _0x135c8d, _0x40a7bd), _0x3a981b = Math['max'](_0x2665f4, _0x135c8d, _0x40a7bd), _0x2a6495 = _0x3a981b - _0x303ab5;
            let _0x159086, _0xa23009;
            if (_0x3a981b === _0x303ab5)
                _0x159086 = 0x0;
            else {
                if (_0x2665f4 === _0x3a981b)
                    _0x159086 = (_0x135c8d - _0x40a7bd) / _0x2a6495;
                else {
                    if (_0x135c8d === _0x3a981b)
                        _0x159086 = 0x2 + (_0x40a7bd - _0x2665f4) / _0x2a6495;
                    else
                        _0x40a7bd === _0x3a981b && (_0x159086 = 0x4 + (_0x2665f4 - _0x135c8d) / _0x2a6495);
                }
            }
            _0x159086 = Math['min'](_0x159086 * 0x3c, 0x168);
            _0x159086 < 0x0 && (_0x159086 += 0x168);
            const _0x18d683 = (_0x303ab5 + _0x3a981b) / 0x2;
            if (_0x3a981b === _0x303ab5)
                _0xa23009 = 0x0;
            else
                _0x18d683 <= 0.5 ? _0xa23009 = _0x2a6495 / (_0x3a981b + _0x303ab5) : _0xa23009 = _0x2a6495 / (0x2 - _0x3a981b - _0x303ab5);
            return [
                _0x159086,
                _0xa23009 * 0x64,
                _0x18d683 * 0x64
            ];
        }, _0x464970['rgb']['hsv'] = function (_0x46795a) {
            let _0x5a68f3, _0x3b4547, _0x57ea2d, _0x309f61, _0x224c10;
            const _0x19d48e = _0x46795a[0x0] / 0xff, _0x1dc1b3 = _0x46795a[0x1] / 0xff, _0x2c14b4 = _0x46795a[0x2] / 0xff, _0x4c6d13 = Math['max'](_0x19d48e, _0x1dc1b3, _0x2c14b4), _0x355209 = _0x4c6d13 - Math['min'](_0x19d48e, _0x1dc1b3, _0x2c14b4), _0x25a0f5 = function (_0x87b570) {
                    return (_0x4c6d13 - _0x87b570) / 0x6 / _0x355209 + 0x1 / 0x2;
                };
            if (_0x355209 === 0x0)
                _0x309f61 = 0x0, _0x224c10 = 0x0;
            else {
                _0x224c10 = _0x355209 / _0x4c6d13, _0x5a68f3 = _0x25a0f5(_0x19d48e), _0x3b4547 = _0x25a0f5(_0x1dc1b3), _0x57ea2d = _0x25a0f5(_0x2c14b4);
                if (_0x19d48e === _0x4c6d13)
                    _0x309f61 = _0x57ea2d - _0x3b4547;
                else {
                    if (_0x1dc1b3 === _0x4c6d13)
                        _0x309f61 = 0x1 / 0x3 + _0x5a68f3 - _0x57ea2d;
                    else
                        _0x2c14b4 === _0x4c6d13 && (_0x309f61 = 0x2 / 0x3 + _0x3b4547 - _0x5a68f3);
                }
                if (_0x309f61 < 0x0)
                    _0x309f61 += 0x1;
                else
                    _0x309f61 > 0x1 && (_0x309f61 -= 0x1);
            }
            return [
                _0x309f61 * 0x168,
                _0x224c10 * 0x64,
                _0x4c6d13 * 0x64
            ];
        }, _0x464970['rgb']['hwb'] = function (_0x53503c) {
            const _0xe7c348 = _0x53503c[0x0], _0x4eed03 = _0x53503c[0x1];
            let _0x13df98 = _0x53503c[0x2];
            const _0x291e11 = _0x464970['rgb']['hsl'](_0x53503c)[0x0], _0x304243 = 0x1 / 0xff * Math['min'](_0xe7c348, Math['min'](_0x4eed03, _0x13df98));
            return _0x13df98 = 0x1 - 0x1 / 0xff * Math['max'](_0xe7c348, Math['max'](_0x4eed03, _0x13df98)), [
                _0x291e11,
                _0x304243 * 0x64,
                _0x13df98 * 0x64
            ];
        }, _0x464970['rgb']['cmyk'] = function (_0x30be46) {
            const _0x5ec4a9 = _0x30be46[0x0] / 0xff, _0x59b9dc = _0x30be46[0x1] / 0xff, _0x2f61d6 = _0x30be46[0x2] / 0xff, _0x8d150b = Math['min'](0x1 - _0x5ec4a9, 0x1 - _0x59b9dc, 0x1 - _0x2f61d6), _0x27c4fc = (0x1 - _0x5ec4a9 - _0x8d150b) / (0x1 - _0x8d150b) || 0x0, _0x333a70 = (0x1 - _0x59b9dc - _0x8d150b) / (0x1 - _0x8d150b) || 0x0, _0x206029 = (0x1 - _0x2f61d6 - _0x8d150b) / (0x1 - _0x8d150b) || 0x0;
            return [
                _0x27c4fc * 0x64,
                _0x333a70 * 0x64,
                _0x206029 * 0x64,
                _0x8d150b * 0x64
            ];
        };
        function _0x4f7b97(_0x475850, _0x521a1e) {
            return (_0x475850[0x0] - _0x521a1e[0x0]) ** 0x2 + (_0x475850[0x1] - _0x521a1e[0x1]) ** 0x2 + (_0x475850[0x2] - _0x521a1e[0x2]) ** 0x2;
        }
        _0x464970['rgb']['keyword'] = function (_0x417c0b) {
            const _0x531edb = _0x345464[_0x417c0b];
            if (_0x531edb)
                return _0x531edb;
            let _0x55c3e7 = Infinity, _0x3a3c85;
            for (const _0x3ed2d2 of Object['keys'](_0x10de59)) {
                const _0x41add1 = _0x10de59[_0x3ed2d2], _0x54626d = _0x4f7b97(_0x417c0b, _0x41add1);
                _0x54626d < _0x55c3e7 && (_0x55c3e7 = _0x54626d, _0x3a3c85 = _0x3ed2d2);
            }
            return _0x3a3c85;
        }, _0x464970['keyword']['rgb'] = function (_0x27655e) {
            return _0x10de59[_0x27655e];
        }, _0x464970['rgb']['xyz'] = function (_0xc2a74b) {
            let _0x4710eb = _0xc2a74b[0x0] / 0xff, _0x2df11c = _0xc2a74b[0x1] / 0xff, _0x2728b4 = _0xc2a74b[0x2] / 0xff;
            _0x4710eb = _0x4710eb > 0.04045 ? ((_0x4710eb + 0.055) / 1.055) ** 2.4 : _0x4710eb / 12.92, _0x2df11c = _0x2df11c > 0.04045 ? ((_0x2df11c + 0.055) / 1.055) ** 2.4 : _0x2df11c / 12.92, _0x2728b4 = _0x2728b4 > 0.04045 ? ((_0x2728b4 + 0.055) / 1.055) ** 2.4 : _0x2728b4 / 12.92;
            const _0x1f79a8 = _0x4710eb * 0.4124 + _0x2df11c * 0.3576 + _0x2728b4 * 0.1805, _0x180c84 = _0x4710eb * 0.2126 + _0x2df11c * 0.7152 + _0x2728b4 * 0.0722, _0x3d21ca = _0x4710eb * 0.0193 + _0x2df11c * 0.1192 + _0x2728b4 * 0.9505;
            return [
                _0x1f79a8 * 0x64,
                _0x180c84 * 0x64,
                _0x3d21ca * 0x64
            ];
        }, _0x464970['rgb']['lab'] = function (_0x17f1e0) {
            const _0x3fe39d = _0x464970['rgb']['xyz'](_0x17f1e0);
            let _0x4f5512 = _0x3fe39d[0x0], _0x104ee7 = _0x3fe39d[0x1], _0x2212b8 = _0x3fe39d[0x2];
            _0x4f5512 /= 95.047, _0x104ee7 /= 0x64, _0x2212b8 /= 108.883, _0x4f5512 = _0x4f5512 > 0.008856 ? _0x4f5512 ** (0x1 / 0x3) : 7.787 * _0x4f5512 + 0x10 / 0x74, _0x104ee7 = _0x104ee7 > 0.008856 ? _0x104ee7 ** (0x1 / 0x3) : 7.787 * _0x104ee7 + 0x10 / 0x74, _0x2212b8 = _0x2212b8 > 0.008856 ? _0x2212b8 ** (0x1 / 0x3) : 7.787 * _0x2212b8 + 0x10 / 0x74;
            const _0xf91c2 = 0x74 * _0x104ee7 - 0x10, _0x1ad392 = 0x1f4 * (_0x4f5512 - _0x104ee7), _0x319ec4 = 0xc8 * (_0x104ee7 - _0x2212b8);
            return [
                _0xf91c2,
                _0x1ad392,
                _0x319ec4
            ];
        }, _0x464970['hsl']['rgb'] = function (_0x563c17) {
            const _0x1f0042 = _0x563c17[0x0] / 0x168, _0x862b48 = _0x563c17[0x1] / 0x64, _0x5e755f = _0x563c17[0x2] / 0x64;
            let _0x26d2f4, _0x43995a, _0x39cc2f;
            if (_0x862b48 === 0x0)
                return _0x39cc2f = _0x5e755f * 0xff, [
                    _0x39cc2f,
                    _0x39cc2f,
                    _0x39cc2f
                ];
            _0x5e755f < 0.5 ? _0x26d2f4 = _0x5e755f * (0x1 + _0x862b48) : _0x26d2f4 = _0x5e755f + _0x862b48 - _0x5e755f * _0x862b48;
            const _0x5b96af = 0x2 * _0x5e755f - _0x26d2f4, _0x3af9a7 = [
                    0x0,
                    0x0,
                    0x0
                ];
            for (let _0x106d33 = 0x0; _0x106d33 < 0x3; _0x106d33++) {
                _0x43995a = _0x1f0042 + 0x1 / 0x3 * -(_0x106d33 - 0x1);
                _0x43995a < 0x0 && _0x43995a++;
                _0x43995a > 0x1 && _0x43995a--;
                if (0x6 * _0x43995a < 0x1)
                    _0x39cc2f = _0x5b96af + (_0x26d2f4 - _0x5b96af) * 0x6 * _0x43995a;
                else {
                    if (0x2 * _0x43995a < 0x1)
                        _0x39cc2f = _0x26d2f4;
                    else
                        0x3 * _0x43995a < 0x2 ? _0x39cc2f = _0x5b96af + (_0x26d2f4 - _0x5b96af) * (0x2 / 0x3 - _0x43995a) * 0x6 : _0x39cc2f = _0x5b96af;
                }
                _0x3af9a7[_0x106d33] = _0x39cc2f * 0xff;
            }
            return _0x3af9a7;
        }, _0x464970['hsl']['hsv'] = function (_0x1468d6) {
            const _0x5bf60a = _0x1468d6[0x0];
            let _0x2ffa8b = _0x1468d6[0x1] / 0x64, _0x24fb52 = _0x1468d6[0x2] / 0x64, _0x2d2fbe = _0x2ffa8b;
            const _0x3295af = Math['max'](_0x24fb52, 0.01);
            _0x24fb52 *= 0x2, _0x2ffa8b *= _0x24fb52 <= 0x1 ? _0x24fb52 : 0x2 - _0x24fb52, _0x2d2fbe *= _0x3295af <= 0x1 ? _0x3295af : 0x2 - _0x3295af;
            const _0x3f4378 = (_0x24fb52 + _0x2ffa8b) / 0x2, _0x4f8551 = _0x24fb52 === 0x0 ? 0x2 * _0x2d2fbe / (_0x3295af + _0x2d2fbe) : 0x2 * _0x2ffa8b / (_0x24fb52 + _0x2ffa8b);
            return [
                _0x5bf60a,
                _0x4f8551 * 0x64,
                _0x3f4378 * 0x64
            ];
        }, _0x464970['hsv']['rgb'] = function (_0x1b7a32) {
            const _0x4bc618 = _0x1b7a32[0x0] / 0x3c, _0x41d015 = _0x1b7a32[0x1] / 0x64;
            let _0xb7f6e9 = _0x1b7a32[0x2] / 0x64;
            const _0x412a4e = Math['floor'](_0x4bc618) % 0x6, _0x44dd63 = _0x4bc618 - Math['floor'](_0x4bc618), _0x405982 = 0xff * _0xb7f6e9 * (0x1 - _0x41d015), _0x1fd6ab = 0xff * _0xb7f6e9 * (0x1 - _0x41d015 * _0x44dd63), _0x1f7f8e = 0xff * _0xb7f6e9 * (0x1 - _0x41d015 * (0x1 - _0x44dd63));
            _0xb7f6e9 *= 0xff;
            switch (_0x412a4e) {
            case 0x0:
                return [
                    _0xb7f6e9,
                    _0x1f7f8e,
                    _0x405982
                ];
            case 0x1:
                return [
                    _0x1fd6ab,
                    _0xb7f6e9,
                    _0x405982
                ];
            case 0x2:
                return [
                    _0x405982,
                    _0xb7f6e9,
                    _0x1f7f8e
                ];
            case 0x3:
                return [
                    _0x405982,
                    _0x1fd6ab,
                    _0xb7f6e9
                ];
            case 0x4:
                return [
                    _0x1f7f8e,
                    _0x405982,
                    _0xb7f6e9
                ];
            case 0x5:
                return [
                    _0xb7f6e9,
                    _0x405982,
                    _0x1fd6ab
                ];
            }
        }, _0x464970['hsv']['hsl'] = function (_0x54ca0a) {
            const _0x4ccdf4 = _0x54ca0a[0x0], _0x20ae05 = _0x54ca0a[0x1] / 0x64, _0x1fad57 = _0x54ca0a[0x2] / 0x64, _0x3f3514 = Math['max'](_0x1fad57, 0.01);
            let _0x461082, _0x8d8c19;
            _0x8d8c19 = (0x2 - _0x20ae05) * _0x1fad57;
            const _0xfd217 = (0x2 - _0x20ae05) * _0x3f3514;
            return _0x461082 = _0x20ae05 * _0x3f3514, _0x461082 /= _0xfd217 <= 0x1 ? _0xfd217 : 0x2 - _0xfd217, _0x461082 = _0x461082 || 0x0, _0x8d8c19 /= 0x2, [
                _0x4ccdf4,
                _0x461082 * 0x64,
                _0x8d8c19 * 0x64
            ];
        }, _0x464970['hwb']['rgb'] = function (_0x2e416c) {
            const _0x116add = _0x2e416c[0x0] / 0x168;
            let _0x13a3ec = _0x2e416c[0x1] / 0x64, _0x4ed150 = _0x2e416c[0x2] / 0x64;
            const _0x394479 = _0x13a3ec + _0x4ed150;
            let _0x206035;
            _0x394479 > 0x1 && (_0x13a3ec /= _0x394479, _0x4ed150 /= _0x394479);
            const _0x192102 = Math['floor'](0x6 * _0x116add), _0x2a6b00 = 0x1 - _0x4ed150;
            _0x206035 = 0x6 * _0x116add - _0x192102;
            (_0x192102 & 0x1) !== 0x0 && (_0x206035 = 0x1 - _0x206035);
            const _0x129fe1 = _0x13a3ec + _0x206035 * (_0x2a6b00 - _0x13a3ec);
            let _0x5eeb31, _0x1f9660, _0x52347d;
            switch (_0x192102) {
            default:
            case 0x6:
            case 0x0:
                _0x5eeb31 = _0x2a6b00, _0x1f9660 = _0x129fe1, _0x52347d = _0x13a3ec;
                break;
            case 0x1:
                _0x5eeb31 = _0x129fe1, _0x1f9660 = _0x2a6b00, _0x52347d = _0x13a3ec;
                break;
            case 0x2:
                _0x5eeb31 = _0x13a3ec, _0x1f9660 = _0x2a6b00, _0x52347d = _0x129fe1;
                break;
            case 0x3:
                _0x5eeb31 = _0x13a3ec, _0x1f9660 = _0x129fe1, _0x52347d = _0x2a6b00;
                break;
            case 0x4:
                _0x5eeb31 = _0x129fe1, _0x1f9660 = _0x13a3ec, _0x52347d = _0x2a6b00;
                break;
            case 0x5:
                _0x5eeb31 = _0x2a6b00, _0x1f9660 = _0x13a3ec, _0x52347d = _0x129fe1;
                break;
            }
            return [
                _0x5eeb31 * 0xff,
                _0x1f9660 * 0xff,
                _0x52347d * 0xff
            ];
        }, _0x464970['cmyk']['rgb'] = function (_0x5c3e0b) {
            const _0x30fbb6 = _0x5c3e0b[0x0] / 0x64, _0x57f78a = _0x5c3e0b[0x1] / 0x64, _0xd52408 = _0x5c3e0b[0x2] / 0x64, _0x20be31 = _0x5c3e0b[0x3] / 0x64, _0x37b460 = 0x1 - Math['min'](0x1, _0x30fbb6 * (0x1 - _0x20be31) + _0x20be31), _0x2c1262 = 0x1 - Math['min'](0x1, _0x57f78a * (0x1 - _0x20be31) + _0x20be31), _0xbbe518 = 0x1 - Math['min'](0x1, _0xd52408 * (0x1 - _0x20be31) + _0x20be31);
            return [
                _0x37b460 * 0xff,
                _0x2c1262 * 0xff,
                _0xbbe518 * 0xff
            ];
        }, _0x464970['xyz']['rgb'] = function (_0x5cae15) {
            const _0x251d9d = _0x5cae15[0x0] / 0x64, _0x342ef1 = _0x5cae15[0x1] / 0x64, _0x22081d = _0x5cae15[0x2] / 0x64;
            let _0x4cb075, _0x99d5fd, _0x4c3b89;
            return _0x4cb075 = _0x251d9d * 3.2406 + _0x342ef1 * -1.5372 + _0x22081d * -0.4986, _0x99d5fd = _0x251d9d * -0.9689 + _0x342ef1 * 1.8758 + _0x22081d * 0.0415, _0x4c3b89 = _0x251d9d * 0.0557 + _0x342ef1 * -0.204 + _0x22081d * 1.057, _0x4cb075 = _0x4cb075 > 0.0031308 ? 1.055 * _0x4cb075 ** (0x1 / 2.4) - 0.055 : _0x4cb075 * 12.92, _0x99d5fd = _0x99d5fd > 0.0031308 ? 1.055 * _0x99d5fd ** (0x1 / 2.4) - 0.055 : _0x99d5fd * 12.92, _0x4c3b89 = _0x4c3b89 > 0.0031308 ? 1.055 * _0x4c3b89 ** (0x1 / 2.4) - 0.055 : _0x4c3b89 * 12.92, _0x4cb075 = Math['min'](Math['max'](0x0, _0x4cb075), 0x1), _0x99d5fd = Math['min'](Math['max'](0x0, _0x99d5fd), 0x1), _0x4c3b89 = Math['min'](Math['max'](0x0, _0x4c3b89), 0x1), [
                _0x4cb075 * 0xff,
                _0x99d5fd * 0xff,
                _0x4c3b89 * 0xff
            ];
        }, _0x464970['xyz']['lab'] = function (_0x3c3eff) {
            let _0x56ef06 = _0x3c3eff[0x0], _0x559af9 = _0x3c3eff[0x1], _0x43b58d = _0x3c3eff[0x2];
            _0x56ef06 /= 95.047, _0x559af9 /= 0x64, _0x43b58d /= 108.883, _0x56ef06 = _0x56ef06 > 0.008856 ? _0x56ef06 ** (0x1 / 0x3) : 7.787 * _0x56ef06 + 0x10 / 0x74, _0x559af9 = _0x559af9 > 0.008856 ? _0x559af9 ** (0x1 / 0x3) : 7.787 * _0x559af9 + 0x10 / 0x74, _0x43b58d = _0x43b58d > 0.008856 ? _0x43b58d ** (0x1 / 0x3) : 7.787 * _0x43b58d + 0x10 / 0x74;
            const _0x1ab32e = 0x74 * _0x559af9 - 0x10, _0x313814 = 0x1f4 * (_0x56ef06 - _0x559af9), _0x4fb777 = 0xc8 * (_0x559af9 - _0x43b58d);
            return [
                _0x1ab32e,
                _0x313814,
                _0x4fb777
            ];
        }, _0x464970['lab']['xyz'] = function (_0x13d251) {
            const _0xbcbb2a = _0x13d251[0x0], _0x1e463d = _0x13d251[0x1], _0x27c1d6 = _0x13d251[0x2];
            let _0x4d03d4, _0x5127bd, _0x10bc64;
            _0x5127bd = (_0xbcbb2a + 0x10) / 0x74, _0x4d03d4 = _0x1e463d / 0x1f4 + _0x5127bd, _0x10bc64 = _0x5127bd - _0x27c1d6 / 0xc8;
            const _0x57e3a2 = _0x5127bd ** 0x3, _0x2a7e09 = _0x4d03d4 ** 0x3, _0x49cf07 = _0x10bc64 ** 0x3;
            return _0x5127bd = _0x57e3a2 > 0.008856 ? _0x57e3a2 : (_0x5127bd - 0x10 / 0x74) / 7.787, _0x4d03d4 = _0x2a7e09 > 0.008856 ? _0x2a7e09 : (_0x4d03d4 - 0x10 / 0x74) / 7.787, _0x10bc64 = _0x49cf07 > 0.008856 ? _0x49cf07 : (_0x10bc64 - 0x10 / 0x74) / 7.787, _0x4d03d4 *= 95.047, _0x5127bd *= 0x64, _0x10bc64 *= 108.883, [
                _0x4d03d4,
                _0x5127bd,
                _0x10bc64
            ];
        }, _0x464970['lab']['lch'] = function (_0x2f9ecf) {
            const _0x2d2a5a = _0x2f9ecf[0x0], _0x258366 = _0x2f9ecf[0x1], _0x592d63 = _0x2f9ecf[0x2];
            let _0x50dfdd;
            const _0x612c72 = Math['atan2'](_0x592d63, _0x258366);
            _0x50dfdd = _0x612c72 * 0x168 / 0x2 / Math['PI'];
            _0x50dfdd < 0x0 && (_0x50dfdd += 0x168);
            const _0x2ef25a = Math['sqrt'](_0x258366 * _0x258366 + _0x592d63 * _0x592d63);
            return [
                _0x2d2a5a,
                _0x2ef25a,
                _0x50dfdd
            ];
        }, _0x464970['lch']['lab'] = function (_0x8d2617) {
            const _0x5875ad = _0x8d2617[0x0], _0x4fab6a = _0x8d2617[0x1], _0x2a20d4 = _0x8d2617[0x2], _0x1f8013 = _0x2a20d4 / 0x168 * 0x2 * Math['PI'], _0x4f275e = _0x4fab6a * Math['cos'](_0x1f8013), _0x4c0f45 = _0x4fab6a * Math['sin'](_0x1f8013);
            return [
                _0x5875ad,
                _0x4f275e,
                _0x4c0f45
            ];
        }, _0x464970['rgb']['ansi16'] = function (_0x11822f, _0x1be23a = null) {
            const [_0x4e586b, _0x20a6d0, _0x56a512] = _0x11822f;
            let _0x52d2b6 = _0x1be23a === null ? _0x464970['rgb']['hsv'](_0x11822f)[0x2] : _0x1be23a;
            _0x52d2b6 = Math['round'](_0x52d2b6 / 0x32);
            if (_0x52d2b6 === 0x0)
                return 0x1e;
            let _0x4ab4b7 = 0x1e + (Math['round'](_0x56a512 / 0xff) << 0x2 | Math['round'](_0x20a6d0 / 0xff) << 0x1 | Math['round'](_0x4e586b / 0xff));
            return _0x52d2b6 === 0x2 && (_0x4ab4b7 += 0x3c), _0x4ab4b7;
        }, _0x464970['hsv']['ansi16'] = function (_0x414457) {
            return _0x464970['rgb']['ansi16'](_0x464970['hsv']['rgb'](_0x414457), _0x414457[0x2]);
        }, _0x464970['rgb']['ansi256'] = function (_0x145a54) {
            const _0x4b1f2b = _0x145a54[0x0], _0x3eb359 = _0x145a54[0x1], _0x2cab57 = _0x145a54[0x2];
            if (_0x4b1f2b === _0x3eb359 && _0x3eb359 === _0x2cab57) {
                if (_0x4b1f2b < 0x8)
                    return 0x10;
                if (_0x4b1f2b > 0xf8)
                    return 0xe7;
                return Math['round']((_0x4b1f2b - 0x8) / 0xf7 * 0x18) + 0xe8;
            }
            const _0x107977 = 0x10 + 0x24 * Math['round'](_0x4b1f2b / 0xff * 0x5) + 0x6 * Math['round'](_0x3eb359 / 0xff * 0x5) + Math['round'](_0x2cab57 / 0xff * 0x5);
            return _0x107977;
        }, _0x464970['ansi16']['rgb'] = function (_0x3bca55) {
            let _0x23ada9 = _0x3bca55 % 0xa;
            if (_0x23ada9 === 0x0 || _0x23ada9 === 0x7)
                return _0x3bca55 > 0x32 && (_0x23ada9 += 3.5), _0x23ada9 = _0x23ada9 / 10.5 * 0xff, [
                    _0x23ada9,
                    _0x23ada9,
                    _0x23ada9
                ];
            const _0x407964 = (~~(_0x3bca55 > 0x32) + 0x1) * 0.5, _0xf28b9d = (_0x23ada9 & 0x1) * _0x407964 * 0xff, _0x131dd7 = (_0x23ada9 >> 0x1 & 0x1) * _0x407964 * 0xff, _0x130faf = (_0x23ada9 >> 0x2 & 0x1) * _0x407964 * 0xff;
            return [
                _0xf28b9d,
                _0x131dd7,
                _0x130faf
            ];
        }, _0x464970['ansi256']['rgb'] = function (_0x2c453f) {
            if (_0x2c453f >= 0xe8) {
                const _0x5552b3 = (_0x2c453f - 0xe8) * 0xa + 0x8;
                return [
                    _0x5552b3,
                    _0x5552b3,
                    _0x5552b3
                ];
            }
            _0x2c453f -= 0x10;
            let _0x4498a6;
            const _0x5468ec = Math['floor'](_0x2c453f / 0x24) / 0x5 * 0xff, _0x243a99 = Math['floor']((_0x4498a6 = _0x2c453f % 0x24) / 0x6) / 0x5 * 0xff, _0x52d46d = _0x4498a6 % 0x6 / 0x5 * 0xff;
            return [
                _0x5468ec,
                _0x243a99,
                _0x52d46d
            ];
        }, _0x464970['rgb']['hex'] = function (_0x4054cc) {
            const _0xc036e6 = ((Math['round'](_0x4054cc[0x0]) & 0xff) << 0x10) + ((Math['round'](_0x4054cc[0x1]) & 0xff) << 0x8) + (Math['round'](_0x4054cc[0x2]) & 0xff), _0x3aafa5 = _0xc036e6['toString'](0x10)['toUpperCase']();
            return '000000'['substring'](_0x3aafa5['length']) + _0x3aafa5;
        }, _0x464970['hex']['rgb'] = function (_0x3346cb) {
            const _0x11fb28 = _0x3346cb['toString'](0x10)['match'](/[a-f0-9]{6}|[a-f0-9]{3}/i);
            if (!_0x11fb28)
                return [
                    0x0,
                    0x0,
                    0x0
                ];
            let _0x360c04 = _0x11fb28[0x0];
            _0x11fb28[0x0]['length'] === 0x3 && (_0x360c04 = _0x360c04['split']('')['map'](_0x136137 => {
                return _0x136137 + _0x136137;
            })['join'](''));
            const _0x4fb6ad = parseInt(_0x360c04, 0x10), _0x57eb38 = _0x4fb6ad >> 0x10 & 0xff, _0x37b86f = _0x4fb6ad >> 0x8 & 0xff, _0x1b037a = _0x4fb6ad & 0xff;
            return [
                _0x57eb38,
                _0x37b86f,
                _0x1b037a
            ];
        }, _0x464970['rgb']['hcg'] = function (_0x2bdbfd) {
            const _0x228ea3 = _0x2bdbfd[0x0] / 0xff, _0x373927 = _0x2bdbfd[0x1] / 0xff, _0x7a9a04 = _0x2bdbfd[0x2] / 0xff, _0xa8190e = Math['max'](Math['max'](_0x228ea3, _0x373927), _0x7a9a04), _0x1088ed = Math['min'](Math['min'](_0x228ea3, _0x373927), _0x7a9a04), _0x136519 = _0xa8190e - _0x1088ed;
            let _0x4ed160, _0x3d08ae;
            _0x136519 < 0x1 ? _0x4ed160 = _0x1088ed / (0x1 - _0x136519) : _0x4ed160 = 0x0;
            if (_0x136519 <= 0x0)
                _0x3d08ae = 0x0;
            else {
                if (_0xa8190e === _0x228ea3)
                    _0x3d08ae = (_0x373927 - _0x7a9a04) / _0x136519 % 0x6;
                else
                    _0xa8190e === _0x373927 ? _0x3d08ae = 0x2 + (_0x7a9a04 - _0x228ea3) / _0x136519 : _0x3d08ae = 0x4 + (_0x228ea3 - _0x373927) / _0x136519;
            }
            return _0x3d08ae /= 0x6, _0x3d08ae %= 0x1, [
                _0x3d08ae * 0x168,
                _0x136519 * 0x64,
                _0x4ed160 * 0x64
            ];
        }, _0x464970['hsl']['hcg'] = function (_0x1e57d6) {
            const _0x91567d = _0x1e57d6[0x1] / 0x64, _0x1f5ae3 = _0x1e57d6[0x2] / 0x64, _0x4fe805 = _0x1f5ae3 < 0.5 ? 0x2 * _0x91567d * _0x1f5ae3 : 0x2 * _0x91567d * (0x1 - _0x1f5ae3);
            let _0x3998f8 = 0x0;
            return _0x4fe805 < 0x1 && (_0x3998f8 = (_0x1f5ae3 - 0.5 * _0x4fe805) / (0x1 - _0x4fe805)), [
                _0x1e57d6[0x0],
                _0x4fe805 * 0x64,
                _0x3998f8 * 0x64
            ];
        }, _0x464970['hsv']['hcg'] = function (_0x30b2d6) {
            const _0x20c62e = _0x30b2d6[0x1] / 0x64, _0x48cbc1 = _0x30b2d6[0x2] / 0x64, _0x4e47f6 = _0x20c62e * _0x48cbc1;
            let _0x5169f7 = 0x0;
            return _0x4e47f6 < 0x1 && (_0x5169f7 = (_0x48cbc1 - _0x4e47f6) / (0x1 - _0x4e47f6)), [
                _0x30b2d6[0x0],
                _0x4e47f6 * 0x64,
                _0x5169f7 * 0x64
            ];
        }, _0x464970['hcg']['rgb'] = function (_0x5329e5) {
            const _0x53ac20 = _0x5329e5[0x0] / 0x168, _0x503e48 = _0x5329e5[0x1] / 0x64, _0x42fc3b = _0x5329e5[0x2] / 0x64;
            if (_0x503e48 === 0x0)
                return [
                    _0x42fc3b * 0xff,
                    _0x42fc3b * 0xff,
                    _0x42fc3b * 0xff
                ];
            const _0x138232 = [
                    0x0,
                    0x0,
                    0x0
                ], _0x15a372 = _0x53ac20 % 0x1 * 0x6, _0xf8fc3b = _0x15a372 % 0x1, _0x2f80a9 = 0x1 - _0xf8fc3b;
            let _0x1dd7b7 = 0x0;
            switch (Math['floor'](_0x15a372)) {
            case 0x0:
                _0x138232[0x0] = 0x1, _0x138232[0x1] = _0xf8fc3b, _0x138232[0x2] = 0x0;
                break;
            case 0x1:
                _0x138232[0x0] = _0x2f80a9, _0x138232[0x1] = 0x1, _0x138232[0x2] = 0x0;
                break;
            case 0x2:
                _0x138232[0x0] = 0x0, _0x138232[0x1] = 0x1, _0x138232[0x2] = _0xf8fc3b;
                break;
            case 0x3:
                _0x138232[0x0] = 0x0, _0x138232[0x1] = _0x2f80a9, _0x138232[0x2] = 0x1;
                break;
            case 0x4:
                _0x138232[0x0] = _0xf8fc3b, _0x138232[0x1] = 0x0, _0x138232[0x2] = 0x1;
                break;
            default:
                _0x138232[0x0] = 0x1, _0x138232[0x1] = 0x0, _0x138232[0x2] = _0x2f80a9;
            }
            return _0x1dd7b7 = (0x1 - _0x503e48) * _0x42fc3b, [
                (_0x503e48 * _0x138232[0x0] + _0x1dd7b7) * 0xff,
                (_0x503e48 * _0x138232[0x1] + _0x1dd7b7) * 0xff,
                (_0x503e48 * _0x138232[0x2] + _0x1dd7b7) * 0xff
            ];
        }, _0x464970['hcg']['hsv'] = function (_0x2b9e12) {
            const _0x3fa2c6 = _0x2b9e12[0x1] / 0x64, _0x8d8e99 = _0x2b9e12[0x2] / 0x64, _0x32eae7 = _0x3fa2c6 + _0x8d8e99 * (0x1 - _0x3fa2c6);
            let _0x1a5283 = 0x0;
            return _0x32eae7 > 0x0 && (_0x1a5283 = _0x3fa2c6 / _0x32eae7), [
                _0x2b9e12[0x0],
                _0x1a5283 * 0x64,
                _0x32eae7 * 0x64
            ];
        }, _0x464970['hcg']['hsl'] = function (_0x29671c) {
            const _0x4b4594 = _0x29671c[0x1] / 0x64, _0x3d62ab = _0x29671c[0x2] / 0x64, _0x30333c = _0x3d62ab * (0x1 - _0x4b4594) + 0.5 * _0x4b4594;
            let _0x29fea5 = 0x0;
            if (_0x30333c > 0x0 && _0x30333c < 0.5)
                _0x29fea5 = _0x4b4594 / (0x2 * _0x30333c);
            else
                _0x30333c >= 0.5 && _0x30333c < 0x1 && (_0x29fea5 = _0x4b4594 / (0x2 * (0x1 - _0x30333c)));
            return [
                _0x29671c[0x0],
                _0x29fea5 * 0x64,
                _0x30333c * 0x64
            ];
        }, _0x464970['hcg']['hwb'] = function (_0x363554) {
            const _0x209844 = _0x363554[0x1] / 0x64, _0x5a7141 = _0x363554[0x2] / 0x64, _0x5245ed = _0x209844 + _0x5a7141 * (0x1 - _0x209844);
            return [
                _0x363554[0x0],
                (_0x5245ed - _0x209844) * 0x64,
                (0x1 - _0x5245ed) * 0x64
            ];
        }, _0x464970['hwb']['hcg'] = function (_0x114c1c) {
            const _0xe41b5a = _0x114c1c[0x1] / 0x64, _0x361866 = _0x114c1c[0x2] / 0x64, _0x830f7b = 0x1 - _0x361866, _0x4e6fd7 = _0x830f7b - _0xe41b5a;
            let _0x405f8b = 0x0;
            return _0x4e6fd7 < 0x1 && (_0x405f8b = (_0x830f7b - _0x4e6fd7) / (0x1 - _0x4e6fd7)), [
                _0x114c1c[0x0],
                _0x4e6fd7 * 0x64,
                _0x405f8b * 0x64
            ];
        }, _0x464970['apple']['rgb'] = function (_0x3882a7) {
            return [
                _0x3882a7[0x0] / 0xffff * 0xff,
                _0x3882a7[0x1] / 0xffff * 0xff,
                _0x3882a7[0x2] / 0xffff * 0xff
            ];
        }, _0x464970['rgb']['apple'] = function (_0x3dfbfb) {
            return [
                _0x3dfbfb[0x0] / 0xff * 0xffff,
                _0x3dfbfb[0x1] / 0xff * 0xffff,
                _0x3dfbfb[0x2] / 0xff * 0xffff
            ];
        }, _0x464970['gray']['rgb'] = function (_0x3ddab4) {
            return [
                _0x3ddab4[0x0] / 0x64 * 0xff,
                _0x3ddab4[0x0] / 0x64 * 0xff,
                _0x3ddab4[0x0] / 0x64 * 0xff
            ];
        }, _0x464970['gray']['hsl'] = function (_0x4db683) {
            return [
                0x0,
                0x0,
                _0x4db683[0x0]
            ];
        }, _0x464970['gray']['hsv'] = _0x464970['gray']['hsl'], _0x464970['gray']['hwb'] = function (_0x28c02e) {
            return [
                0x0,
                0x64,
                _0x28c02e[0x0]
            ];
        }, _0x464970['gray']['cmyk'] = function (_0x10c6e2) {
            return [
                0x0,
                0x0,
                0x0,
                _0x10c6e2[0x0]
            ];
        }, _0x464970['gray']['lab'] = function (_0x418326) {
            return [
                _0x418326[0x0],
                0x0,
                0x0
            ];
        }, _0x464970['gray']['hex'] = function (_0x5e5c64) {
            const _0x1cf566 = Math['round'](_0x5e5c64[0x0] / 0x64 * 0xff) & 0xff, _0x187119 = (_0x1cf566 << 0x10) + (_0x1cf566 << 0x8) + _0x1cf566, _0xbd24fc = _0x187119['toString'](0x10)['toUpperCase']();
            return '000000'['substring'](_0xbd24fc['length']) + _0xbd24fc;
        }, _0x464970['rgb']['gray'] = function (_0x1b6f86) {
            const _0x11a200 = (_0x1b6f86[0x0] + _0x1b6f86[0x1] + _0x1b6f86[0x2]) / 0x3;
            return [_0x11a200 / 0xff * 0x64];
        };
    },
    0xad3: (_0x26e22a, _0x5c4a63, _0x271983) => {
        const _0x4f9cfa = _0x271983(0x13ed), _0xe13be9 = _0x271983(0xc51), _0x4157da = {}, _0x498eb5 = Object['keys'](_0x4f9cfa);
        function _0xa1ca52(_0x55ef5c) {
            const _0x449a8 = function (..._0x54f84c) {
                const _0x59de5c = _0x54f84c[0x0];
                if (_0x59de5c === undefined || _0x59de5c === null)
                    return _0x59de5c;
                return _0x59de5c['length'] > 0x1 && (_0x54f84c = _0x59de5c), _0x55ef5c(_0x54f84c);
            };
            return 'conversion' in _0x55ef5c && (_0x449a8['conversion'] = _0x55ef5c['conversion']), _0x449a8;
        }
        function _0x2263ce(_0x2667ac) {
            const _0x3be7cb = function (..._0x1888f4) {
                const _0x18abb0 = _0x1888f4[0x0];
                if (_0x18abb0 === undefined || _0x18abb0 === null)
                    return _0x18abb0;
                _0x18abb0['length'] > 0x1 && (_0x1888f4 = _0x18abb0);
                const _0x12ce0d = _0x2667ac(_0x1888f4);
                if (typeof _0x12ce0d === 'object')
                    for (let _0x2c6e4e = _0x12ce0d['length'], _0x366eea = 0x0; _0x366eea < _0x2c6e4e; _0x366eea++) {
                        _0x12ce0d[_0x366eea] = Math['round'](_0x12ce0d[_0x366eea]);
                    }
                return _0x12ce0d;
            };
            return 'conversion' in _0x2667ac && (_0x3be7cb['conversion'] = _0x2667ac['conversion']), _0x3be7cb;
        }
        _0x498eb5['forEach'](_0xe6322e => {
            _0x4157da[_0xe6322e] = {}, Object['defineProperty'](_0x4157da[_0xe6322e], 'channels', { 'value': _0x4f9cfa[_0xe6322e]['channels'] }), Object['defineProperty'](_0x4157da[_0xe6322e], 'labels', { 'value': _0x4f9cfa[_0xe6322e]['labels'] });
            const _0x4e8b78 = _0xe13be9(_0xe6322e), _0x51c174 = Object['keys'](_0x4e8b78);
            _0x51c174['forEach'](_0xc844a5 => {
                const _0x4f51f8 = _0x4e8b78[_0xc844a5];
                _0x4157da[_0xe6322e][_0xc844a5] = _0x2263ce(_0x4f51f8), _0x4157da[_0xe6322e][_0xc844a5]['raw'] = _0xa1ca52(_0x4f51f8);
            });
        }), _0x26e22a['exports'] = _0x4157da;
    },
    0xc51: (_0x3f7019, _0x3114f7, _0x1d4ef4) => {
        const _0x2d3735 = _0x1d4ef4(0x13ed);
        function _0x3597ea() {
            const _0x182dd2 = {}, _0x2ac821 = Object['keys'](_0x2d3735);
            for (let _0x3f1faa = _0x2ac821['length'], _0x25dc99 = 0x0; _0x25dc99 < _0x3f1faa; _0x25dc99++) {
                _0x182dd2[_0x2ac821[_0x25dc99]] = {
                    'distance': -0x1,
                    'parent': null
                };
            }
            return _0x182dd2;
        }
        function _0x7e0429(_0x25e9d9) {
            const _0x3e5959 = _0x3597ea(), _0x2e1cee = [_0x25e9d9];
            _0x3e5959[_0x25e9d9]['distance'] = 0x0;
            while (_0x2e1cee['length']) {
                const _0xbe36a0 = _0x2e1cee['pop'](), _0x531890 = Object['keys'](_0x2d3735[_0xbe36a0]);
                for (let _0x596ded = _0x531890['length'], _0x27515f = 0x0; _0x27515f < _0x596ded; _0x27515f++) {
                    const _0x325ca5 = _0x531890[_0x27515f], _0x153837 = _0x3e5959[_0x325ca5];
                    _0x153837['distance'] === -0x1 && (_0x153837['distance'] = _0x3e5959[_0xbe36a0]['distance'] + 0x1, _0x153837['parent'] = _0xbe36a0, _0x2e1cee['unshift'](_0x325ca5));
                }
            }
            return _0x3e5959;
        }
        function _0x5d74be(_0x1ffb11, _0x25b4c6) {
            return function (_0x5d9879) {
                return _0x25b4c6(_0x1ffb11(_0x5d9879));
            };
        }
        function _0x3ec04c(_0x5048aa, _0x365f5b) {
            const _0x2cb561 = [
                _0x365f5b[_0x5048aa]['parent'],
                _0x5048aa
            ];
            let _0x3c9aea = _0x2d3735[_0x365f5b[_0x5048aa]['parent']][_0x5048aa], _0x32c014 = _0x365f5b[_0x5048aa]['parent'];
            while (_0x365f5b[_0x32c014]['parent']) {
                _0x2cb561['unshift'](_0x365f5b[_0x32c014]['parent']), _0x3c9aea = _0x5d74be(_0x2d3735[_0x365f5b[_0x32c014]['parent']][_0x32c014], _0x3c9aea), _0x32c014 = _0x365f5b[_0x32c014]['parent'];
            }
            return _0x3c9aea['conversion'] = _0x2cb561, _0x3c9aea;
        }
        _0x3f7019['exports'] = function (_0x5e0690) {
            const _0x1794ea = _0x7e0429(_0x5e0690), _0x445bf6 = {}, _0x5d1d5b = Object['keys'](_0x1794ea);
            for (let _0x4d767b = _0x5d1d5b['length'], _0x3b113d = 0x0; _0x3b113d < _0x4d767b; _0x3b113d++) {
                const _0x344763 = _0x5d1d5b[_0x3b113d], _0x5c4398 = _0x1794ea[_0x344763];
                if (_0x5c4398['parent'] === null)
                    continue;
                _0x445bf6[_0x344763] = _0x3ec04c(_0x344763, _0x1794ea);
            }
            return _0x445bf6;
        };
    },
    0x1845b: _0x3b1218 => {
        'use strict';
        _0x3b1218['exports'] = {
            'aliceblue': [
                0xf0,
                0xf8,
                0xff
            ],
            'antiquewhite': [
                0xfa,
                0xeb,
                0xd7
            ],
            'aqua': [
                0x0,
                0xff,
                0xff
            ],
            'aquamarine': [
                0x7f,
                0xff,
                0xd4
            ],
            'azure': [
                0xf0,
                0xff,
                0xff
            ],
            'beige': [
                0xf5,
                0xf5,
                0xdc
            ],
            'bisque': [
                0xff,
                0xe4,
                0xc4
            ],
            'black': [
                0x0,
                0x0,
                0x0
            ],
            'blanchedalmond': [
                0xff,
                0xeb,
                0xcd
            ],
            'blue': [
                0x0,
                0x0,
                0xff
            ],
            'blueviolet': [
                0x8a,
                0x2b,
                0xe2
            ],
            'brown': [
                0xa5,
                0x2a,
                0x2a
            ],
            'burlywood': [
                0xde,
                0xb8,
                0x87
            ],
            'cadetblue': [
                0x5f,
                0x9e,
                0xa0
            ],
            'chartreuse': [
                0x7f,
                0xff,
                0x0
            ],
            'chocolate': [
                0xd2,
                0x69,
                0x1e
            ],
            'coral': [
                0xff,
                0x7f,
                0x50
            ],
            'cornflowerblue': [
                0x64,
                0x95,
                0xed
            ],
            'cornsilk': [
                0xff,
                0xf8,
                0xdc
            ],
            'crimson': [
                0xdc,
                0x14,
                0x3c
            ],
            'cyan': [
                0x0,
                0xff,
                0xff
            ],
            'darkblue': [
                0x0,
                0x0,
                0x8b
            ],
            'darkcyan': [
                0x0,
                0x8b,
                0x8b
            ],
            'darkgoldenrod': [
                0xb8,
                0x86,
                0xb
            ],
            'darkgray': [
                0xa9,
                0xa9,
                0xa9
            ],
            'darkgreen': [
                0x0,
                0x64,
                0x0
            ],
            'darkgrey': [
                0xa9,
                0xa9,
                0xa9
            ],
            'darkkhaki': [
                0xbd,
                0xb7,
                0x6b
            ],
            'darkmagenta': [
                0x8b,
                0x0,
                0x8b
            ],
            'darkolivegreen': [
                0x55,
                0x6b,
                0x2f
            ],
            'darkorange': [
                0xff,
                0x8c,
                0x0
            ],
            'darkorchid': [
                0x99,
                0x32,
                0xcc
            ],
            'darkred': [
                0x8b,
                0x0,
                0x0
            ],
            'darksalmon': [
                0xe9,
                0x96,
                0x7a
            ],
            'darkseagreen': [
                0x8f,
                0xbc,
                0x8f
            ],
            'darkslateblue': [
                0x48,
                0x3d,
                0x8b
            ],
            'darkslategray': [
                0x2f,
                0x4f,
                0x4f
            ],
            'darkslategrey': [
                0x2f,
                0x4f,
                0x4f
            ],
            'darkturquoise': [
                0x0,
                0xce,
                0xd1
            ],
            'darkviolet': [
                0x94,
                0x0,
                0xd3
            ],
            'deeppink': [
                0xff,
                0x14,
                0x93
            ],
            'deepskyblue': [
                0x0,
                0xbf,
                0xff
            ],
            'dimgray': [
                0x69,
                0x69,
                0x69
            ],
            'dimgrey': [
                0x69,
                0x69,
                0x69
            ],
            'dodgerblue': [
                0x1e,
                0x90,
                0xff
            ],
            'firebrick': [
                0xb2,
                0x22,
                0x22
            ],
            'floralwhite': [
                0xff,
                0xfa,
                0xf0
            ],
            'forestgreen': [
                0x22,
                0x8b,
                0x22
            ],
            'fuchsia': [
                0xff,
                0x0,
                0xff
            ],
            'gainsboro': [
                0xdc,
                0xdc,
                0xdc
            ],
            'ghostwhite': [
                0xf8,
                0xf8,
                0xff
            ],
            'gold': [
                0xff,
                0xd7,
                0x0
            ],
            'goldenrod': [
                0xda,
                0xa5,
                0x20
            ],
            'gray': [
                0x80,
                0x80,
                0x80
            ],
            'green': [
                0x0,
                0x80,
                0x0
            ],
            'greenyellow': [
                0xad,
                0xff,
                0x2f
            ],
            'grey': [
                0x80,
                0x80,
                0x80
            ],
            'honeydew': [
                0xf0,
                0xff,
                0xf0
            ],
            'hotpink': [
                0xff,
                0x69,
                0xb4
            ],
            'indianred': [
                0xcd,
                0x5c,
                0x5c
            ],
            'indigo': [
                0x4b,
                0x0,
                0x82
            ],
            'ivory': [
                0xff,
                0xff,
                0xf0
            ],
            'khaki': [
                0xf0,
                0xe6,
                0x8c
            ],
            'lavender': [
                0xe6,
                0xe6,
                0xfa
            ],
            'lavenderblush': [
                0xff,
                0xf0,
                0xf5
            ],
            'lawngreen': [
                0x7c,
                0xfc,
                0x0
            ],
            'lemonchiffon': [
                0xff,
                0xfa,
                0xcd
            ],
            'lightblue': [
                0xad,
                0xd8,
                0xe6
            ],
            'lightcoral': [
                0xf0,
                0x80,
                0x80
            ],
            'lightcyan': [
                0xe0,
                0xff,
                0xff
            ],
            'lightgoldenrodyellow': [
                0xfa,
                0xfa,
                0xd2
            ],
            'lightgray': [
                0xd3,
                0xd3,
                0xd3
            ],
            'lightgreen': [
                0x90,
                0xee,
                0x90
            ],
            'lightgrey': [
                0xd3,
                0xd3,
                0xd3
            ],
            'lightpink': [
                0xff,
                0xb6,
                0xc1
            ],
            'lightsalmon': [
                0xff,
                0xa0,
                0x7a
            ],
            'lightseagreen': [
                0x20,
                0xb2,
                0xaa
            ],
            'lightskyblue': [
                0x87,
                0xce,
                0xfa
            ],
            'lightslategray': [
                0x77,
                0x88,
                0x99
            ],
            'lightslategrey': [
                0x77,
                0x88,
                0x99
            ],
            'lightsteelblue': [
                0xb0,
                0xc4,
                0xde
            ],
            'lightyellow': [
                0xff,
                0xff,
                0xe0
            ],
            'lime': [
                0x0,
                0xff,
                0x0
            ],
            'limegreen': [
                0x32,
                0xcd,
                0x32
            ],
            'linen': [
                0xfa,
                0xf0,
                0xe6
            ],
            'magenta': [
                0xff,
                0x0,
                0xff
            ],
            'maroon': [
                0x80,
                0x0,
                0x0
            ],
            'mediumaquamarine': [
                0x66,
                0xcd,
                0xaa
            ],
            'mediumblue': [
                0x0,
                0x0,
                0xcd
            ],
            'mediumorchid': [
                0xba,
                0x55,
                0xd3
            ],
            'mediumpurple': [
                0x93,
                0x70,
                0xdb
            ],
            'mediumseagreen': [
                0x3c,
                0xb3,
                0x71
            ],
            'mediumslateblue': [
                0x7b,
                0x68,
                0xee
            ],
            'mediumspringgreen': [
                0x0,
                0xfa,
                0x9a
            ],
            'mediumturquoise': [
                0x48,
                0xd1,
                0xcc
            ],
            'mediumvioletred': [
                0xc7,
                0x15,
                0x85
            ],
            'midnightblue': [
                0x19,
                0x19,
                0x70
            ],
            'mintcream': [
                0xf5,
                0xff,
                0xfa
            ],
            'mistyrose': [
                0xff,
                0xe4,
                0xe1
            ],
            'moccasin': [
                0xff,
                0xe4,
                0xb5
            ],
            'navajowhite': [
                0xff,
                0xde,
                0xad
            ],
            'navy': [
                0x0,
                0x0,
                0x80
            ],
            'oldlace': [
                0xfd,
                0xf5,
                0xe6
            ],
            'olive': [
                0x80,
                0x80,
                0x0
            ],
            'olivedrab': [
                0x6b,
                0x8e,
                0x23
            ],
            'orange': [
                0xff,
                0xa5,
                0x0
            ],
            'orangered': [
                0xff,
                0x45,
                0x0
            ],
            'orchid': [
                0xda,
                0x70,
                0xd6
            ],
            'palegoldenrod': [
                0xee,
                0xe8,
                0xaa
            ],
            'palegreen': [
                0x98,
                0xfb,
                0x98
            ],
            'paleturquoise': [
                0xaf,
                0xee,
                0xee
            ],
            'palevioletred': [
                0xdb,
                0x70,
                0x93
            ],
            'papayawhip': [
                0xff,
                0xef,
                0xd5
            ],
            'peachpuff': [
                0xff,
                0xda,
                0xb9
            ],
            'peru': [
                0xcd,
                0x85,
                0x3f
            ],
            'pink': [
                0xff,
                0xc0,
                0xcb
            ],
            'plum': [
                0xdd,
                0xa0,
                0xdd
            ],
            'powderblue': [
                0xb0,
                0xe0,
                0xe6
            ],
            'purple': [
                0x80,
                0x0,
                0x80
            ],
            'rebeccapurple': [
                0x66,
                0x33,
                0x99
            ],
            'red': [
                0xff,
                0x0,
                0x0
            ],
            'rosybrown': [
                0xbc,
                0x8f,
                0x8f
            ],
            'royalblue': [
                0x41,
                0x69,
                0xe1
            ],
            'saddlebrown': [
                0x8b,
                0x45,
                0x13
            ],
            'salmon': [
                0xfa,
                0x80,
                0x72
            ],
            'sandybrown': [
                0xf4,
                0xa4,
                0x60
            ],
            'seagreen': [
                0x2e,
                0x8b,
                0x57
            ],
            'seashell': [
                0xff,
                0xf5,
                0xee
            ],
            'sienna': [
                0xa0,
                0x52,
                0x2d
            ],
            'silver': [
                0xc0,
                0xc0,
                0xc0
            ],
            'skyblue': [
                0x87,
                0xce,
                0xeb
            ],
            'slateblue': [
                0x6a,
                0x5a,
                0xcd
            ],
            'slategray': [
                0x70,
                0x80,
                0x90
            ],
            'slategrey': [
                0x70,
                0x80,
                0x90
            ],
            'snow': [
                0xff,
                0xfa,
                0xfa
            ],
            'springgreen': [
                0x0,
                0xff,
                0x7f
            ],
            'steelblue': [
                0x46,
                0x82,
                0xb4
            ],
            'tan': [
                0xd2,
                0xb4,
                0x8c
            ],
            'teal': [
                0x0,
                0x80,
                0x80
            ],
            'thistle': [
                0xd8,
                0xbf,
                0xd8
            ],
            'tomato': [
                0xff,
                0x63,
                0x47
            ],
            'turquoise': [
                0x40,
                0xe0,
                0xd0
            ],
            'violet': [
                0xee,
                0x82,
                0xee
            ],
            'wheat': [
                0xf5,
                0xde,
                0xb3
            ],
            'white': [
                0xff,
                0xff,
                0xff
            ],
            'whitesmoke': [
                0xf5,
                0xf5,
                0xf5
            ],
            'yellow': [
                0xff,
                0xff,
                0x0
            ],
            'yellowgreen': [
                0x9a,
                0xcd,
                0x32
            ]
        };
    },
    0x908f: _0x2d789e => {
        'use strict';
        _0x2d789e['exports'] = (_0x3603bd, _0x5c2066 = process['argv']) => {
            const _0x2a03ab = _0x3603bd['startsWith']('-') ? '' : _0x3603bd['length'] === 0x1 ? '-' : '--', _0x2408fe = _0x5c2066['indexOf'](_0x2a03ab + _0x3603bd), _0x56db1b = _0x5c2066['indexOf']('--');
            return _0x2408fe !== -0x1 && (_0x56db1b === -0x1 || _0x2408fe < _0x56db1b);
        };
    },
    0x9583: (_0x2decfd, _0x183a6d, _0x58c122) => {
        'use strict';
        const _0x542904 = _0x58c122(0x5615), _0x5e6842 = _0x58c122(0x129c0), _0x1cda8b = _0x58c122(0x908f), {env: _0x43f9d6} = process;
        let _0x415301;
        if (_0x1cda8b('no-color') || _0x1cda8b('no-colors') || _0x1cda8b('color=false') || _0x1cda8b('color=never'))
            _0x415301 = 0x0;
        else
            (_0x1cda8b('color') || _0x1cda8b('colors') || _0x1cda8b('color=true') || _0x1cda8b('color=always')) && (_0x415301 = 0x1);
        if ('FORCE_COLOR' in _0x43f9d6) {
            if (_0x43f9d6['FORCE_COLOR'] === 'true')
                _0x415301 = 0x1;
            else
                _0x43f9d6['FORCE_COLOR'] === 'false' ? _0x415301 = 0x0 : _0x415301 = _0x43f9d6['FORCE_COLOR']['length'] === 0x0 ? 0x1 : Math['min'](parseInt(_0x43f9d6['FORCE_COLOR'], 0xa), 0x3);
        }
        function _0x410764(_0x32aecd) {
            if (_0x32aecd === 0x0)
                return ![];
            return {
                'level': _0x32aecd,
                'hasBasic': !![],
                'has256': _0x32aecd >= 0x2,
                'has16m': _0x32aecd >= 0x3
            };
        }
        function _0x4918ff(_0x476371, _0x578a16) {
            if (_0x415301 === 0x0)
                return 0x0;
            if (_0x1cda8b('color=16m') || _0x1cda8b('color=full') || _0x1cda8b('color=truecolor'))
                return 0x3;
            if (_0x1cda8b('color=256'))
                return 0x2;
            if (_0x476371 && !_0x578a16 && _0x415301 === undefined)
                return 0x0;
            const _0x8f26d = _0x415301 || 0x0;
            if (_0x43f9d6['TERM'] === 'dumb')
                return _0x8f26d;
            if (process['platform'] === 'win32') {
                const _0x467cfd = _0x542904['release']()['split']('.');
                if (Number(_0x467cfd[0x0]) >= 0xa && Number(_0x467cfd[0x2]) >= 0x295a)
                    return Number(_0x467cfd[0x2]) >= 0x3a53 ? 0x3 : 0x2;
                return 0x1;
            }
            if ('CI' in _0x43f9d6) {
                if ([
                        'TRAVIS',
                        'CIRCLECI',
                        'APPVEYOR',
                        'GITLAB_CI',
                        'GITHUB_ACTIONS',
                        'BUILDKITE'
                    ]['some'](_0x4c1f9b => _0x4c1f9b in _0x43f9d6) || _0x43f9d6['CI_NAME'] === 'codeship')
                    return 0x1;
                return _0x8f26d;
            }
            if ('TEAMCITY_VERSION' in _0x43f9d6)
                return /^(9\.(0*[1-9]\d*)\.|\d{2,}\.)/['test'](_0x43f9d6['TEAMCITY_VERSION']) ? 0x1 : 0x0;
            if (_0x43f9d6['COLORTERM'] === 'truecolor')
                return 0x3;
            if ('TERM_PROGRAM' in _0x43f9d6) {
                const _0x3f71e4 = parseInt((_0x43f9d6['TERM_PROGRAM_VERSION'] || '')['split']('.')[0x0], 0xa);
                switch (_0x43f9d6['TERM_PROGRAM']) {
                case 'iTerm.app':
                    return _0x3f71e4 >= 0x3 ? 0x3 : 0x2;
                case 'Apple_Terminal':
                    return 0x2;
                }
            }
            if (/-256(color)?$/i['test'](_0x43f9d6['TERM']))
                return 0x2;
            if (/^screen|^xterm|^vt100|^vt220|^rxvt|color|ansi|cygwin|linux/i['test'](_0x43f9d6['TERM']))
                return 0x1;
            if ('COLORTERM' in _0x43f9d6)
                return 0x1;
            return _0x8f26d;
        }
        function _0x5e97b3(_0x41a4a3) {
            const _0x448ffc = _0x4918ff(_0x41a4a3, _0x41a4a3 && _0x41a4a3['isTTY']);
            return _0x410764(_0x448ffc);
        }
        _0x2decfd['exports'] = {
            'supportsColor': _0x5e97b3,
            'stdout': _0x410764(_0x4918ff(!![], _0x5e6842['isatty'](0x1))),
            'stderr': _0x410764(_0x4918ff(!![], _0x5e6842['isatty'](0x2)))
        };
    },
    0x3b1: (_0x4eb18e, _0x29fd85, _0x38c857) => {
        var _0x206002 = '2.0.1', _0xa4a65c, _0x438de8, _0xf01398, _0x439f53, _0x56ef6c, _0x4bc308, _0x11e957, _0x36f486, _0x208c1a, _0x442ace, _0x178eff, _0x34facd = []['slice'], _0x304989 = []['indexOf'] || function (_0x18bbc7) {
                for (var _0x466329 = 0x0, _0x5a2f55 = this['length']; _0x466329 < _0x5a2f55; _0x466329++) {
                    if (_0x466329 in this && this[_0x466329] === _0x18bbc7)
                        return _0x466329;
                }
                return -0x1;
            }, _0x49b5a1 = {}['hasOwnProperty'];
        _0x11e957 = _0x38c857(0x11569), _0xf01398 = function (_0x3854a7) {
            return typeof _0x3854a7 === 'function';
        }, _0x439f53 = function (_0xaff30) {
            return typeof _0xaff30 === 'string' || !!_0xaff30 && typeof _0xaff30 === 'object' && Object['prototype']['toString']['call'](_0xaff30) === '[object\x20String]';
        }, _0x178eff = _0x29fd85, _0x178eff['VERSION'] = typeof _0x206002 !== 'undefined' && _0x206002 !== null ? _0x206002 : 'NO-VERSION', _0x442ace = function (_0x44a316) {
            return _0x44a316 = _0x44a316['replace'](/\\/g, '/'), _0x44a316 = _0x44a316['replace'](/(?<!^)\/+/g, '/'), _0x44a316;
        };
        for (_0x36f486 in _0x11e957) {
            _0x208c1a = _0x11e957[_0x36f486], _0xf01398(_0x208c1a) ? _0x178eff[_0x36f486] = function (_0x5ad706) {
                return function () {
                    var _0x1e3044, _0x248f9e;
                    return _0x1e3044 = 0x1 <= arguments['length'] ? _0x34facd['call'](arguments, 0x0) : [], _0x1e3044 = _0x1e3044['map'](function (_0x410410) {
                        return _0x439f53(_0x410410) ? _0x442ace(_0x410410) : _0x410410;
                    }), _0x248f9e = _0x11e957[_0x5ad706]['apply'](_0x11e957, _0x1e3044), _0x439f53(_0x248f9e) ? _0x442ace(_0x248f9e) : _0x248f9e;
                };
            }(_0x36f486) : _0x178eff[_0x36f486] = _0x208c1a;
        }
        _0x178eff['sep'] = '/', _0x438de8 = {
            'toUnix': _0x442ace,
            'normalizeSafe': function (_0x4d4e8d) {
                var _0x4a5aa1;
                _0x4d4e8d = _0x442ace(_0x4d4e8d), _0x4a5aa1 = _0x178eff['normalize'](_0x4d4e8d);
                if (_0x4d4e8d['startsWith']('./') && !_0x4a5aa1['startsWith']('./') && !_0x4a5aa1['startsWith']('..'))
                    _0x4a5aa1 = './' + _0x4a5aa1;
                else
                    _0x4d4e8d['startsWith']('//') && !_0x4a5aa1['startsWith']('//') && (_0x4d4e8d['startsWith']('//./') ? _0x4a5aa1 = '//.' + _0x4a5aa1 : _0x4a5aa1 = '/' + _0x4a5aa1);
                return _0x4a5aa1;
            },
            'normalizeTrim': function (_0x3030f) {
                return _0x3030f = _0x178eff['normalizeSafe'](_0x3030f), _0x3030f['endsWith']('/') ? _0x3030f['slice'](0x0, +(_0x3030f['length'] - 0x2) + 0x1 || 0x218711a00) : _0x3030f;
            },
            'joinSafe': function () {
                var _0x2a2b97, _0x181254, _0x5b1826;
                _0x2a2b97 = 0x1 <= arguments['length'] ? _0x34facd['call'](arguments, 0x0) : [], _0x5b1826 = _0x178eff['join']['apply'](null, _0x2a2b97);
                if (_0x2a2b97['length'] > 0x0) {
                    _0x181254 = _0x442ace(_0x2a2b97[0x0]);
                    if (_0x181254['startsWith']('./') && !_0x5b1826['startsWith']('./') && !_0x5b1826['startsWith']('..'))
                        _0x5b1826 = './' + _0x5b1826;
                    else
                        _0x181254['startsWith']('//') && !_0x5b1826['startsWith']('//') && (_0x181254['startsWith']('//./') ? _0x5b1826 = '//.' + _0x5b1826 : _0x5b1826 = '/' + _0x5b1826);
                }
                return _0x5b1826;
            },
            'addExt': function (_0xa450ec, _0x1542ee) {
                return !_0x1542ee ? _0xa450ec : (_0x1542ee[0x0] !== '.' && (_0x1542ee = '.' + _0x1542ee), _0xa450ec + (_0xa450ec['endsWith'](_0x1542ee) ? '' : _0x1542ee));
            },
            'trimExt': function (_0x23d207, _0x2b2b82, _0x350e40) {
                var _0x26a904;
                return _0x350e40 == null && (_0x350e40 = 0x7), _0x26a904 = _0x178eff['extname'](_0x23d207), _0x56ef6c(_0x26a904, _0x2b2b82, _0x350e40) ? _0x23d207['slice'](0x0, +(_0x23d207['length'] - _0x26a904['length'] - 0x1) + 0x1 || 0x218711a00) : _0x23d207;
            },
            'removeExt': function (_0x33947d, _0x29b74c) {
                return !_0x29b74c ? _0x33947d : (_0x29b74c = _0x29b74c[0x0] === '.' ? _0x29b74c : '.' + _0x29b74c, _0x178eff['extname'](_0x33947d) === _0x29b74c ? _0x178eff['trimExt'](_0x33947d, [], _0x29b74c['length']) : _0x33947d);
            },
            'changeExt': function (_0xdb37c2, _0x3645de, _0x3b15c3, _0x4fda4e) {
                return _0x4fda4e == null && (_0x4fda4e = 0x7), _0x178eff['trimExt'](_0xdb37c2, _0x3b15c3, _0x4fda4e) + (!_0x3645de ? '' : _0x3645de[0x0] === '.' ? _0x3645de : '.' + _0x3645de);
            },
            'defaultExt': function (_0x1e8dd3, _0x5a63e9, _0x53fb52, _0x34d3dc) {
                var _0x2ed87f;
                return _0x34d3dc == null && (_0x34d3dc = 0x7), _0x2ed87f = _0x178eff['extname'](_0x1e8dd3), _0x56ef6c(_0x2ed87f, _0x53fb52, _0x34d3dc) ? _0x1e8dd3 : _0x178eff['addExt'](_0x1e8dd3, _0x5a63e9);
            }
        }, _0x56ef6c = function (_0x41e20c, _0x14ce69, _0x2843c8) {
            return _0x14ce69 == null && (_0x14ce69 = []), _0x41e20c && _0x41e20c['length'] <= _0x2843c8 && _0x304989['call'](_0x14ce69['map'](function (_0x2d896c) {
                return (_0x2d896c && _0x2d896c[0x0] !== '.' ? '.' : '') + _0x2d896c;
            }), _0x41e20c) < 0x0;
        };
        for (_0x4bc308 in _0x438de8) {
            if (!_0x49b5a1['call'](_0x438de8, _0x4bc308))
                continue;
            _0xa4a65c = _0x438de8[_0x4bc308];
            if (_0x178eff[_0x4bc308] !== void 0x0)
                throw new Error('path.' + _0x4bc308 + '\x20already\x20exists.');
            else
                _0x178eff[_0x4bc308] = _0xa4a65c;
        }
        ;
    }
};
;
