// test-maker/dist/400.js
'use strict';
exports['id'] = 0x190, exports['ids'] = [
    0x190,
    0x23d,
    0x373,
    0x2d5
], exports['modules'] = {
    0x686d: _0xd5efb8 => {
        var _0x94b3c8 = /[|\\{}()[\]^$+*?.]/g;
        _0xd5efb8['exports'] = function (_0x33a42c) {
            if (typeof _0x33a42c !== 'string')
                throw new TypeError('Expected\x20a\x20string');
            return _0x33a42c['replace'](_0x94b3c8, '\x5c$&');
        };
    },
    0x845b: (_0xd74c86, _0x3ae7e0, _0x32f9b5) => {
        _0x32f9b5['r'](_0x3ae7e0), _0x32f9b5['d'](_0x3ae7e0, {
            'generateId': () => _0x14b6c1,
            'generateIdAsync': () => _0x248dfa,
            'generateIdCustom': () => _0x34d793,
            'generateIdCustomAsync': () => _0x5d9808
        });
        var _0x3769b0 = _0x32f9b5(0x16925), _0x27e430 = _0x32f9b5(0xccb6);
        function _0x14b6c1(_0x817907) {
            return (0x0, _0x3769b0['nanoid'])(_0x817907);
        }
        async function _0x248dfa(_0x371b18) {
            return await (0x0, _0x27e430['nanoid'])(_0x371b18);
        }
        function _0x34d793(_0x213351, _0x55b4d6) {
            return (0x0, _0x3769b0['customAlphabet'])(_0x213351, _0x55b4d6)();
        }
        async function _0x5d9808(_0x5905a8, _0x201515) {
            return await (0x0, _0x27e430['customAlphabet'])(_0x5905a8, _0x201515)();
        }
    },
    0x38ed: (_0xd3052a, _0x34c44d, _0x1ae673) => {
        _0x1ae673['r'](_0x34c44d), _0x1ae673['d'](_0x34c44d, { 'logGenericError': () => _0x2be2c6 });
        var _0x1cb698 = _0x1ae673(0x1145c), _0x411474 = _0x1ae673['n'](_0x1cb698);
        async function _0x2be2c6({
            error: _0xfcad43,
            isWarning: _0x2a3935,
            show: show = !![]
        }) {
            let _0xc95d1c;
            try {
                _0xc95d1c ? _0x2a3935 ? console['warn'](_0xc95d1c['formatted']['ansi']) : console['error'](_0xc95d1c['formatted']['ansi']) : _0x2a3935 ? console['warn'](_0xfcad43) : console['error'](_0x411474()['redBright'](_0xfcad43['stack'] || _0xfcad43));
            } catch (_0x1af095) {
                console['error'](_0x411474()['redBright'](_0x1af095['stack'] || _0x1af095));
            } finally {
                console['info'](_0x411474()['cyanBright']('See\x20Log\x20file\x20for\x20more\x20Details...'));
            }
        }
    },
    0xccb6: (_0x322e84, _0x17772c, _0x3b4464) => {
        _0x3b4464['r'](_0x17772c), _0x3b4464['d'](_0x17772c, {
            'customAlphabet': () => _0x78d501,
            'nanoid': () => _0x446758,
            'random': () => _0x369082
        });
        var _0x7e3bad = _0x3b4464(0x17e1), _0x375524 = _0x3b4464(0x12bca);
        let _0x369082 = _0x32c3ed => new Promise((_0x9c0c39, _0x346322) => {
                _0x7e3bad['randomFill'](Buffer['allocUnsafe'](_0x32c3ed), (_0x256bd0, _0x2a1e3b) => {
                    _0x256bd0 ? _0x346322(_0x256bd0) : _0x9c0c39(_0x2a1e3b);
                });
            }), _0x78d501 = (_0xf8bd6f, _0x47bb36 = 0x15) => {
                let _0x19d2b2 = (0x2 << 0x1f - Math['clz32'](_0xf8bd6f['length'] - 0x1 | 0x1)) - 0x1, _0x651b05 = Math['ceil'](1.6 * _0x19d2b2 * _0x47bb36 / _0xf8bd6f['length']), _0x3fbdc5 = (_0x5f23ff, _0xea4056 = _0x47bb36) => _0x369082(_0x651b05)['then'](_0x674dba => {
                        let _0x424c10 = _0x651b05;
                        while (_0x424c10--) {
                            _0x5f23ff += _0xf8bd6f[_0x674dba[_0x424c10] & _0x19d2b2] || '';
                            if (_0x5f23ff['length'] === _0xea4056)
                                return _0x5f23ff;
                        }
                        return _0x3fbdc5(_0x5f23ff, _0xea4056);
                    });
                return _0x5db2db => _0x3fbdc5('', _0x5db2db);
            }, _0x446758 = (_0x4cde3e = 0x15) => _0x369082(_0x4cde3e)['then'](_0x23ada8 => {
                let _0x11145d = '';
                while (_0x4cde3e--) {
                    _0x11145d += _0x375524['urlAlphabet'][_0x23ada8[_0x4cde3e] & 0x3f];
                }
                return _0x11145d;
            });
    },
    0x16925: (_0x21d82e, _0x3bf8cc, _0xdbd847) => {
        _0xdbd847['r'](_0x3bf8cc), _0xdbd847['d'](_0x3bf8cc, {
            'customAlphabet': () => _0x9fb0df,
            'customRandom': () => _0x10b2cf,
            'nanoid': () => _0x4374d7,
            'random': () => _0xbf27,
            'urlAlphabet': () => _0x1b2189['urlAlphabet']
        });
        var _0x1bee1a = _0xdbd847(0x17e1), _0x1b2189 = _0xdbd847(0x12bca);
        const _0x27e217 = 0x80;
        let _0x346daf, _0x28b9db, _0x3ec6ea = _0x35aef3 => {
                if (!_0x346daf || _0x346daf['length'] < _0x35aef3)
                    _0x346daf = Buffer['allocUnsafe'](_0x35aef3 * _0x27e217), _0x1bee1a['randomFillSync'](_0x346daf), _0x28b9db = 0x0;
                else
                    _0x28b9db + _0x35aef3 > _0x346daf['length'] && (_0x1bee1a['randomFillSync'](_0x346daf), _0x28b9db = 0x0);
                _0x28b9db += _0x35aef3;
            }, _0xbf27 = _0x5c5cef => {
                return _0x3ec6ea(_0x5c5cef -= 0x0), _0x346daf['subarray'](_0x28b9db - _0x5c5cef, _0x28b9db);
            }, _0x10b2cf = (_0xbc6dd2, _0x38e252, _0x47693d) => {
                let _0x4d37a4 = (0x2 << 0x1f - Math['clz32'](_0xbc6dd2['length'] - 0x1 | 0x1)) - 0x1, _0x32cc6e = Math['ceil'](1.6 * _0x4d37a4 * _0x38e252 / _0xbc6dd2['length']);
                return (_0x137b09 = _0x38e252) => {
                    let _0x3ac7ba = '';
                    while (!![]) {
                        let _0xf518dd = _0x47693d(_0x32cc6e), _0x2551ba = _0x32cc6e;
                        while (_0x2551ba--) {
                            _0x3ac7ba += _0xbc6dd2[_0xf518dd[_0x2551ba] & _0x4d37a4] || '';
                            if (_0x3ac7ba['length'] === _0x137b09)
                                return _0x3ac7ba;
                        }
                    }
                };
            }, _0x9fb0df = (_0x2777a6, _0x1ef7d1 = 0x15) => _0x10b2cf(_0x2777a6, _0x1ef7d1, _0xbf27), _0x4374d7 = (_0x4c62d1 = 0x15) => {
                _0x3ec6ea(_0x4c62d1 -= 0x0);
                let _0xa09b1f = '';
                for (let _0x18b94c = _0x28b9db - _0x4c62d1; _0x18b94c < _0x28b9db; _0x18b94c++) {
                    _0xa09b1f += _0x1b2189['urlAlphabet'][_0x346daf[_0x18b94c] & 0x3f];
                }
                return _0xa09b1f;
            };
    },
    0x12bca: (_0x29b75e, _0x51961e, _0xbbbe08) => {
        _0xbbbe08['r'](_0x51961e), _0xbbbe08['d'](_0x51961e, { 'urlAlphabet': () => _0x3ed9b9 });
        let _0x3ed9b9 = 'useandom-26T198340PX75pxJACKVERYMINDBUSHWOLF_GQZbfghjklqvwyzrict';
    }
};
;
