// test-maker/dist/573.js
'use strict';
exports['id'] = 0x23d, exports['ids'] = [0x23d], exports['modules'] = {
    0x38ed: (_0x59e923, _0x59dc95, _0x4582d8) => {
        _0x4582d8['r'](_0x59dc95), _0x4582d8['d'](_0x59dc95, { 'logGenericError': () => _0x10bf09 });
        var _0x31d417 = _0x4582d8(0x1145c), _0x63d11c = _0x4582d8['n'](_0x31d417);
        async function _0x10bf09({
            error: _0x88dbb2,
            isWarning: _0x1218cd,
            show: show = !![]
        }) {
            let _0x565ff7;
            try {
                _0x565ff7 ? _0x1218cd ? console['warn'](_0x565ff7['formatted']['ansi']) : console['error'](_0x565ff7['formatted']['ansi']) : _0x1218cd ? console['warn'](_0x88dbb2) : console['error'](_0x63d11c()['redBright'](_0x88dbb2['stack'] || _0x88dbb2));
            } catch (_0x367167) {
                console['error'](_0x63d11c()['redBright'](_0x367167['stack'] || _0x367167));
            } finally {
                console['info'](_0x63d11c()['cyanBright']('See\x20Log\x20file\x20for\x20more\x20Details...'));
            }
        }
    }
};
;
