// test-maker/dist/739.js
exports['id'] = 0x2e3, exports['ids'] = [
    0x2e3,
    0x3a4
], exports['modules'] = {
    0xc30c: _0x31180e => {
        function _0x1376b0(_0x22fa70) {
            var _0x56b547 = new Error('Cannot\x20find\x20module\x20\x27' + _0x22fa70 + '\x27');
            _0x56b547['code'] = 'MODULE_NOT_FOUND';
            throw _0x56b547;
        }
        _0x1376b0['keys'] = () => [], _0x1376b0['resolve'] = _0x1376b0, _0x1376b0['id'] = 0xc30c, _0x31180e['exports'] = _0x1376b0;
    },
    0x10703: (_0x589ab9, _0xb52e55, _0x3e9aef) => {
        'use strict';
        _0x3e9aef['r'](_0xb52e55), _0x3e9aef['d'](_0xb52e55, { 'Validate': () => _0x1d2fb1 });
        var _0x71c39b = _0x3e9aef(0xc165), _0x35d86a = _0x3e9aef['n'](_0x71c39b), _0x27b4ed = _0x3e9aef(0x13caf);
        class _0x1d2fb1 {
            constructor() {
                _0x71c39b['program']['command']('validate', 'Validate\x20Installation')['cast'](![])['strict'](!![])['action'](async () => {
                    await (0x0, _0x27b4ed['validateDeps'])(), process['exit'](0x0);
                });
            }
        }
    },
    0x13caf: (_0xec4240, _0x206722, _0x5f02b4) => {
        'use strict';
        _0x5f02b4['r'](_0x206722), _0x5f02b4['d'](_0x206722, { 'validateDeps': () => _0x3617ea });
        var _0x179294 = _0x5f02b4(0x5395), _0x2bc8c5 = _0x5f02b4['n'](_0x179294), _0xbd40b0 = _0x5f02b4(0x1145c), _0x3184eb = _0x5f02b4['n'](_0xbd40b0), _0x5615c4 = _0x5f02b4(0x17fbe), _0x3e9f6c = _0x5f02b4['n'](_0x5615c4), _0x458d20 = _0x5f02b4(0x3b1), _0x96dad9 = _0x5f02b4['n'](_0x458d20), _0xe49b27 = _0x5f02b4(0x13833);
        async function _0x3617ea() {
            let _0x330ed0 = ![];
            const _0x487da4 = _0xe49b27['testMakerInstallPath'] + '/node_modules/test-maker-dummy/bundles-hash.txt', _0x5caca9 = await _0x3e9f6c()['pathExists'](_0x487da4), _0x147cb6 = './package-lock.json', _0x2093a1 = './pnpm-lock.yaml', _0x50bc26 = './yarn.lock';
            let _0x40b9c6 = '', _0x3b0578 = '', _0x2a3df1 = '';
            await _0x3e9f6c()['pathExists'](_0x147cb6) && (_0x40b9c6 = (await _0x3e9f6c()['stat'](_0x147cb6))['mtime']['toString']());
            await _0x3e9f6c()['pathExists'](_0x2093a1) && (_0x3b0578 = (await _0x3e9f6c()['stat'](_0x2093a1))['mtime']['toString']());
            await _0x3e9f6c()['pathExists'](_0x50bc26) && (_0x2a3df1 = (await _0x3e9f6c()['stat'](_0x50bc26))['mtime']['toString']());
            const _0x4f644d = _0x40b9c6 + _0x3b0578 + _0x2a3df1;
            if (_0x5caca9) {
            } else
                _0x330ed0 = !![];
            if (_0x330ed0) {
                console['info'](_0x3184eb()['cyanBright']('\x0aValidating\x20Test\x20Maker\x20installation\x20(This\x20is\x20only\x20required\x20on\x20first\x20time\x20install\x20or\x20when\x20extra\x20packages\x20are\x20installed/uninstalled)\x0a'));
                let _0x9db26e = new (_0x2bc8c5())(_0xe49b27['testMakerDistPath'] + '/bundled_modules.zip');
                _0x9db26e['extractAllTo'](_0xe49b27['testMakerInstallPath'] + '/node_modules', !![]), await _0x3e9f6c()['outputFile'](_0x487da4, _0x4f644d);
                const _0x25fc92 = [];
                for (let _0x39ddf1 of _0x25fc92) {
                    await _0x3e9f6c()['copy'](_0xe49b27['testMakerInstallPath'] + '/node_modules/' + _0x39ddf1, '' + _0x96dad9()['join'](_0xe49b27['testMakerInstallPath'], '../', _0x39ddf1), { 'overwrite': !![] });
                }
            }
        }
    }
};
;
