// test-maker/dist/906.js
'use strict';
exports['id'] = 0x38a, exports['ids'] = [0x38a], exports['modules'] = {
    0xc6da: (_0x5907cb, _0x4ae340, _0x2cca0a) => {
        _0x2cca0a['r'](_0x4ae340), _0x2cca0a['d'](_0x4ae340, {
            'IpcSlave': () => _0xccb0b3,
            'ipcSlave': () => _0x156d8a
        });
        var _0x56a466 = _0x2cca0a(0x167e3), _0x30c39a = _0x2cca0a['n'](_0x56a466), _0x42b0af = _0x2cca0a(0x10895);
        class _0xccb0b3 {
            constructor() {
                const _0x21b9f3 = process['argv']['indexOf']('--terminal-eye-slave-uid') + 0x1;
                let _0x325e52 = process['argv'][_0x21b9f3];
                const _0x5c7def = 'test-maker-terminal-eye-master-' + _0x325e52;
                _0x56a466['config']['id'] = 'test-maker-terminal-eye-slave', _0x56a466['config']['retry'] = 0x5dc, _0x56a466['config']['silent'] = !![], _0x56a466['connectTo'](_0x5c7def, () => {
                    this['messenger'] = _0x56a466['of'][_0x5c7def], _0x56a466['of'][_0x5c7def]['on']('connect', async () => {
                        this['messenger']['emit']('message', { 'action': 'connected' }), setInterval(() => {
                            this['updateResourcesUsage']();
                        }, 0x3e8), this['messenger']['on']('message', _0x446d9a => {
                            _0x446d9a['action'] === 'updateScreenInfo' && (process['stdout']['columns'] = _0x446d9a['columns'] - 0x4, process['stdout']['rows'] = _0x446d9a['rows']);
                        });
                    });
                });
            }
            async ['updateResourcesUsage']() {
                this['messenger']['emit']('message', {
                    'action': 'updateResourcesUsage',
                    'cpu': await (0x0, _0x42b0af['getCpuAverageLoad'])(),
                    'memory': process['memoryUsage']()['heapUsed']
                });
            }
        }
        const _0x35a68b = global['__TEST_MAKER__IPC_SLAVE__'], _0x156d8a = global['__TEST_MAKER__IPC_SLAVE__'] = _0x35a68b ? _0x35a68b : new _0xccb0b3();
    },
    0x10895: (_0x4dbf90, _0x4d179a, _0x4176f3) => {
        _0x4176f3['r'](_0x4d179a), _0x4176f3['d'](_0x4d179a, { 'getCpuAverageLoad': () => _0x50af44 });
        var _0x3adc49 = _0x4176f3(0x5615), _0x443b22 = _0x4176f3['n'](_0x3adc49);
        function _0x91cc09() {
            let _0x57d0bd = 0x0, _0x4bad06 = 0x0, _0x56069c = _0x443b22()['cpus']();
            for (let _0x5912cf = 0x0, _0xdcd66f = _0x56069c['length']; _0x5912cf < _0xdcd66f; _0x5912cf++) {
                let _0x173b23 = _0x56069c[_0x5912cf];
                for (let _0xe3da8 in _0x173b23['times']) {
                    _0x4bad06 += _0x173b23['times'][_0xe3da8];
                }
                _0x57d0bd += _0x173b23['times']['idle'];
            }
            return {
                'idle': _0x57d0bd / _0x56069c['length'],
                'total': _0x4bad06 / _0x56069c['length']
            };
        }
        const _0x1819d7 = _0x346500 => {
            if (_0x346500 && _0x346500['length'] >= 0x1) {
                const _0x2b53a7 = _0x346500['reduce']((_0x4159ad, _0x56e5bd) => _0x4159ad + _0x56e5bd, 0x0);
                return _0x2b53a7 / _0x346500['length'];
            } else
                return 0x0;
        };
        function _0x50af44(_0xdcdfa4 = 0x3e8, _0x30fcf7 = 0x64) {
            return new Promise((_0x46003e, _0x102cbe) => {
                const _0x2f4a77 = ~~(_0xdcdfa4 / _0x30fcf7);
                _0x2f4a77 <= 0x1 && _0x102cbe(new Error('Error:\x20interval\x20to\x20small'));
                let _0x5a047e = 0x0, _0x3e04f1 = [];
                const _0x1f620f = _0x91cc09();
                let _0x19c336 = setInterval(() => {
                    _0x5a047e >= _0x2f4a77 && (clearInterval(_0x19c336), _0x46003e(~~(_0x1819d7(_0x3e04f1) * 0x64)));
                    const _0x42b5da = _0x91cc09(), _0x4ef5fc = _0x42b5da['total'] - _0x1f620f['total'], _0x4dcedd = _0x42b5da['idle'] - _0x1f620f['idle'];
                    _0x3e04f1[_0x5a047e] = 0x1 - _0x4dcedd / _0x4ef5fc, _0x5a047e++;
                }, _0x30fcf7);
            });
        }
    }
};
;
