// test-maker/dist/feature-loader.js
(function webpackUniversalModuleDefinition(_0x3eca0e, _0x809550) {
    if (typeof exports === 'object' && typeof module === 'object')
        module['exports'] = _0x809550();
    else {
        if (typeof define === 'function' && define['amd'])
            define([], _0x809550);
        else {
            var _0xdd8678 = _0x809550();
            for (var _0x436836 in _0xdd8678)
                (typeof exports === 'object' ? exports : _0x3eca0e)[_0x436836] = _0xdd8678[_0x436836];
        }
    }
}(this, () => {
    return ((() => {
        'use strict';
        var _0x144f26 = {
                0x148b0: (_0x387d80, _0x36c370, _0x400e23) => {
                    _0x400e23['r'](_0x36c370), _0x400e23['d'](_0x36c370, {
                        'FeatureLoader': () => _0x4fcc40,
                        'default': () => _0x334c8f
                    });
                    var _0x375e99 = _0x400e23(0x1145c), _0x55db26 = _0x400e23['n'](_0x375e99), _0x30aa81 = _0x400e23(0x3942), _0x18a880 = _0x400e23(0x17fbe), _0x538076 = _0x400e23['n'](_0x18a880), _0x484dd8 = _0x400e23(0xf3e4);
                    ;
                    const _0x55741c = require('typescript');
                    var _0x3441af = _0x400e23(0x3b1);
                    ;
                    var _0x30ba20;
                    (function (_0x4dc1ad) {
                        _0x4dc1ad['Feature'] = 'Feature', _0x4dc1ad['Background'] = 'Background', _0x4dc1ad['Example'] = 'Example', _0x4dc1ad['Scenario'] = 'Scenario', _0x4dc1ad['ScenarioOutline'] = 'ScenarioOutline', _0x4dc1ad['Given'] = 'Given', _0x4dc1ad['When'] = 'When', _0x4dc1ad['And'] = 'And', _0x4dc1ad['Then'] = 'Then', _0x4dc1ad['But'] = 'But';
                    }(_0x30ba20 || (_0x30ba20 = {})));
                    ;
                    const _0x373a99 = [_0x30ba20['Feature']], _0x1e6c92 = [
                            _0x30ba20['Scenario'],
                            _0x30ba20['ScenarioOutline'],
                            _0x30ba20['Example'],
                            _0x30ba20['Background']
                        ], _0x5d6899 = [
                            _0x30ba20['Given'],
                            _0x30ba20['And'],
                            _0x30ba20['When'],
                            _0x30ba20['Then'],
                            _0x30ba20['But']
                        ], _0x1d7577 = [
                            ..._0x373a99,
                            ..._0x1e6c92,
                            ..._0x5d6899
                        ];
                    ;
                    var _0x2f0c5c;
                    (function (_0x267ac4) {
                        _0x267ac4['TestMaker'] = 'Test\x20Maker', _0x267ac4['BeforeAll'] = 'Before\x20All', _0x267ac4['AfterAll'] = 'After\x20All', _0x267ac4['Feature'] = 'Feature', _0x267ac4['BeforeEachFeature'] = 'Before\x20Each\x20Feature', _0x267ac4['BeforeFeature'] = 'Before\x20Feature', _0x267ac4['AfterFeature'] = 'After\x20Feature', _0x267ac4['AfterEachFeature'] = 'After\x20Each\x20Feature', _0x267ac4['Scenario'] = 'Scenario', _0x267ac4['BeforeEachScenario'] = 'Before\x20Each\x20Scenario', _0x267ac4['BeforeScenario'] = 'Before\x20Scenario', _0x267ac4['AfterScenario'] = 'After\x20Scenario', _0x267ac4['AfterEachScenario'] = 'After\x20Each\x20Scenario', _0x267ac4['Step'] = 'Step', _0x267ac4['BeforeEachStep'] = 'Before\x20Each\x20Step\x20In\x20Scenario', _0x267ac4['AfterEachStep'] = 'After\x20Each\x20Step\x20In\x20Scenario', _0x267ac4['SupStep'] = 'Sup\x20Step';
                    }(_0x2f0c5c || (_0x2f0c5c = {})));
                    ;
                    function _0x111e0b(_0xfe1c6f, _0x253a61 = !![]) {
                        if (_0x253a61)
                            console['log']('\x0aExiting\x20Test\x20Maker\x20Process.\x0a');
                        process['exit'](_0xfe1c6f);
                    }
                    var _0x3dc94a = _0x400e23(0x16925), _0x42ea0e = _0x400e23(0xccb6);
                    ;
                    function _0x23392f(_0xc5dc9b) {
                        return (0x0, _0x3dc94a['nanoid'])(_0xc5dc9b);
                    }
                    async function _0x206105(_0x3f9e72) {
                        return await (0x0, _0x42ea0e['nanoid'])(_0x3f9e72);
                    }
                    function _0x4dadf8(_0x24c414, _0x888b4d) {
                        return (0x0, _0x3dc94a['customAlphabet'])(_0x24c414, _0x888b4d)();
                    }
                    async function _0x40789c(_0x4692f5, _0x3ad934) {
                        return await (0x0, _0x42ea0e['customAlphabet'])(_0x4692f5, _0x3ad934)();
                    }
                    var _0x13114b = _0x400e23(0x9e6);
                    ;
                    class _0x4fcc40 {
                        constructor(_0x22eb50, _0x57f63f, _0x43b23f) {
                            this['source'] = '', this['tsSource'] = '', this['traverse'] = _0x30aa81['traverse'], this['feature'] = {
                                'id': '',
                                'name': '',
                                'didRun': ![],
                                'keyword': _0x30ba20['Feature'],
                                'only': ![],
                                'skip': ![],
                                'isLast': !![],
                                'sourcePath': '',
                                'distPath': '',
                                'OriginalSource': '',
                                'location': {
                                    'start': {
                                        'line': 0x1,
                                        'column': 0x1
                                    },
                                    'end': {
                                        'line': 0x1,
                                        'column': 0x1
                                    }
                                },
                                'adapters': [],
                                'clients': [],
                                'scenarios': []
                            }, this['features'] = [], this['scenarios'] = [], this['steps'] = [], this['sourcePath'] = '', this['distPath'] = '', this['source'] = _0x22eb50, this['map'] = _0x43b23f, this['compiler'] = _0x57f63f, this['renderCodeFrame'] = _0x57f63f['getOptions']()['renderCodeFrame'];
                        }
                        async ['load']() {
                            const _0x1b8428 = this['compiler']['async']();
                            try {
                                this['stateManager'] = this['compiler']['getOptions']()['stateManager'], this['configManger'] = this['compiler']['getOptions']()['configManager'];
                                const _0x48a677 = Object['values'](this['compiler']['getOptions']()['whiteList'])['map'](_0x97a187 => _0x3441af['resolve'](_0x97a187)), _0x575c1e = this['compiler']['_compiler']['outputPath'];
                                this['sourcePath'] = _0x3441af['normalizeTrim'](this['compiler']['resourcePath']), this['distPath'] = _0x3441af['joinSafe'](_0x3441af['changeExt'](_0x3441af['joinSafe'](_0x575c1e, this['compiler']['_module']['rawRequest']), 'js'));
                                for (let _0x13723f of this['stateManager']['state']['runner']['featuresAst']['values']()) {
                                    if (_0x13723f && _0x13723f['sourcePath'] === this['sourcePath']) {
                                        this['stateManager']['state']['runner']['featuresAst']['delete'](_0x13723f['name']);
                                        break;
                                    }
                                }
                                this['stateManager']['state']['onlyFeatures']['delete'](this['sourcePath']), this['stateManager']['state']['onlyScenarios']['delete'](this['sourcePath']), this['stateManager']['state']['onlySteps']['delete'](this['sourcePath']);
                                if (!_0x48a677['includes'](this['sourcePath']))
                                    this['stateManager']['state']['runner']['featuresToRun']['delete'](this['sourcePath']);
                                else {
                                    this['configManger']['configuration']['build']['cleanUp']['tests'] && this['stateManager']['state']['runner']['runs'] > 0x1 && (console['info']('-\x20Deleting\x20Old\x20Spec\x20File:\x20' + this['distPath'] + '\x0a'), await _0x538076()['remove'](_0x3441af['resolve'](this['distPath'])));
                                    const _0x1ec5e6 = (await _0x538076()['readFile'](this['sourcePath']))['toString'](), _0x1769d5 = _0x55741c['transpileModule'](_0x1ec5e6, {
                                            'compilerOptions': {
                                                'module': _0x55741c['ModuleKind']['ESNext'],
                                                'target': _0x55741c['ScriptTarget']['ESNext'],
                                                'importsNotUsedAsValue': _0x55741c['ImportsNotUsedAsValues']['Preserve'],
                                                'preserveValueImports': !![],
                                                'preserveConstEnums': !![],
                                                'removeComments': ![],
                                                'pretty': ![]
                                            }
                                        })['outputText'], _0x45686c = _0x13114b['parsePatch'](_0x13114b['createPatch']('file', _0x1ec5e6, _0x1769d5, '', '')), _0x555a81 = _0x45686c[0x0]['hunks'];
                                    for (let _0x2d807b = 0x0; _0x2d807b < _0x555a81['length']; ++_0x2d807b) {
                                        let _0x500919 = 0x0;
                                        const _0x33a80e = _0x555a81[_0x2d807b];
                                        _0x33a80e['lines'] = _0x33a80e['lines']['map'](_0x4c4fa8 => {
                                            if (_0x4c4fa8 === '-')
                                                return _0x500919++, '\x20';
                                            return _0x4c4fa8;
                                        }), _0x33a80e['newLines'] += _0x500919;
                                        for (let _0xb3478c = _0x2d807b + 0x1; _0xb3478c < _0x555a81['length']; ++_0xb3478c) {
                                            _0x555a81[_0xb3478c]['newStart'] += _0x500919;
                                        }
                                    }
                                    this['tsSource'] = _0x13114b['applyPatch'](_0x1ec5e6, _0x45686c), await this['parseCode'](), this['feature']['OriginalSource'] = this['tsSource'];
                                    this['feature']['name'] && (await this['checkDuplicateSameFileFeatures'](), await this['checkDuplicateMultipleFeatures'](), await this['checkDuplicateScenarios'](), this['setIsLast']());
                                    if (this['feature']['name'])
                                        this['stateManager']['state']['runner']['featuresAst']['set'](this['feature']['name'], this['feature']), this['stateManager']['state']['runner']['featuresToRun']['set'](this['sourcePath'], this['feature']);
                                    else {
                                        for (let _0x1ffdce of this['stateManager']['state']['runner']['featuresAst']['values']()) {
                                            if (_0x1ffdce['sourcePath'] === this['sourcePath'])
                                                this['stateManager']['state']['runner']['featuresAst']['delete'](_0x1ffdce['name']);
                                        }
                                        this['stateManager']['state']['runner']['featuresToRun']['delete'](this['sourcePath']);
                                    }
                                }
                                return this['stateManager']['state']['filesCache']['set'](this['sourcePath'], (await _0x538076()['readFile'](this['sourcePath']))['toString']()), _0x1b8428(null, this['source'], this['map']);
                            } catch (_0x3fd5b5) {
                                return console['error']('Spec\x20Compilation\x20Error\x20for\x20File:\x20' + this['sourcePath'] + '\x0a'), _0x1b8428(_0x3fd5b5);
                            }
                        }
                        async ['parseCode']() {
                            let _0x25f6ea = ![];
                            this['ast'] = (0x0, _0x484dd8['parseScript'])(this['tsSource']['toString'](), {
                                'module': !![],
                                'loc': !![],
                                'raw': !![],
                                'ranges': !![]
                            }), this['traverse'](this['ast'], {
                                'enter': _0x504b23 => {
                                    var _0x1a9b45, _0x4470fa, _0x4a7fdd, _0x2b31e2, _0x2ad374;
                                    if (this['isFeature'](_0x504b23)) {
                                        const _0x4362fa = _0x504b23['arguments'][0x0]['quasis'] ? (_0x1a9b45 = _0x504b23['arguments'][0x0]['quasis'][0x0]) === null || _0x1a9b45 === void 0x0 ? void 0x0 : _0x1a9b45['value']['cooked'] : _0x504b23['arguments'][0x0]['value'], _0x1ca21d = {
                                                'id': _0x23392f(),
                                                'didRun': ![],
                                                'name': this['removeQuotes'](_0x4362fa),
                                                'only': ![],
                                                'skip': ![],
                                                'isLast': !![],
                                                'keyword': _0x30ba20['Feature'],
                                                'sourcePath': this['sourcePath'],
                                                'distPath': this['distPath'],
                                                'location': _0x504b23['loc'],
                                                'scenarios': this['scenarios']['reverse'](),
                                                'adapters': [],
                                                'clients': []
                                            };
                                        this['scenarios'] = [], this['features']['push'](JSON['parse'](JSON['stringify'](_0x1ca21d))), this['feature'] = _0x1ca21d, _0x25f6ea && (this['stateManager']['state']['onlyFeatures']['set'](this['sourcePath'], this['feature']['name']), _0x1ca21d['only'] = !![], _0x25f6ea = ![]);
                                    }
                                    if (this['isScenario'](_0x504b23)) {
                                        const _0x53bd6a = _0x504b23['arguments'][0x0]['quasis'] ? (_0x4470fa = _0x504b23['arguments'][0x0]['quasis'][0x0]) === null || _0x4470fa === void 0x0 ? void 0x0 : _0x4470fa['value']['cooked'] : _0x504b23['arguments'][0x0]['value'], _0x17934d = {
                                                'id': _0x23392f(),
                                                'name': this['removeQuotes'](_0x53bd6a),
                                                'keyword': _0x30ba20['Scenario'],
                                                'only': ![],
                                                'skip': ![],
                                                'isLast': ![],
                                                'sourcePath': this['sourcePath'],
                                                'distPath': this['distPath'],
                                                'adapters': [],
                                                'clients': [],
                                                'location': (_0x4a7fdd = _0x504b23['callee']) === null || _0x4a7fdd === void 0x0 ? void 0x0 : _0x4a7fdd['property']['loc'],
                                                'steps': JSON['parse'](JSON['stringify'](this['steps']['reverse']()))
                                            };
                                        this['scenarios']['push'](_0x17934d), this['steps'] = [], _0x25f6ea && (this['stateManager']['state']['onlyScenarios']['add'](this['sourcePath']), _0x17934d['only'] = !![], _0x25f6ea = ![]);
                                    }
                                    if (this['isStep'](_0x504b23)) {
                                        const _0x18412d = _0x504b23['arguments'][0x0]['quasis'] ? (_0x2b31e2 = _0x504b23['arguments'][0x0]['quasis'][0x0]) === null || _0x2b31e2 === void 0x0 ? void 0x0 : _0x2b31e2['value']['cooked'] : _0x504b23['arguments'][0x0]['value'], _0x476032 = {
                                                'id': _0x23392f(),
                                                'name': this['removeQuotes'](_0x18412d),
                                                'keyword': _0x30ba20['Given'],
                                                'only': ![],
                                                'skip': ![],
                                                'isLast': ![],
                                                'sourcePath': this['sourcePath'],
                                                'distPath': this['distPath'],
                                                'adapters': [],
                                                'clients': [],
                                                'location': (_0x2ad374 = _0x504b23['callee']) === null || _0x2ad374 === void 0x0 ? void 0x0 : _0x2ad374['property']['loc']
                                            };
                                        this['steps']['push'](_0x476032), _0x25f6ea && (this['stateManager']['state']['onlySteps']['add'](this['sourcePath']), _0x476032['only'] = !![], _0x25f6ea = ![]);
                                    }
                                    if (this['isOnly'](_0x504b23))
                                        _0x25f6ea = !![];
                                }
                            });
                        }
                        ['isFeature'](_0x26b692) {
                            var _0x4e072d;
                            return _0x26b692['type'] === 'CallExpression' && _0x373a99['includes']((_0x4e072d = _0x26b692['callee']) === null || _0x4e072d === void 0x0 ? void 0x0 : _0x4e072d['name']);
                        }
                        ['isScenario'](_0x5aec66) {
                            var _0x4caa2c, _0x57dd78;
                            return _0x5aec66['type'] === 'CallExpression' && _0x1e6c92['includes']((_0x57dd78 = (_0x4caa2c = _0x5aec66['callee']) === null || _0x4caa2c === void 0x0 ? void 0x0 : _0x4caa2c['property']) === null || _0x57dd78 === void 0x0 ? void 0x0 : _0x57dd78['name']);
                        }
                        ['isStep'](_0x14140e) {
                            var _0x470a96, _0x5807a5;
                            return _0x14140e['type'] === 'CallExpression' && _0x5d6899['includes']((_0x5807a5 = (_0x470a96 = _0x14140e['callee']) === null || _0x470a96 === void 0x0 ? void 0x0 : _0x470a96['property']) === null || _0x5807a5 === void 0x0 ? void 0x0 : _0x5807a5['name']);
                        }
                        ['isHook'](_0x58272b) {
                            var _0x5680c3, _0x115438;
                            const _0x4b719b = [
                                _0x2f0c5c['BeforeAll'],
                                _0x2f0c5c['AfterAll'],
                                _0x2f0c5c['BeforeEachFeature'],
                                _0x2f0c5c['BeforeFeature'],
                                _0x2f0c5c['AfterFeature'],
                                _0x2f0c5c['AfterEachFeature'],
                                _0x2f0c5c['BeforeEachScenario'],
                                _0x2f0c5c['BeforeScenario'],
                                _0x2f0c5c['AfterScenario'],
                                _0x2f0c5c['AfterEachScenario'],
                                _0x2f0c5c['BeforeEachStep'],
                                _0x2f0c5c['AfterEachStep']
                            ];
                            return _0x58272b['type'] === 'CallExpression' && _0x4b719b['includes']((_0x115438 = (_0x5680c3 = _0x58272b['callee']) === null || _0x5680c3 === void 0x0 ? void 0x0 : _0x5680c3['property']) === null || _0x115438 === void 0x0 ? void 0x0 : _0x115438['name']);
                        }
                        ['isOnly'](_0x1809f0) {
                            var _0x4596f4, _0x5b1665;
                            return _0x1809f0['type'] === 'CallExpression' && ((_0x5b1665 = (_0x4596f4 = _0x1809f0['callee']) === null || _0x4596f4 === void 0x0 ? void 0x0 : _0x4596f4['property']) === null || _0x5b1665 === void 0x0 ? void 0x0 : _0x5b1665['name']) === 'only';
                        }
                        ['removeQuotes'](_0x14e4b6) {
                            return _0x14e4b6['replace'](/(\n|\r\n)/g, '\x0a');
                            let _0x5c4dc7 = _0x14e4b6;
                            const _0x5da3d3 = [
                                '\x22',
                                '\x27',
                                '`'
                            ];
                            if (_0x5da3d3['includes'](_0x5c4dc7['charAt'](0x0)) && _0x5da3d3['includes'](_0x5c4dc7['charAt'](_0x5c4dc7['length'] - 0x1)))
                                _0x5c4dc7 = _0x5c4dc7['substr'](0x1, _0x5c4dc7['length'] - 0x2);
                            return _0x5c4dc7;
                        }
                        async ['checkDuplicateMultipleFeatures']() {
                            const _0xf39c75 = this['stateManager']['state']['runner']['featuresAst']['get'](this['feature']['name']), _0x2684c1 = _0xf39c75 === null || _0xf39c75 === void 0x0 ? void 0x0 : _0xf39c75['OriginalSource'];
                            if (!_0x2684c1)
                                return;
                            _0xf39c75 && (console['error'](_0x55db26()['redBright']('Duplicate\x20Features\x20Files\x20found\x0a')), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Feature') + ':\x20' + this['feature']['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], this['feature']['location']['start']['line'], this['feature']['location']['start']['column']))['ansi'] + '\x0a'), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Feature') + ':\x20' + _0xf39c75['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + _0xf39c75['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](_0x2684c1 || '', _0xf39c75['location']['start']['line'], _0xf39c75['location']['start']['column']))['ansi'] + '\x0a'), _0x111e0b(0x1));
                        }
                        async ['checkDuplicateSameFileFeatures']() {
                            if (this['features']['length'] > 0x1) {
                                console['error'](_0x55db26()['redBright']('A\x20file\x20can\x20have\x20only\x20one\x20Feature\x0a'));
                                for (let _0x592337 of this['features']) {
                                    console['log']('\x20\x20' + _0x55db26()['cyanBright']('Feature') + ':\x20' + this['feature']['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x592337['location']['start']['line'], _0x592337['location']['start']['column']))['ansi'] + '\x0a');
                                }
                                _0x111e0b(0x1);
                            }
                        }
                        async ['checkDuplicateScenarios']() {
                            if (!this['feature']['scenarios']['length'])
                                console['error'](_0x55db26()['redBright']('A\x20Feature\x20must\x20have\x20at\x20least\x20one\x20Scenario\x0a')), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Feature') + ':\x20' + this['feature']['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], this['feature']['location']['start']['line'], this['feature']['location']['start']['column']))['ansi'] + '\x0a'), _0x111e0b(0x1);
                            else {
                                const _0x399423 = new Map();
                                for (let _0x4ca300 of this['feature']['scenarios']) {
                                    const _0x5762f7 = _0x399423['get'](_0x4ca300['name']);
                                    _0x5762f7 && (console['error'](_0x55db26()['redBright']('A\x20Feature\x20can\x20not\x20have\x20duplicate\x20Scenarios\x20names\x0a')), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Scenario') + ':\x20' + _0x5762f7['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x5762f7['location']['start']['line'], _0x5762f7['location']['start']['column']))['ansi'] + '\x0a'), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Scenario') + ':\x20' + _0x4ca300['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x4ca300['location']['start']['line'], _0x4ca300['location']['start']['column']))['ansi'] + '\x0a'), _0x111e0b(0x1)), _0x399423['set'](_0x4ca300['name'], _0x4ca300), await this['checkDuplicateSteps'](_0x4ca300);
                                }
                            }
                        }
                        async ['checkDuplicateSteps'](_0x118cc3) {
                            if (!_0x118cc3['steps']['length'])
                                console['error'](_0x55db26()['redBright']('A\x20Scenario\x20must\x20have\x20at\x20least\x20one\x20Step\x0a')), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Scenario') + ':\x20' + _0x118cc3['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x118cc3['location']['start']['line'], _0x118cc3['location']['start']['column']))['ansi'] + '\x0a'), _0x111e0b(0x1);
                            else {
                                const _0x463a7e = new Map();
                                for (let _0x5a7813 of _0x118cc3['steps']) {
                                    const _0x2658cc = _0x463a7e['get'](_0x5a7813['name']);
                                    _0x2658cc && (console['error'](_0x55db26()['redBright']('A\x20Scenario\x20can\x20not\x20have\x20duplicate\x20Steps\x20names\x0a')), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Step') + ':\x20' + _0x2658cc['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x2658cc['location']['start']['line'], _0x2658cc['location']['start']['column']))['ansi'] + '\x0a'), console['log']('\x20\x20' + _0x55db26()['cyanBright']('Step') + ':\x20' + _0x5a7813['name'] + '\x0a\x20\x20' + _0x55db26()['cyanBright']('File') + ':\x20' + this['sourcePath'] + '\x0a'), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x5a7813['location']['start']['line'], _0x5a7813['location']['start']['column']))['ansi'] + '\x0a'), _0x111e0b(0x1)), _0x463a7e['set'](_0x5a7813['name'], _0x5a7813);
                                }
                            }
                        }
                        async ['checkNames'](_0x5b9dfd) {
                            await this['logNoName'](_0x5b9dfd['name'], _0x5b9dfd['keyword'], _0x5b9dfd['location']);
                            for (let _0x11bb3f of _0x5b9dfd['scenarios']) {
                                await this['logNoName'](_0x11bb3f['name'], _0x11bb3f['keyword'], _0x11bb3f['location']);
                                for (let _0x39e142 of _0x11bb3f['steps']) {
                                    await this['logNoName'](_0x39e142['name'], _0x39e142['keyword'], _0x39e142['location']);
                                }
                            }
                        }
                        async ['logNoName'](_0xc3743e, _0x1ce849, _0x355200) {
                            !_0xc3743e['trim']() && (console['error'](_0x55db26()['redBright'](_0x1ce849 + '\x20name\x20can\x20not\x20be\x20empty\x0a')), console['log']((await this['renderCodeFrame'](this['tsSource'], _0x355200['start']['line'], _0x355200['start']['column']))['ansi'] + '\x0a')), _0x111e0b(0x1);
                        }
                        ['setIsLast']() {
                            const _0x2fcd72 = this['feature']['scenarios'][this['feature']['scenarios']['length'] - 0x1];
                            _0x2fcd72['isLast'] = !![];
                            const _0x45ff17 = _0x2fcd72['steps'][_0x2fcd72['steps']['length'] - 0x1];
                            _0x45ff17['isLast'] = !![];
                        }
                    }
                    function _0x334c8f(_0x3c327e, _0xbd91e9) {
                        const _0x214321 = new _0x4fcc40(_0x3c327e, this, _0xbd91e9);
                        return _0x214321['load']();
                    }
                },
                0x9a43: _0xdbf28e => {
                    _0xdbf28e['exports'] = require('assert');
                },
                0x5629: _0x181429 => {
                    _0x181429['exports'] = require('constants');
                },
                0x17e1: _0x2ab282 => {
                    _0x2ab282['exports'] = require('crypto');
                },
                0xdf3b: _0xc7cd64 => {
                    _0xc7cd64['exports'] = require('fs');
                },
                0x5615: _0x14dd01 => {
                    _0x14dd01['exports'] = require('os');
                },
                0x11569: _0xf50e25 => {
                    _0xf50e25['exports'] = require('path');
                },
                0x31ed: _0x38ce55 => {
                    _0x38ce55['exports'] = require('stream');
                },
                0x129c0: _0x59acc8 => {
                    _0x59acc8['exports'] = require('tty');
                },
                0x1206d: _0x28b93e => {
                    _0x28b93e['exports'] = require('util');
                }
            }, _0x108027 = {};
        function _0x1ff818(_0x2065f4) {
            var _0x20209a = _0x108027[_0x2065f4];
            if (_0x20209a !== undefined)
                return _0x20209a['exports'];
            var _0x16dda0 = _0x108027[_0x2065f4] = {
                'id': _0x2065f4,
                'loaded': ![],
                'exports': {}
            };
            return _0x144f26[_0x2065f4](_0x16dda0, _0x16dda0['exports'], _0x1ff818), _0x16dda0['loaded'] = !![], _0x16dda0['exports'];
        }
        _0x1ff818['m'] = _0x144f26, _0x1ff818['x'] = () => {
            var _0x38dedb = _0x1ff818['O'](undefined, [
                0xee,
                0x182,
                0x240
            ], () => _0x1ff818(0x148b0));
            return _0x38dedb = _0x1ff818['O'](_0x38dedb), _0x38dedb;
        }, ((() => {
            var _0x6647ea = [];
            _0x1ff818['O'] = (_0x17908d, _0x345de0, _0x15389a, _0x29fd0d) => {
                if (_0x345de0) {
                    _0x29fd0d = _0x29fd0d || 0x0;
                    for (var _0x19b8ca = _0x6647ea['length']; _0x19b8ca > 0x0 && _0x6647ea[_0x19b8ca - 0x1][0x2] > _0x29fd0d; _0x19b8ca--)
                        _0x6647ea[_0x19b8ca] = _0x6647ea[_0x19b8ca - 0x1];
                    _0x6647ea[_0x19b8ca] = [
                        _0x345de0,
                        _0x15389a,
                        _0x29fd0d
                    ];
                    return;
                }
                var _0x4061d8 = Infinity;
                for (var _0x19b8ca = 0x0; _0x19b8ca < _0x6647ea['length']; _0x19b8ca++) {
                    var [_0x345de0, _0x15389a, _0x29fd0d] = _0x6647ea[_0x19b8ca], _0x3270a0 = !![];
                    for (var _0x232908 = 0x0; _0x232908 < _0x345de0['length']; _0x232908++) {
                        if ((_0x29fd0d & 0x1 === 0x0 || _0x4061d8 >= _0x29fd0d) && Object['keys'](_0x1ff818['O'])['every'](_0x41aa3b => _0x1ff818['O'][_0x41aa3b](_0x345de0[_0x232908])))
                            _0x345de0['splice'](_0x232908--, 0x1);
                        else {
                            _0x3270a0 = ![];
                            if (_0x29fd0d < _0x4061d8)
                                _0x4061d8 = _0x29fd0d;
                        }
                    }
                    if (_0x3270a0) {
                        _0x6647ea['splice'](_0x19b8ca--, 0x1);
                        var _0x3e02f1 = _0x15389a();
                        if (_0x3e02f1 !== undefined)
                            _0x17908d = _0x3e02f1;
                    }
                }
                return _0x17908d;
            };
        })()), ((() => {
            _0x1ff818['n'] = _0x583a50 => {
                var _0x1acc84 = _0x583a50 && _0x583a50['__esModule'] ? () => _0x583a50['default'] : () => _0x583a50;
                return _0x1ff818['d'](_0x1acc84, { 'a': _0x1acc84 }), _0x1acc84;
            };
        })()), ((() => {
            _0x1ff818['d'] = (_0x4b5725, _0x13cda6) => {
                for (var _0x117f3a in _0x13cda6) {
                    _0x1ff818['o'](_0x13cda6, _0x117f3a) && !_0x1ff818['o'](_0x4b5725, _0x117f3a) && Object['defineProperty'](_0x4b5725, _0x117f3a, {
                        'enumerable': !![],
                        'get': _0x13cda6[_0x117f3a]
                    });
                }
            };
        })()), ((() => {
            _0x1ff818['f'] = {}, _0x1ff818['e'] = _0x59688f => {
                return Promise['all'](Object['keys'](_0x1ff818['f'])['reduce']((_0xa7bd30, _0x3b7255) => {
                    return _0x1ff818['f'][_0x3b7255](_0x59688f, _0xa7bd30), _0xa7bd30;
                }, []));
            };
        })()), ((() => {
            _0x1ff818['u'] = _0x1fd5c1 => {
                return '' + _0x1fd5c1 + '.js';
            };
        })()), ((() => {
            _0x1ff818['o'] = (_0x1d4ec9, _0x3573d7) => Object['prototype']['hasOwnProperty']['call'](_0x1d4ec9, _0x3573d7);
        })()), ((() => {
            _0x1ff818['r'] = _0x443aaf => {
                typeof Symbol !== 'undefined' && Symbol['toStringTag'] && Object['defineProperty'](_0x443aaf, Symbol['toStringTag'], { 'value': 'Module' }), Object['defineProperty'](_0x443aaf, '__esModule', { 'value': !![] });
            };
        })()), ((() => {
            _0x1ff818['nmd'] = _0x2883e0 => {
                _0x2883e0['paths'] = [];
                if (!_0x2883e0['children'])
                    _0x2883e0['children'] = [];
                return _0x2883e0;
            };
        })()), ((() => {
            var _0x16e924 = { 0x144: 0x1 };
            _0x1ff818['O']['require'] = _0x575fd0 => _0x16e924[_0x575fd0];
            var _0x9dd95e = _0x392ff8 => {
                var _0x40d6ea = _0x392ff8['modules'], _0x1f0292 = _0x392ff8['ids'], _0x3edb8d = _0x392ff8['runtime'];
                for (var _0x5b869d in _0x40d6ea) {
                    _0x1ff818['o'](_0x40d6ea, _0x5b869d) && (_0x1ff818['m'][_0x5b869d] = _0x40d6ea[_0x5b869d]);
                }
                if (_0x3edb8d)
                    _0x3edb8d(_0x1ff818);
                for (var _0x21f3f7 = 0x0; _0x21f3f7 < _0x1f0292['length']; _0x21f3f7++)
                    _0x16e924[_0x1f0292[_0x21f3f7]] = 0x1;
                _0x1ff818['O']();
            };
            _0x1ff818['f']['require'] = (_0x4c68c4, _0x2a05e2) => {
                if (!_0x16e924[_0x4c68c4]) {
                    if (!![])
                        _0x9dd95e(require('./' + _0x1ff818['u'](_0x4c68c4)));
                    else
                        _0x16e924[_0x4c68c4] = 0x1;
                }
            };
        })()), ((() => {
            var _0x3b9206 = _0x1ff818['x'];
            _0x1ff818['x'] = () => {
                return _0x1ff818['e'](0xee), _0x1ff818['e'](0x182), _0x1ff818['e'](0x240), _0x3b9206();
            };
        })());
        var _0x579d88 = _0x1ff818['x']();
        return _0x579d88;
    })());
}));
